{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pipeline for RNA-Seq analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Mark : Autogenerated with openBIS-ELN/LIMS (System Cell, Don't delete!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "matplotlib.rcParams['font.family'] = 'Arial'\n",
    "\n",
    "import seaborn as sns\n",
    "sns.set(color_codes=True)\n",
    "\n",
    "from pyCRAC.Parsers import GTF2\n",
    "from collections import defaultdict\n",
    "from sklearn.cluster import KMeans\n",
    "from scipy.stats import pearsonr,spearmanr\n",
    "\n",
    "from sg_ngs_tools import *\n",
    "import obis_connector\n",
    "\n",
    "%matplotlib inline\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variable - File Name (System Cell, Don't delete!, Update if renamed)\n",
    "fileName='RNA_pipeline.ipynb'\n",
    "workspace='ngs/sander/phage_infection_time_course_e_coli_rnaseq'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variable : Parents used by the automaticaly generated result dataset (System Cell, Don't delete!)\n",
    "resultDatasetParents=[\"20211105180528381-5936\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Connect to openBIS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using hardcoded url instead: http://openngs2.bio.ed.ac.uk:8443\n"
     ]
    }
   ],
   "source": [
    "import obis_connector\n",
    "o = obis_connector.connect_to_openbis(url='http://openngs2.bio.ed.ac.uk:8443')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Datasets Information and Download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing of fastq files\n",
      "Skipping the existing: RNASeq_data_analyses.ipynb\n",
      "Skipping the existing: RNASeq_data_analyses.ipynb.html\n",
      "Skipping the existing: a10_rep2_+T4_t15_1.fastq\n",
      "Skipping the existing: a10_rep2_+T4_t15_2.fastq\n",
      "Skipping the existing: a11_rep2_+T4_t20_1.fastq\n",
      "Skipping the existing: a11_rep2_+T4_t20_2.fastq\n",
      "Skipping the existing: a12_rep2_control_t20_1.fastq\n",
      "Skipping the existing: a12_rep2_control_t20_2.fastq\n",
      "Skipping the existing: a13_rep3_control_t-5_1.fastq\n",
      "Skipping the existing: a13_rep3_control_t-5_2.fastq\n",
      "Skipping the existing: a14_rep3_+T4_t2_1.fastq\n",
      "Skipping the existing: a14_rep3_+T4_t2_2.fastq\n",
      "Skipping the existing: a15_rep3_+T4_t5_1.fastq\n",
      "Skipping the existing: a15_rep3_+T4_t5_2.fastq\n",
      "Skipping the existing: a16_rep3_+T4_t10_1.fastq\n",
      "Skipping the existing: a16_rep3_+T4_t10_2.fastq\n",
      "Skipping the existing: a17_rep3_+T4_t20_1.fastq\n",
      "Skipping the existing: a17_rep3_+T4_t20_2.fastq\n",
      "Skipping the existing: a18_rep3_control_t20_1.fastq\n",
      "Skipping the existing: a18_rep3_control_t20_2.fastq\n",
      "Skipping the existing: a1_rep1_control_t-5_1.fastq\n",
      "Skipping the existing: a1_rep1_control_t-5_2.fastq\n",
      "Skipping the existing: a2_rep1_+T4_t5_1.fastq\n",
      "Skipping the existing: a2_rep1_+T4_t5_2.fastq\n",
      "Skipping the existing: a3_rep1_+T4_t10_1.fastq\n",
      "Skipping the existing: a3_rep1_+T4_t10_2.fastq\n",
      "Skipping the existing: a4_rep1_+T4_t15_1.fastq\n",
      "Skipping the existing: a4_rep1_+T4_t15_2.fastq\n",
      "Skipping the existing: a5_rep1_+T4_t20_1.fastq\n",
      "Skipping the existing: a5_rep1_+T4_t20_2.fastq\n",
      "Skipping the existing: a6_rep1_control_t20_1.fastq\n",
      "Skipping the existing: a6_rep1_control_t20_2.fastq\n",
      "Skipping the existing: a7_rep2_control_t-5_1.fastq\n",
      "Skipping the existing: a7_rep2_control_t-5_2.fastq\n",
      "Skipping the existing: a8_rep2_+T4_t5_1.fastq\n",
      "Skipping the existing: a8_rep2_+T4_t5_2.fastq\n",
      "Skipping the existing: a9_rep2_+T4_t10_1.fastq\n",
      "Skipping the existing: a9_rep2_+T4_t10_2.fastq\n",
      "Skipping the existing: python_env.1636130848.2266634.txt\n",
      "Files downloaded to: 20211105180528381-5936\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['20211105180528381-5936/original/DEFAULT/RNASeq_data_analyses.ipynb',\n",
       " '20211105180528381-5936/original/DEFAULT/RNASeq_data_analyses.ipynb.html',\n",
       " '20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/python_env.1636130848.2266634.txt']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Dataset ''20211105180528381-5936:\n",
    "s0 = o.get_object('20211105164909360-5934')\n",
    "#print(s0.summary)\n",
    "\n",
    "ds0 = o.get_dataset('20211105180528381-5936')\n",
    "#print(ds0.summary)\n",
    "\n",
    "print(ds0.props.name)\n",
    "# files0 = ds0.download(files=['original/DEFAULT/file1']) # to download selected, ds0.file_list lists present\n",
    "\n",
    "files0 = ds0.download()\n",
    "\n",
    "###################\n",
    "# you can use files object to easily access the full file paths, for example:\n",
    "# files0.path('filename')\n",
    "# files0.paths lists all paths\n",
    "# files0.glob('*.txt') lists paths matching glob pattern\n",
    "# files0.help() to see commands help\n",
    "###################\n",
    "\n",
    "files0.paths"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process your data here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "###### MAKE A VARIABLE THAT CONTAINS A LIST OF ALL THE FILES THAT YOU WANT TO UPLOAD #######\n",
    "analysisfile = list()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Checkig /homes2/openngs/openngs/scripts2\n",
      "All content is recorded as:\n",
      "\n",
      "## v.1.0.0...origin/v.1.0.0\n",
      "commit 5f1c9cb1a751385062cefdf1a2bc7c298402cb08\n",
      "Author: openngs2 <bio_rdm@ed.ac.uk>\n",
      "Date:   Tue May 3 00:29:23 2022 +0100\n",
      "\n",
      "    removed python2 checking\n",
      "\n",
      "Checkig /homes2/openngs/openngs/annotation_files2\n",
      "All content is recorded as:\n",
      "\n",
      "## master...origin/master\n",
      "commit 641054659eb96a75fca1a051a5f5e956c4b04dfc\n",
      "Author: openngs2 <bio_rdm@ed.ac.uk>\n",
      "Date:   Mon Jul 4 18:56:49 2022 +0100\n",
      "\n",
      "    Fixed PROKKA GTF file\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Please only use scripts and annotations files from these folders so they will be tracked with the analysis\n",
    "SCRIPTS=/homes2/openngs/openngs/scripts2\n",
    "ANNOTATIONS=/homes2/openngs/openngs/annotation_files2\n",
    "\n",
    "$SCRIPTS/check_dir_status.sh $SCRIPTS || exit 1\n",
    "$SCRIPTS/check_dir_status.sh $ANNOTATIONS || exit 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['20211105180528381-5936/original/DEFAULT/RNASeq_data_analyses.ipynb',\n",
       " '20211105180528381-5936/original/DEFAULT/RNASeq_data_analyses.ipynb.html',\n",
       " '20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/python_env.1636130848.2266634.txt']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files0.paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1.fastq',\n",
       " '20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2.fastq']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files0.glob(\"*.fastq\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## the pipeline for processing the raw fastq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP1: run the fastqc on all the fastq files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1.fastq\n",
      "finished the analysis of ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_fastq=\"./20211105180528381-5936/original/DEFAULT/\"\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/*.fastq | grep -Eo 'a[0-9]+_.*t.*[0-9]+' | sort | uniq )\n",
    "R1_file=$(ls ./20211105180528381-5936/original/DEFAULT/*.fastq |grep -Eo 'a[0-9]+_.*t.*[0-9]+_1'| sort | uniq )\n",
    "R2_file=$(ls ./20211105180528381-5936/original/DEFAULT/*.fastq |grep -Eo 'a[0-9]+_.*t.*[0-9]+_2'| sort | uniq )\n",
    "#echo $R1_file\n",
    "for ID in $fastq_name\n",
    "    do\n",
    "    fastqc -o ./yan_results/fastqc_result/ -q -t 10 $path_fastq$ID\".fastq\"\n",
    "    echo 'finished the analysis of' $path_fastq$ID\".fastq\"\n",
    "    done\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP2: run the cutadapt on all the fastq files\n",
    "( need to distinguish forward file and reverse file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt version 4.0\n",
      "\n",
      "Copyright (C) 2010-2022 Marcel Martin <marcel.martin@scilifelab.se>\n",
      "\n",
      "cutadapt removes adapter sequences from high-throughput sequencing reads.\n",
      "\n",
      "Usage:\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\n",
      "\n",
      "For paired-end reads:\n",
      "    cutadapt -a ADAPT1 -A ADAPT2 [options] -o out1.fastq -p out2.fastq in1.fastq in2.fastq\n",
      "\n",
      "Replace \"ADAPTER\" with the actual sequence of your 3' adapter. IUPAC wildcard\n",
      "characters are supported. All reads from input.fastq will be written to\n",
      "output.fastq with the adapter sequence removed. Adapter matching is\n",
      "error-tolerant. Multiple adapter sequences can be given (use further -a\n",
      "options), but only the best-matching adapter will be removed.\n",
      "\n",
      "Input may also be in FASTA format. Compressed input and output is supported and\n",
      "auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for\n",
      "standard input/output. Without the -o option, output is sent to standard output.\n",
      "\n",
      "Citation:\n",
      "\n",
      "Marcel Martin. Cutadapt removes adapter sequences from high-throughput\n",
      "sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.\n",
      "http://dx.doi.org/10.14806/ej.17.1.200\n",
      "\n",
      "Run \"cutadapt --help\" to see all command-line options.\n",
      "See https://cutadapt.readthedocs.io/ for full documentation.\n",
      "\n",
      "Options:\n",
      "  -h, --help            Show this help message and exit\n",
      "  --version             Show version number and exit\n",
      "  --debug               Print debug log. Use twice to also print DP matrices\n",
      "  -j CORES, --cores CORES\n",
      "                        Number of CPU cores to use. Use 0 to auto-detect.\n",
      "                        Default: 1\n",
      "\n",
      "Finding adapters:\n",
      "  Parameters -a, -g, -b specify adapters to be removed from each read (or from\n",
      "  R1 if data is paired-end. If specified multiple times, only the best\n",
      "  matching adapter is trimmed (but see the --times option). Use notation\n",
      "  'file:FILE' to read adapter sequences from a FASTA file.\n",
      "\n",
      "  -a ADAPTER, --adapter ADAPTER\n",
      "                        Sequence of an adapter ligated to the 3' end (paired\n",
      "                        data: of the first read). The adapter and subsequent\n",
      "                        bases are trimmed. If a '$' character is appended\n",
      "                        ('anchoring'), the adapter is only found if it is a\n",
      "                        suffix of the read.\n",
      "  -g ADAPTER, --front ADAPTER\n",
      "                        Sequence of an adapter ligated to the 5' end (paired\n",
      "                        data: of the first read). The adapter and any preceding\n",
      "                        bases are trimmed. Partial matches at the 5' end are\n",
      "                        allowed. If a '^' character is prepended ('anchoring'),\n",
      "                        the adapter is only found if it is a prefix of the read.\n",
      "  -b ADAPTER, --anywhere ADAPTER\n",
      "                        Sequence of an adapter that may be ligated to the 5' or\n",
      "                        3' end (paired data: of the first read). Both types of\n",
      "                        matches as described under -a and -g are allowed. If the\n",
      "                        first base of the read is part of the match, the\n",
      "                        behavior is as with -g, otherwise as with -a. This\n",
      "                        option is mostly for rescuing failed library\n",
      "                        preparations - do not use if you know which end your\n",
      "                        adapter was ligated to!\n",
      "  -e E, --error-rate E, --errors E\n",
      "                        Maximum allowed error rate (if 0 <= E < 1), or absolute\n",
      "                        number of errors for full-length adapter match (if E is\n",
      "                        an integer >= 1). Error rate = no. of errors divided by\n",
      "                        length of matching region. Default: 0.1 (10%)\n",
      "  --no-indels           Allow only mismatches in alignments. Default: allow both\n",
      "                        mismatches and indels\n",
      "  -n COUNT, --times COUNT\n",
      "                        Remove up to COUNT adapters from each read. Default: 1\n",
      "  -O MINLENGTH, --overlap MINLENGTH\n",
      "                        Require MINLENGTH overlap between read and adapter for\n",
      "                        an adapter to be found. Default: 3\n",
      "  --match-read-wildcards\n",
      "                        Interpret IUPAC wildcards in reads. Default: False\n",
      "  -N, --no-match-adapter-wildcards\n",
      "                        Do not interpret IUPAC wildcards in adapters.\n",
      "  --action {trim,retain,mask,lowercase,none}\n",
      "                        What to do if a match was found. trim: trim adapter and\n",
      "                        up- or downstream sequence; retain: trim, but retain\n",
      "                        adapter; mask: replace with 'N' characters; lowercase:\n",
      "                        convert to lowercase; none: leave unchanged. Default:\n",
      "                        trim\n",
      "  --rc, --revcomp       Check both the read and its reverse complement for\n",
      "                        adapter matches. If match is on reverse-complemented\n",
      "                        version, output that one. Default: check only read\n",
      "\n",
      "Additional read modifications:\n",
      "  -u LENGTH, --cut LENGTH\n",
      "                        Remove bases from each read (first read only if paired).\n",
      "                        If LENGTH is positive, remove bases from the beginning.\n",
      "                        If LENGTH is negative, remove bases from the end. Can be\n",
      "                        used twice if LENGTHs have different signs. This is\n",
      "                        applied *before* adapter trimming.\n",
      "  --nextseq-trim 3'CUTOFF\n",
      "                        NextSeq-specific quality trimming (each read). Trims\n",
      "                        also dark cycles appearing as high-quality G bases.\n",
      "  -q [5'CUTOFF,]3'CUTOFF, --quality-cutoff [5'CUTOFF,]3'CUTOFF\n",
      "                        Trim low-quality bases from 5' and/or 3' ends of each\n",
      "                        read before adapter removal. Applied to both reads if\n",
      "                        data is paired. If one value is given, only the 3' end\n",
      "                        is trimmed. If two comma-separated cutoffs are given,\n",
      "                        the 5' end is trimmed with the first cutoff, the 3' end\n",
      "                        with the second.\n",
      "  --quality-base N      Assume that quality values in FASTQ are encoded as\n",
      "                        ascii(quality + N). This needs to be set to 64 for some\n",
      "                        old Illumina FASTQ files. Default: 33\n",
      "  --length LENGTH, -l LENGTH\n",
      "                        Shorten reads to LENGTH. Positive values remove bases at\n",
      "                        the end while negative ones remove bases at the\n",
      "                        beginning. This and the following modifications are\n",
      "                        applied after adapter trimming.\n",
      "  --trim-n              Trim N's on ends of reads.\n",
      "  --length-tag TAG      Search for TAG followed by a decimal number in the\n",
      "                        description field of the read. Replace the decimal\n",
      "                        number with the correct length of the trimmed read. For\n",
      "                        example, use --length-tag 'length=' to correct fields\n",
      "                        like 'length=123'.\n",
      "  --strip-suffix STRIP_SUFFIX\n",
      "                        Remove this suffix from read names if present. Can be\n",
      "                        given multiple times.\n",
      "  -x PREFIX, --prefix PREFIX\n",
      "                        Add this prefix to read names. Use {name} to insert the\n",
      "                        name of the matching adapter.\n",
      "  -y SUFFIX, --suffix SUFFIX\n",
      "                        Add this suffix to read names; can also include {name}\n",
      "  --rename TEMPLATE     Rename reads using TEMPLATE containing variables such as\n",
      "                        {id}, {adapter_name} etc. (see documentation)\n",
      "  --zero-cap, -z        Change negative quality values to zero.\n",
      "\n",
      "Filtering of processed reads:\n",
      "  Filters are applied after above read modifications. Paired-end reads are\n",
      "  always discarded pairwise (see also --pair-filter).\n",
      "\n",
      "  -m LEN[:LEN2], --minimum-length LEN[:LEN2]\n",
      "                        Discard reads shorter than LEN. Default: 0\n",
      "  -M LEN[:LEN2], --maximum-length LEN[:LEN2]\n",
      "                        Discard reads longer than LEN. Default: no limit\n",
      "  --max-n COUNT         Discard reads with more than COUNT 'N' bases. If COUNT\n",
      "                        is a number between 0 and 1, it is interpreted as a\n",
      "                        fraction of the read length.\n",
      "  --max-expected-errors ERRORS, --max-ee ERRORS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                        Discard reads whose expected number of errors (computed\n",
      "                        from quality values) exceeds ERRORS.\n",
      "  --discard-trimmed, --discard\n",
      "                        Discard reads that contain an adapter. Use also -O to\n",
      "                        avoid discarding too many randomly matching reads.\n",
      "  --discard-untrimmed, --trimmed-only\n",
      "                        Discard reads that do not contain an adapter.\n",
      "  --discard-casava      Discard reads that did not pass CASAVA filtering (header\n",
      "                        has :Y:).\n",
      "\n",
      "Output:\n",
      "  --quiet               Print only error messages.\n",
      "  --report {full,minimal}\n",
      "                        Which type of report to print: 'full' or 'minimal'.\n",
      "                        Default: full\n",
      "  --json FILE           Dump report in JSON format to FILE\n",
      "  -o FILE, --output FILE\n",
      "                        Write trimmed reads to FILE. FASTQ or FASTA format is\n",
      "                        chosen depending on input. Summary report is sent to\n",
      "                        standard output. Use '{name}' for demultiplexing (see\n",
      "                        docs). Default: write to standard output\n",
      "  --fasta               Output FASTA to standard output even on FASTQ input.\n",
      "  -Z                    Use compression level 1 for gzipped output files\n",
      "                        (faster, but uses more space)\n",
      "  --info-file FILE      Write information about each read and its adapter\n",
      "                        matches into FILE. See the documentation for the file\n",
      "                        format.\n",
      "  -r FILE, --rest-file FILE\n",
      "                        When the adapter matches in the middle of a read, write\n",
      "                        the rest (after the adapter) to FILE.\n",
      "  --wildcard-file FILE  When the adapter has N wildcard bases, write adapter\n",
      "                        bases matching wildcard positions to FILE. (Inaccurate\n",
      "                        with indels.)\n",
      "  --too-short-output FILE\n",
      "                        Write reads that are too short (according to length\n",
      "                        specified by -m) to FILE. Default: discard reads\n",
      "  --too-long-output FILE\n",
      "                        Write reads that are too long (according to length\n",
      "                        specified by -M) to FILE. Default: discard reads\n",
      "  --untrimmed-output FILE\n",
      "                        Write reads that do not contain any adapter to FILE.\n",
      "                        Default: output to same file as trimmed reads\n",
      "\n",
      "Paired-end options:\n",
      "  The -A/-G/-B/-U/-Q options work like their lowercase counterparts, but are\n",
      "  applied to R2 (second read in pair)\n",
      "\n",
      "  -A ADAPTER            3' adapter to be removed from R2\n",
      "  -G ADAPTER            5' adapter to be removed from R2\n",
      "  -B ADAPTER            5'/3 adapter to be removed from R2\n",
      "  -U LENGTH             Remove LENGTH bases from R2\n",
      "  -Q [5'CUTOFF,]3'CUTOFF\n",
      "                        Quality-trimming cutoff for R2. Default: same as for R1\n",
      "  -p FILE, --paired-output FILE\n",
      "                        Write R2 to FILE.\n",
      "  --pair-adapters       Treat adapters given with -a/-A etc. as pairs. Either\n",
      "                        both or none are removed from each read pair.\n",
      "  --pair-filter {any,both,first}\n",
      "                        Which of the reads in a paired-end read have to match\n",
      "                        the filtering criterion in order for the pair to be\n",
      "                        filtered. Default: any\n",
      "  --interleaved         Read and/or write interleaved paired-end reads.\n",
      "  --untrimmed-paired-output FILE\n",
      "                        Write second read in a pair to this FILE when no adapter\n",
      "                        was found. Use with --untrimmed-output. Default: output\n",
      "                        to same file as trimmed reads\n",
      "  --too-short-paired-output FILE\n",
      "                        Write second read in a pair to this file if pair is too\n",
      "                        short.\n",
      "  --too-long-paired-output FILE\n",
      "                        Write second read in a pair to this file if pair is too\n",
      "                        long.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cutadapt --h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_1.fastq ./20211105180528381-5936/original/DEFAULT/a10_rep2_+T4_t15_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 30.99 s (1 µs/read; 49.66 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         25,647,520\n",
      "  Read 1 with adapter:               1,257,717 (4.9%)\n",
      "  Read 2 with adapter:               1,111,663 (4.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               2,327 (0.0%)\n",
      "Pairs written (passing filters):    25,645,193 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,694,256,000 bp\n",
      "  Read 1: 3,847,128,000 bp\n",
      "  Read 2: 3,847,128,000 bp\n",
      "Total written (filtered):  7,666,979,026 bp (99.6%)\n",
      "  Read 1: 3,831,697,747 bp\n",
      "  Read 2: 3,835,281,279 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 1257717 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 58.7%\n",
      "  C: 13.8%\n",
      "  G: 10.2%\n",
      "  T: 17.3%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t495113\t400742.5\t0\t495113\n",
      "4\t159822\t100185.6\t0\t159822\n",
      "5\t56415\t25046.4\t0\t56415\n",
      "6\t30661\t6261.6\t0\t30661\n",
      "7\t24237\t1565.4\t0\t24237\n",
      "8\t22144\t391.4\t0\t22144\n",
      "9\t21836\t97.8\t0\t21151 685\n",
      "10\t21228\t24.5\t1\t20144 1084\n",
      "11\t20440\t6.1\t1\t19560 880\n",
      "12\t19296\t1.5\t1\t18582 714\n",
      "13\t18527\t0.4\t1\t17811 716\n",
      "14\t17479\t0.1\t1\t16745 734\n",
      "15\t17079\t0.0\t1\t16429 650\n",
      "16\t16498\t0.0\t1\t15816 682\n",
      "17\t15666\t0.0\t1\t15023 643\n",
      "18\t15444\t0.0\t1\t14723 720 1\n",
      "19\t14840\t0.0\t1\t14193 638 9\n",
      "20\t14535\t0.0\t2\t13715 673 147\n",
      "21\t13554\t0.0\t2\t12759 647 148\n",
      "22\t13176\t0.0\t2\t12436 610 130\n",
      "23\t12418\t0.0\t2\t11712 578 128\n",
      "24\t11444\t0.0\t2\t10785 537 122\n",
      "25\t11233\t0.0\t2\t10569 568 96\n",
      "26\t10662\t0.0\t2\t10039 501 122\n",
      "27\t10153\t0.0\t2\t9512 529 112\n",
      "28\t9795\t0.0\t2\t9229 466 100\n",
      "29\t9412\t0.0\t2\t8826 487 95 4\n",
      "30\t8781\t0.0\t3\t8166 483 85 47\n",
      "31\t8701\t0.0\t3\t8091 443 116 51\n",
      "32\t8233\t0.0\t3\t7685 412 92 44\n",
      "33\t7806\t0.0\t3\t7319 384 67 36\n",
      "34\t7261\t0.0\t3\t6715 285 203 58\n",
      "35\t6825\t0.0\t3\t0 6345 303 177\n",
      "36\t6395\t0.0\t3\t0 4 6026 365\n",
      "37\t6279\t0.0\t3\t0 4 5926 349\n",
      "38\t5837\t0.0\t3\t0 6 5496 335\n",
      "39\t5830\t0.0\t3\t0 4 5468 267 91\n",
      "40\t5299\t0.0\t4\t0 0 6 5002 291\n",
      "41\t5100\t0.0\t4\t0 0 4 4773 323\n",
      "42\t4949\t0.0\t4\t0 0 3 4638 308\n",
      "43\t4387\t0.0\t4\t0 0 4 4126 257\n",
      "44\t4107\t0.0\t4\t0 0 2 3849 256\n",
      "45\t3940\t0.0\t4\t0 0 3 3306 631\n",
      "46\t3704\t0.0\t4\t0 0 0 3137 567\n",
      "47\t3496\t0.0\t4\t0 0 2 2928 566\n",
      "48\t3298\t0.0\t4\t0 0 0 2745 553\n",
      "49\t3128\t0.0\t4\t0 0 2 2616 498 12\n",
      "50\t2940\t0.0\t5\t0 0 2 2460 419 59\n",
      "51\t2929\t0.0\t5\t0 0 0 2459 398 72\n",
      "52\t2639\t0.0\t5\t0 0 3 2136 435 65\n",
      "53\t2386\t0.0\t5\t0 0 2 1960 359 65\n",
      "54\t2239\t0.0\t5\t0 0 0 1801 381 57\n",
      "55\t1978\t0.0\t5\t0 0 2 1596 318 62\n",
      "56\t1817\t0.0\t5\t0 0 1 1447 321 48\n",
      "57\t1616\t0.0\t5\t0 0 0 1306 267 43\n",
      "58\t1541\t0.0\t5\t0 0 1 1218 269 52 1\n",
      "59\t1535\t0.0\t5\t0 0 0 1202 274 58 1\n",
      "60\t1451\t0.0\t6\t0 0 0 1107 259 67 18\n",
      "61\t1422\t0.0\t6\t0 0 0 1073 274 48 27\n",
      "62\t1333\t0.0\t6\t0 0 0 993 252 62 26\n",
      "63\t1209\t0.0\t6\t0 0 1 897 207 79 25\n",
      "64\t1103\t0.0\t6\t0 0 1 774 236 60 32\n",
      "65\t908\t0.0\t6\t0 0 0 650 193 47 18\n",
      "66\t823\t0.0\t6\t0 0 0 581 166 50 26\n",
      "67\t719\t0.0\t6\t0 0 0 510 153 37 19\n",
      "68\t696\t0.0\t6\t0 0 0 500 139 47 10\n",
      "69\t698\t0.0\t6\t0 0 0 509 137 36 16\n",
      "70\t642\t0.0\t6\t0 0 1 458 133 35 15\n",
      "71\t542\t0.0\t6\t0 0 0 392 117 23 10\n",
      "72\t557\t0.0\t6\t0 0 0 407 108 34 8\n",
      "73\t523\t0.0\t6\t0 0 0 372 105 30 16\n",
      "74\t508\t0.0\t6\t0 0 0 352 112 31 13\n",
      "75\t398\t0.0\t6\t0 0 0 298 72 20 8\n",
      "76\t368\t0.0\t6\t0 0 0 264 79 20 5\n",
      "77\t316\t0.0\t6\t0 0 0 237 55 17 7\n",
      "78\t307\t0.0\t6\t0 0 0 215 60 25 7\n",
      "79\t288\t0.0\t6\t0 0 0 198 72 12 6\n",
      "80\t227\t0.0\t6\t0 0 0 161 42 20 4\n",
      "81\t201\t0.0\t6\t0 0 0 148 36 12 5\n",
      "82\t232\t0.0\t6\t0 0 0 173 41 15 3\n",
      "83\t235\t0.0\t6\t0 0 0 183 39 12 1\n",
      "84\t204\t0.0\t6\t0 0 0 137 46 14 7\n",
      "85\t175\t0.0\t6\t0 0 0 124 35 13 3\n",
      "86\t137\t0.0\t6\t0 0 0 85 36 11 5\n",
      "87\t163\t0.0\t6\t0 0 0 114 35 7 7\n",
      "88\t136\t0.0\t6\t0 0 0 86 33 14 3\n",
      "89\t109\t0.0\t6\t0 0 0 82 14 7 6\n",
      "90\t93\t0.0\t6\t0 0 0 63 20 10\n",
      "91\t76\t0.0\t6\t0 0 0 55 12 5 4\n",
      "92\t77\t0.0\t6\t0 0 0 48 22 4 3\n",
      "93\t84\t0.0\t6\t0 0 0 59 9 13 3\n",
      "94\t63\t0.0\t6\t0 0 0 46 13 4\n",
      "95\t69\t0.0\t6\t0 0 0 47 16 3 3\n",
      "96\t62\t0.0\t6\t0 0 0 42 13 4 3\n",
      "97\t54\t0.0\t6\t0 0 0 33 13 5 3\n",
      "98\t42\t0.0\t6\t0 0 0 21 12 6 3\n",
      "99\t29\t0.0\t6\t0 0 0 16 7 4 2\n",
      "100\t28\t0.0\t6\t0 0 0 20 3 3 2\n",
      "101\t17\t0.0\t6\t0 0 0 8 3 2 4\n",
      "102\t14\t0.0\t6\t0 0 0 9 3 1 1\n",
      "103\t16\t0.0\t6\t0 0 0 9 4 2 1\n",
      "104\t13\t0.0\t6\t0 0 0 9 3 0 1\n",
      "105\t14\t0.0\t6\t0 0 0 7 2 4 1\n",
      "106\t13\t0.0\t6\t0 0 0 9 1 2 1\n",
      "107\t18\t0.0\t6\t0 0 0 12 1 2 3\n",
      "108\t6\t0.0\t6\t0 0 0 2 3 1\n",
      "109\t14\t0.0\t6\t0 0 0 6 3 2 3\n",
      "110\t10\t0.0\t6\t0 0 0 8 1 0 1\n",
      "111\t5\t0.0\t6\t0 0 0 2 3\n",
      "112\t12\t0.0\t6\t0 0 0 8 2 2\n",
      "113\t20\t0.0\t6\t0 0 0 8 4 4 4\n",
      "114\t12\t0.0\t6\t0 0 0 3 3 5 1\n",
      "115\t6\t0.0\t6\t0 0 0 3 2 0 1\n",
      "116\t3\t0.0\t6\t0 0 0 1 2\n",
      "117\t8\t0.0\t6\t0 0 0 3 1 3 1\n",
      "118\t40\t0.0\t6\t0 0 0 17 13 7 3\n",
      "119\t8\t0.0\t6\t0 0 0 4 2 1 1\n",
      "120\t19\t0.0\t6\t0 0 0 13 3 2 1\n",
      "121\t10\t0.0\t6\t0 0 0 4 5 0 1\n",
      "122\t9\t0.0\t6\t0 0 0 6 0 2 1\n",
      "123\t5\t0.0\t6\t0 0 0 1 3 1\n",
      "124\t18\t0.0\t6\t0 0 0 11 5 1 1\n",
      "125\t13\t0.0\t6\t0 0 0 8 3 2\n",
      "126\t12\t0.0\t6\t0 0 0 6 3 1 2\n",
      "127\t36\t0.0\t6\t0 0 0 16 11 6 3\n",
      "128\t34\t0.0\t6\t0 0 0 17 13 3 1\n",
      "129\t7\t0.0\t6\t0 0 0 3 3 1\n",
      "130\t33\t0.0\t6\t0 0 0 17 9 2 5\n",
      "131\t28\t0.0\t6\t0 0 0 17 6 2 3\n",
      "132\t12\t0.0\t6\t0 0 0 7 4 1\n",
      "133\t24\t0.0\t6\t0 0 0 14 8 2\n",
      "134\t26\t0.0\t6\t0 0 0 18 6 1 1\n",
      "135\t89\t0.0\t6\t0 0 0 59 19 10 1\n",
      "136\t21\t0.0\t6\t0 0 0 8 3 7 3\n",
      "137\t26\t0.0\t6\t0 0 0 14 7 2 3\n",
      "138\t42\t0.0\t6\t0 0 0 26 13 2 1\n",
      "139\t33\t0.0\t6\t0 0 0 21 10 2\n",
      "140\t113\t0.0\t6\t0 0 0 71 26 12 4\n",
      "141\t31\t0.0\t6\t0 0 0 21 8 1 1\n",
      "142\t45\t0.0\t6\t0 0 0 26 15 3 1\n",
      "143\t50\t0.0\t6\t0 0 0 28 17 3 2\n",
      "144\t113\t0.0\t6\t0 0 0 62 29 15 7\n",
      "145\t34\t0.0\t6\t0 0 0 17 13 4\n",
      "146\t66\t0.0\t6\t0 0 0 39 19 7 1\n",
      "147\t17\t0.0\t6\t0 0 0 7 7 2 1\n",
      "148\t4\t0.0\t6\t0 0 0 1 2 1\n",
      "149\t71\t0.0\t6\t0 0 0 45 23 2 1\n",
      "150\t1037\t0.0\t6\t0 0 0 564 288 106 79\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 1111663 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.8%\n",
      "  C: 29.0%\n",
      "  G: 29.8%\n",
      "  T: 10.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t426608\t400742.5\t0\t426608\n",
      "4\t134738\t100185.6\t0\t134738\n",
      "5\t50088\t25046.4\t0\t50088\n",
      "6\t32690\t6261.6\t0\t32690\n",
      "7\t25935\t1565.4\t0\t25935\n",
      "8\t23115\t391.4\t0\t23115\n",
      "9\t22830\t97.8\t0\t22038 792\n",
      "10\t22527\t24.5\t1\t21210 1317\n",
      "11\t21278\t6.1\t1\t20286 992\n",
      "12\t20438\t1.5\t1\t19570 868\n",
      "13\t19373\t0.4\t1\t18612 761\n",
      "14\t18665\t0.1\t1\t17948 717\n",
      "15\t17639\t0.0\t1\t16911 728\n",
      "16\t17201\t0.0\t1\t16407 794\n",
      "17\t16622\t0.0\t1\t15926 696\n",
      "18\t15738\t0.0\t1\t14735 1001 2\n",
      "19\t15567\t0.0\t1\t14587 960 20\n",
      "20\t15022\t0.0\t2\t13734 1129 159\n",
      "21\t14637\t0.0\t2\t13233 1245 159\n",
      "22\t13652\t0.0\t2\t12362 1146 144\n",
      "23\t13284\t0.0\t2\t12047 1098 139\n",
      "24\t12523\t0.0\t2\t11316 1067 140\n",
      "25\t11529\t0.0\t2\t10464 928 137\n",
      "26\t11323\t0.0\t2\t10007 1157 159\n",
      "27\t10753\t0.0\t2\t9521 1054 178\n",
      "28\t10242\t0.0\t2\t9006 1058 177 1\n",
      "29\t9883\t0.0\t2\t8631 1074 177 1\n",
      "30\t9574\t0.0\t3\t8364 991 165 54\n",
      "31\t8870\t0.0\t3\t7777 899 142 52\n",
      "32\t8770\t0.0\t3\t7730 825 169 46\n",
      "33\t8302\t0.0\t3\t7287 854 119 42\n",
      "34\t7862\t0.0\t3\t6854 727 231 50\n",
      "35\t7286\t0.0\t3\t0 6345 810 131\n",
      "36\t6898\t0.0\t3\t1 5958 710 229\n",
      "37\t6432\t0.0\t3\t0 0 5786 646\n",
      "38\t6178\t0.0\t3\t0 1 5595 581 1\n",
      "39\t5333\t0.0\t3\t0 1 1 5313 18\n",
      "40\t630\t0.0\t4\t0 0 0 619 11\n",
      "41\t4831\t0.0\t4\t0 0 0 472 4359\n",
      "42\t4644\t0.0\t4\t0 0 0 644 4000\n",
      "43\t1206\t0.0\t4\t0 0 0 122 1084\n",
      "44\t452\t0.0\t4\t0 0 0 0 452\n",
      "45\t379\t0.0\t4\t0 0 0 1 378\n",
      "46\t106\t0.0\t4\t0 0 0 0 106\n",
      "47\t8\t0.0\t4\t0 0 0 0 8\n",
      "48\t2\t0.0\t4\t0 0 0 0 2\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a11_rep2_+T4_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 40.99 s (2 µs/read; 36.10 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         24,665,596\n",
      "  Read 1 with adapter:                 832,401 (3.4%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Read 2 with adapter:                 767,348 (3.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               2,806 (0.0%)\n",
      "Pairs written (passing filters):    24,662,790 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,399,678,800 bp\n",
      "  Read 1: 3,699,839,400 bp\n",
      "  Read 2: 3,699,839,400 bp\n",
      "Total written (filtered):  7,387,044,218 bp (99.8%)\n",
      "  Read 1: 3,692,794,845 bp\n",
      "  Read 2: 3,694,249,373 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 832401 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 41.8%\n",
      "  C: 18.0%\n",
      "  G: 15.1%\n",
      "  T: 25.0%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t460487\t385399.9\t0\t460487\n",
      "4\t132087\t96350.0\t0\t132087\n",
      "5\t37360\t24087.5\t0\t37360\n",
      "6\t14703\t6021.9\t0\t14703\n",
      "7\t9884\t1505.5\t0\t9884\n",
      "8\t8667\t376.4\t0\t8667\n",
      "9\t8328\t94.1\t0\t7894 434\n",
      "10\t8072\t23.5\t1\t7506 566\n",
      "11\t7662\t5.9\t1\t7245 417\n",
      "12\t7113\t1.5\t1\t6752 361\n",
      "13\t6711\t0.4\t1\t6412 299\n",
      "14\t6465\t0.1\t1\t6144 321\n",
      "15\t6272\t0.0\t1\t5977 295\n",
      "16\t5811\t0.0\t1\t5580 231\n",
      "17\t5620\t0.0\t1\t5380 240\n",
      "18\t5395\t0.0\t1\t5098 295 2\n",
      "19\t5450\t0.0\t1\t5164 280 6\n",
      "20\t5150\t0.0\t2\t4793 284 73\n",
      "21\t4820\t0.0\t2\t4449 295 76\n",
      "22\t4656\t0.0\t2\t4353 257 46\n",
      "23\t4380\t0.0\t2\t4080 249 51\n",
      "24\t4019\t0.0\t2\t3748 221 50\n",
      "25\t3804\t0.0\t2\t3531 219 54\n",
      "26\t3711\t0.0\t2\t3431 232 48\n",
      "27\t3710\t0.0\t2\t3448 213 49\n",
      "28\t3298\t0.0\t2\t3065 190 42 1\n",
      "29\t3199\t0.0\t2\t2980 176 40 3\n",
      "30\t3039\t0.0\t3\t2812 176 33 18\n",
      "31\t2840\t0.0\t3\t2593 196 31 20\n",
      "32\t2624\t0.0\t3\t2392 182 29 21\n",
      "33\t2687\t0.0\t3\t2430 197 35 25\n",
      "34\t2365\t0.0\t3\t2192 103 50 20\n",
      "35\t2201\t0.0\t3\t0 2031 140 30\n",
      "36\t2284\t0.0\t3\t0 2118 143 23\n",
      "37\t2168\t0.0\t3\t0 1996 107 64 1\n",
      "38\t1999\t0.0\t3\t0 3 1868 127 1\n",
      "39\t2043\t0.0\t3\t0 1 1888 96 58\n",
      "40\t1897\t0.0\t4\t0 0 6 1785 106\n",
      "41\t1820\t0.0\t4\t0 0 7 1702 111\n",
      "42\t1665\t0.0\t4\t0 0 7 1536 122\n",
      "43\t1544\t0.0\t4\t0 0 5 1438 101\n",
      "44\t1420\t0.0\t4\t0 0 5 1317 98\n",
      "45\t1248\t0.0\t4\t0 0 3 966 279\n",
      "46\t1243\t0.0\t4\t0 0 0 927 316\n",
      "47\t1209\t0.0\t4\t0 0 2 915 292\n",
      "48\t1060\t0.0\t4\t0 0 1 833 226\n",
      "49\t1163\t0.0\t4\t0 0 1 912 246 4\n",
      "50\t931\t0.0\t5\t0 0 0 682 216 33\n",
      "51\t958\t0.0\t5\t0 0 1 725 193 39\n",
      "52\t938\t0.0\t5\t0 0 0 663 238 37\n",
      "53\t937\t0.0\t5\t0 0 1 717 190 29\n",
      "54\t793\t0.0\t5\t0 0 0 572 187 34\n",
      "55\t648\t0.0\t5\t0 0 2 446 170 30\n",
      "56\t642\t0.0\t5\t0 0 0 442 167 33\n",
      "57\t597\t0.0\t5\t0 0 0 428 137 32\n",
      "58\t546\t0.0\t5\t0 0 2 392 126 26\n",
      "59\t539\t0.0\t5\t0 0 0 378 133 28\n",
      "60\t529\t0.0\t6\t0 0 0 336 138 40 15\n",
      "61\t522\t0.0\t6\t0 0 0 320 143 43 16\n",
      "62\t459\t0.0\t6\t0 0 0 289 129 30 11\n",
      "63\t429\t0.0\t6\t0 0 0 260 116 31 22\n",
      "64\t375\t0.0\t6\t0 0 0 226 102 35 12\n",
      "65\t362\t0.0\t6\t0 0 0 217 102 27 16\n",
      "66\t357\t0.0\t6\t0 0 0 208 104 27 18\n",
      "67\t278\t0.0\t6\t0 0 0 170 76 21 11\n",
      "68\t234\t0.0\t6\t0 0 0 138 60 23 13\n",
      "69\t238\t0.0\t6\t0 0 0 146 64 22 6\n",
      "70\t232\t0.0\t6\t0 0 0 128 69 22 13\n",
      "71\t233\t0.0\t6\t0 0 0 143 56 24 10\n",
      "72\t214\t0.0\t6\t0 0 0 113 67 23 11\n",
      "73\t207\t0.0\t6\t0 0 0 123 50 22 12\n",
      "74\t206\t0.0\t6\t0 0 0 117 61 12 16\n",
      "75\t160\t0.0\t6\t0 0 0 82 54 19 5\n",
      "76\t150\t0.0\t6\t0 0 0 88 43 15 4\n",
      "77\t132\t0.0\t6\t0 0 0 82 32 15 3\n",
      "78\t120\t0.0\t6\t0 0 0 83 24 10 3\n",
      "79\t108\t0.0\t6\t0 0 0 70 26 8 4\n",
      "80\t101\t0.0\t6\t0 0 0 56 30 14 1\n",
      "81\t87\t0.0\t6\t0 0 0 47 27 11 2\n",
      "82\t118\t0.0\t6\t0 0 0 69 39 8 2\n",
      "83\t85\t0.0\t6\t0 0 0 49 21 7 8\n",
      "84\t82\t0.0\t6\t0 0 0 46 27 7 2\n",
      "85\t92\t0.0\t6\t0 0 0 54 26 9 3\n",
      "86\t65\t0.0\t6\t0 0 0 26 28 6 5\n",
      "87\t68\t0.0\t6\t0 0 0 34 13 10 11\n",
      "88\t76\t0.0\t6\t0 0 0 46 17 7 6\n",
      "89\t52\t0.0\t6\t0 0 0 25 22 5\n",
      "90\t30\t0.0\t6\t0 0 0 16 13 1\n",
      "91\t42\t0.0\t6\t0 0 0 22 12 4 4\n",
      "92\t44\t0.0\t6\t0 0 0 25 14 4 1\n",
      "93\t39\t0.0\t6\t0 0 0 20 14 4 1\n",
      "94\t29\t0.0\t6\t0 0 0 13 13 0 3\n",
      "95\t28\t0.0\t6\t0 0 0 12 10 2 4\n",
      "96\t33\t0.0\t6\t0 0 0 18 12 2 1\n",
      "97\t29\t0.0\t6\t0 0 0 13 11 4 1\n",
      "98\t25\t0.0\t6\t0 0 0 15 9 1\n",
      "99\t17\t0.0\t6\t0 0 0 8 4 2 3\n",
      "100\t26\t0.0\t6\t0 0 0 14 9 1 2\n",
      "101\t13\t0.0\t6\t0 0 0 7 3 1 2\n",
      "102\t13\t0.0\t6\t0 0 0 10 2 0 1\n",
      "103\t13\t0.0\t6\t0 0 0 4 3 1 5\n",
      "104\t14\t0.0\t6\t0 0 0 8 3 2 1\n",
      "105\t12\t0.0\t6\t0 0 0 4 3 2 3\n",
      "106\t7\t0.0\t6\t0 0 0 5 1 1\n",
      "107\t11\t0.0\t6\t0 0 0 5 4 2\n",
      "108\t7\t0.0\t6\t0 0 0 6 0 1\n",
      "109\t13\t0.0\t6\t0 0 0 5 5 1 2\n",
      "110\t11\t0.0\t6\t0 0 0 8 0 2 1\n",
      "111\t6\t0.0\t6\t0 0 0 2 0 3 1\n",
      "112\t9\t0.0\t6\t0 0 0 4 1 3 1\n",
      "113\t5\t0.0\t6\t0 0 0 1 2 1 1\n",
      "114\t6\t0.0\t6\t0 0 0 3 1 1 1\n",
      "115\t5\t0.0\t6\t0 0 0 1 0 0 4\n",
      "116\t3\t0.0\t6\t0 0 0 1 1 1\n",
      "117\t5\t0.0\t6\t0 0 0 1 2 1 1\n",
      "118\t22\t0.0\t6\t0 0 0 5 8 4 5\n",
      "119\t16\t0.0\t6\t0 0 0 8 7 1\n",
      "120\t20\t0.0\t6\t0 0 0 5 9 5 1\n",
      "121\t1\t0.0\t6\t0 0 0 0 1\n",
      "122\t7\t0.0\t6\t0 0 0 4 3\n",
      "123\t6\t0.0\t6\t0 0 0 0 1 5\n",
      "124\t11\t0.0\t6\t0 0 0 4 4 2 1\n",
      "125\t4\t0.0\t6\t0 0 0 1 1 2\n",
      "126\t4\t0.0\t6\t0 0 0 3 1\n",
      "127\t41\t0.0\t6\t0 0 0 20 12 4 5\n",
      "128\t22\t0.0\t6\t0 0 0 6 7 7 2\n",
      "129\t19\t0.0\t6\t0 0 0 5 8 2 4\n",
      "130\t31\t0.0\t6\t0 0 0 12 13 4 2\n",
      "131\t37\t0.0\t6\t0 0 0 14 14 6 3\n",
      "132\t15\t0.0\t6\t0 0 0 4 5 3 3\n",
      "133\t15\t0.0\t6\t0 0 0 3 9 3\n",
      "134\t19\t0.0\t6\t0 0 0 10 5 1 3\n",
      "135\t83\t0.0\t6\t0 0 0 35 28 15 5\n",
      "136\t14\t0.0\t6\t0 0 0 4 5 3 2\n",
      "137\t20\t0.0\t6\t0 0 0 7 9 3 1\n",
      "138\t53\t0.0\t6\t0 0 0 23 19 11\n",
      "139\t38\t0.0\t6\t0 0 0 17 9 6 6\n",
      "140\t178\t0.0\t6\t0 0 0 64 67 36 11\n",
      "141\t67\t0.0\t6\t0 0 0 24 29 11 3\n",
      "142\t56\t0.0\t6\t0 0 0 21 24 4 7\n",
      "143\t48\t0.0\t6\t0 0 0 19 14 12 3\n",
      "144\t183\t0.0\t6\t0 0 0 71 67 32 13\n",
      "145\t49\t0.0\t6\t0 0 0 13 17 13 6\n",
      "146\t74\t0.0\t6\t0 0 0 21 36 16 1\n",
      "147\t14\t0.0\t6\t0 0 0 5 6 2 1\n",
      "148\t6\t0.0\t6\t0 0 0 0 5 1\n",
      "149\t70\t0.0\t6\t0 0 0 23 27 14 6\n",
      "150\t1410\t0.0\t6\t0 0 0 480 506 264 160\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 767348 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 38.4%\n",
      "  C: 27.8%\n",
      "  G: 24.9%\n",
      "  T: 8.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t423474\t385399.9\t0\t423474\n",
      "4\t127476\t96350.0\t0\t127476\n",
      "5\t32815\t24087.5\t0\t32815\n",
      "6\t16944\t6021.9\t0\t16944\n",
      "7\t10911\t1505.5\t0\t10911\n",
      "8\t8801\t376.4\t0\t8801\n",
      "9\t9086\t94.1\t0\t8558 528\n",
      "10\t8687\t23.5\t1\t8020 667\n",
      "11\t8119\t5.9\t1\t7628 491\n",
      "12\t7666\t1.5\t1\t7340 326\n",
      "13\t7173\t0.4\t1\t6862 311\n",
      "14\t6781\t0.1\t1\t6532 249\n",
      "15\t6523\t0.0\t1\t6241 282\n",
      "16\t6382\t0.0\t1\t6099 283\n",
      "17\t5892\t0.0\t1\t5617 275\n",
      "18\t5696\t0.0\t1\t5317 379\n",
      "19\t5481\t0.0\t1\t5115 358 8\n",
      "20\t5545\t0.0\t2\t5046 442 57\n",
      "21\t5213\t0.0\t2\t4725 429 59\n",
      "22\t4912\t0.0\t2\t4448 399 65\n",
      "23\t4708\t0.0\t2\t4275 381 52\n",
      "24\t4460\t0.0\t2\t4018 395 47\n",
      "25\t4089\t0.0\t2\t3704 346 39\n",
      "26\t3855\t0.0\t2\t3398 393 64\n",
      "27\t3771\t0.0\t2\t3363 342 66\n",
      "28\t3749\t0.0\t2\t3345 356 48\n",
      "29\t3363\t0.0\t2\t2940 367 53 3\n",
      "30\t3253\t0.0\t3\t2881 311 42 19\n",
      "31\t3088\t0.0\t3\t2739 285 49 15\n",
      "32\t2887\t0.0\t3\t2568 262 43 14\n",
      "33\t2670\t0.0\t3\t2314 249 83 24\n",
      "34\t2690\t0.0\t3\t1 2373 218 98\n",
      "35\t2352\t0.0\t3\t0 5 2091 256\n",
      "36\t2165\t0.0\t3\t0 0 1965 200\n",
      "37\t2065\t0.0\t3\t0 0 7 2058\n",
      "38\t1978\t0.0\t3\t0 0 5 40 1933\n",
      "39\t1842\t0.0\t3\t0 0 0 17 1825\n",
      "40\t504\t0.0\t4\t0 0 1 10 493\n",
      "41\t137\t0.0\t4\t0 0 0 1 136\n",
      "42\t123\t0.0\t4\t0 0 0 3 120\n",
      "43\t21\t0.0\t4\t0 0 0 0 21\n",
      "44\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a12_rep2_control_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 34.66 s (1 µs/read; 43.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         24,933,926\n",
      "  Read 1 with adapter:                 902,301 (3.6%)\n",
      "  Read 2 with adapter:                 744,556 (3.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):    24,933,926 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,480,177,800 bp\n",
      "  Read 1: 3,740,088,900 bp\n",
      "  Read 2: 3,740,088,900 bp\n",
      "Total written (filtered):  7,467,804,808 bp (99.8%)\n",
      "  Read 1: 3,733,989,023 bp\n",
      "  Read 2: 3,733,815,785 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 902301 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 39.8%\n",
      "  C: 20.8%\n",
      "  G: 14.3%\n",
      "  T: 25.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t482893\t389592.6\t0\t482893\n",
      "4\t159048\t97398.1\t0\t159048\n",
      "5\t47342\t24349.5\t0\t47342\n",
      "6\t19400\t6087.4\t0\t19400\n",
      "7\t12042\t1521.8\t0\t12042\n",
      "8\t10044\t380.5\t0\t10044\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9\t10428\t95.1\t0\t9727 701\n",
      "10\t9978\t23.8\t1\t9138 840\n",
      "11\t9174\t5.9\t1\t8687 487\n",
      "12\t8837\t1.5\t1\t8504 333\n",
      "13\t8422\t0.4\t1\t8158 264\n",
      "14\t7987\t0.1\t1\t7716 271\n",
      "15\t7510\t0.0\t1\t7196 314\n",
      "16\t7255\t0.0\t1\t6999 256\n",
      "17\t7059\t0.0\t1\t6788 271\n",
      "18\t6751\t0.0\t1\t6448 302 1\n",
      "19\t6635\t0.0\t1\t6341 289 5\n",
      "20\t6243\t0.0\t2\t5944 254 45\n",
      "21\t5760\t0.0\t2\t5453 261 46\n",
      "22\t5508\t0.0\t2\t5226 229 53\n",
      "23\t5220\t0.0\t2\t4942 236 42\n",
      "24\t5035\t0.0\t2\t4783 219 33\n",
      "25\t4938\t0.0\t2\t4656 232 50\n",
      "26\t4693\t0.0\t2\t4399 249 45\n",
      "27\t4300\t0.0\t2\t4039 211 50\n",
      "28\t4176\t0.0\t2\t3887 242 46 1\n",
      "29\t4126\t0.0\t2\t3840 249 37\n",
      "30\t4024\t0.0\t3\t3734 221 40 29\n",
      "31\t3667\t0.0\t3\t3409 195 41 22\n",
      "32\t3717\t0.0\t3\t3429 229 42 17\n",
      "33\t3294\t0.0\t3\t3058 190 26 20\n",
      "34\t3237\t0.0\t3\t2983 145 81 28\n",
      "35\t2958\t0.0\t3\t1 2780 144 33\n",
      "36\t2760\t0.0\t3\t0 2581 109 70\n",
      "37\t2834\t0.0\t3\t0 0 2680 109 45\n",
      "38\t2573\t0.0\t3\t0 0 13 2457 103\n",
      "39\t2281\t0.0\t3\t0 0 8 2268 5\n",
      "40\t20\t0.0\t4\t0 0 1 13 6\n",
      "41\t16\t0.0\t4\t0 0 0 7 9\n",
      "42\t64\t0.0\t4\t0 0 0 0 64\n",
      "43\t48\t0.0\t4\t0 0 0 1 47\n",
      "45\t1\t0.0\t4\t0 0 0 0 1\n",
      "46\t1\t0.0\t4\t0 0 0 0 1\n",
      "53\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "60\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 744556 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 34.3%\n",
      "  C: 29.8%\n",
      "  G: 25.4%\n",
      "  T: 10.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t371587\t389592.6\t0\t371587\n",
      "4\t101868\t97398.1\t0\t101868\n",
      "5\t34887\t24349.5\t0\t34887\n",
      "6\t19740\t6087.4\t0\t19740\n",
      "7\t11599\t1521.8\t0\t11599\n",
      "8\t10586\t380.5\t0\t10586\n",
      "9\t10929\t95.1\t0\t10025 904\n",
      "10\t10318\t23.8\t1\t9707 611\n",
      "11\t9586\t5.9\t1\t9197 389\n",
      "12\t9104\t1.5\t1\t8735 369\n",
      "13\t8866\t0.4\t1\t8535 331\n",
      "14\t8480\t0.1\t1\t8175 305\n",
      "15\t8031\t0.0\t1\t7755 276\n",
      "16\t7520\t0.0\t1\t7237 283\n",
      "17\t7274\t0.0\t1\t6962 312\n",
      "18\t7060\t0.0\t1\t6672 386 2\n",
      "19\t6785\t0.0\t1\t6373 404 8\n",
      "20\t6718\t0.0\t2\t6178 463 77\n",
      "21\t6273\t0.0\t2\t5699 484 90\n",
      "22\t5796\t0.0\t2\t5230 505 61\n",
      "23\t5541\t0.0\t2\t5072 403 66\n",
      "24\t5235\t0.0\t2\t4797 386 52\n",
      "25\t5066\t0.0\t2\t4672 352 42\n",
      "26\t4958\t0.0\t2\t4379 505 74\n",
      "27\t4716\t0.0\t2\t4150 485 81\n",
      "28\t4336\t0.0\t2\t3831 429 75 1\n",
      "29\t4207\t0.0\t2\t3675 451 78 3\n",
      "30\t4164\t0.0\t3\t3646 434 62 22\n",
      "31\t4051\t0.0\t3\t3512 438 75 26\n",
      "32\t3686\t0.0\t3\t3208 377 75 26\n",
      "33\t3740\t0.0\t3\t3202 388 120 30\n",
      "34\t3297\t0.0\t3\t1 2880 347 69\n",
      "35\t3224\t0.0\t3\t3 2820 294 107\n",
      "36\t2924\t0.0\t3\t0 5 2568 351\n",
      "37\t2644\t0.0\t3\t0 3 1977 664\n",
      "38\t2686\t0.0\t3\t0 4 1777 904 1\n",
      "39\t2419\t0.0\t3\t0 6 1634 725 54\n",
      "40\t2137\t0.0\t4\t0 0 1397 695 45\n",
      "41\t2247\t0.0\t4\t0 4 1318 691 234\n",
      "42\t2151\t0.0\t4\t0 1 1296 598 256\n",
      "43\t2031\t0.0\t4\t0 1 857 875 298\n",
      "44\t1741\t0.0\t4\t0 0 1 1157 583\n",
      "45\t1598\t0.0\t4\t0 0 2 1057 539\n",
      "46\t1440\t0.0\t4\t0 0 0 744 696\n",
      "47\t962\t0.0\t4\t0 0 0 80 882\n",
      "48\t212\t0.0\t4\t0 0 0 19 193\n",
      "49\t93\t0.0\t4\t0 0 0 0 93\n",
      "50\t24\t0.0\t5\t0 0 0 0 24\n",
      "51\t5\t0.0\t5\t0 0 0 0 4 1\n",
      "52\t6\t0.0\t5\t0 0 0 0 0 6\n",
      "53\t8\t0.0\t5\t0 0 0 0 0 8\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_1.fastq ./20211105180528381-5936/original/DEFAULT/a13_rep3_control_t-5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 53.51 s (2 µs/read; 30.82 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         27,491,204\n",
      "  Read 1 with adapter:                 871,023 (3.2%)\n",
      "  Read 2 with adapter:                 674,928 (2.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 667 (0.0%)\n",
      "Pairs written (passing filters):    27,490,537 (100.0%)\n",
      "\n",
      "Total basepairs processed: 8,247,361,200 bp\n",
      "  Read 1: 4,123,680,600 bp\n",
      "  Read 2: 4,123,680,600 bp\n",
      "Total written (filtered):  8,237,982,625 bp (99.9%)\n",
      "  Read 1: 4,118,595,715 bp\n",
      "  Read 2: 4,119,386,910 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 871023 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.1%\n",
      "  C: 22.2%\n",
      "  G: 16.5%\n",
      "  T: 29.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t516678\t429550.1\t0\t516678\n",
      "4\t171105\t107387.5\t0\t171105\n",
      "5\t43041\t26846.9\t0\t43041\n",
      "6\t17142\t6711.7\t0\t17142\n",
      "7\t8051\t1677.9\t0\t8051\n",
      "8\t6262\t419.5\t0\t6262\n",
      "9\t6530\t104.9\t0\t5896 634\n",
      "10\t6785\t26.2\t1\t5820 965\n",
      "11\t5708\t6.6\t1\t5300 408\n",
      "12\t5500\t1.6\t1\t5218 282\n",
      "13\t5118\t0.4\t1\t4941 177\n",
      "14\t4819\t0.1\t1\t4661 158\n",
      "15\t4750\t0.0\t1\t4559 191\n",
      "16\t4577\t0.0\t1\t4408 169\n",
      "17\t4262\t0.0\t1\t4117 145\n",
      "18\t4078\t0.0\t1\t3890 188\n",
      "19\t3958\t0.0\t1\t3769 188 1\n",
      "20\t3857\t0.0\t2\t3640 169 48\n",
      "21\t3697\t0.0\t2\t3500 166 31\n",
      "22\t3476\t0.0\t2\t3287 150 39\n",
      "23\t3205\t0.0\t2\t3002 167 36\n",
      "24\t3047\t0.0\t2\t2865 153 29\n",
      "25\t2927\t0.0\t2\t2761 138 28\n",
      "26\t2677\t0.0\t2\t2478 166 33\n",
      "27\t2582\t0.0\t2\t2440 124 18\n",
      "28\t2508\t0.0\t2\t2370 111 27\n",
      "29\t2381\t0.0\t2\t2239 115 25 2\n",
      "30\t2368\t0.0\t3\t2187 132 34 15\n",
      "31\t2344\t0.0\t3\t2184 123 24 13\n",
      "32\t2132\t0.0\t3\t1956 121 34 21\n",
      "33\t2115\t0.0\t3\t1972 75 56 12\n",
      "34\t1864\t0.0\t3\t1 1728 84 51\n",
      "35\t1704\t0.0\t3\t0 1 1585 118\n",
      "36\t1715\t0.0\t3\t0 1 1615 99\n",
      "37\t1591\t0.0\t3\t0 0 7 1584\n",
      "38\t1474\t0.0\t3\t0 0 0 11 1463\n",
      "39\t1378\t0.0\t3\t0 0 1 5 1372\n",
      "40\t14\t0.0\t4\t0 0 0 7 7\n",
      "41\t7\t0.0\t4\t0 0 0 7\n",
      "42\t1\t0.0\t4\t0 0 0 1\n",
      "43\t5\t0.0\t4\t0 0 0 0 5\n",
      "44\t1\t0.0\t4\t0 0 0 0 1\n",
      "45\t3\t0.0\t4\t0 0 0 0 3\n",
      "50\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "51\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "55\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "60\t220\t0.0\t6\t0 0 0 0 0 1 219\n",
      "61\t230\t0.0\t6\t0 0 0 0 0 0 230\n",
      "62\t231\t0.0\t6\t0 0 0 0 0 0 231\n",
      "63\t201\t0.0\t6\t0 0 0 0 0 0 201\n",
      "64\t164\t0.0\t6\t0 0 0 0 0 1 163\n",
      "65\t146\t0.0\t6\t0 0 0 0 0 0 146\n",
      "66\t125\t0.0\t6\t0 0 0 0 0 0 125\n",
      "67\t135\t0.0\t6\t0 0 0 0 0 1 134\n",
      "68\t115\t0.0\t6\t0 0 0 0 0 0 115\n",
      "69\t111\t0.0\t6\t0 0 0 0 0 0 111\n",
      "70\t98\t0.0\t6\t0 0 0 0 0 0 98\n",
      "71\t103\t0.0\t6\t0 0 0 0 0 0 103\n",
      "72\t97\t0.0\t6\t0 0 0 0 0 0 97\n",
      "73\t105\t0.0\t6\t0 0 0 0 0 0 105\n",
      "74\t76\t0.0\t6\t0 0 0 0 0 0 76\n",
      "75\t71\t0.0\t6\t0 0 0 0 0 0 71\n",
      "76\t49\t0.0\t6\t0 0 0 0 0 0 49\n",
      "77\t47\t0.0\t6\t0 0 0 0 0 0 47\n",
      "78\t46\t0.0\t6\t0 0 0 0 0 0 46\n",
      "79\t43\t0.0\t6\t0 0 0 0 0 0 43\n",
      "80\t51\t0.0\t6\t0 0 0 0 0 0 51\n",
      "81\t30\t0.0\t6\t0 0 0 0 0 0 30\n",
      "82\t58\t0.0\t6\t0 0 0 0 0 0 58\n",
      "83\t38\t0.0\t6\t0 0 0 0 0 0 38\n",
      "84\t44\t0.0\t6\t0 0 0 0 0 0 44\n",
      "85\t44\t0.0\t6\t0 0 0 0 0 1 43\n",
      "86\t32\t0.0\t6\t0 0 0 0 0 0 32\n",
      "87\t30\t0.0\t6\t0 0 0 0 0 0 30\n",
      "88\t29\t0.0\t6\t0 0 0 0 0 1 28\n",
      "89\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "90\t19\t0.0\t6\t0 0 0 0 0 0 19\n",
      "91\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "92\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "93\t14\t0.0\t6\t0 0 0 0 0 0 14\n",
      "94\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "95\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "96\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "97\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "98\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "99\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "100\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "101\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "102\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "103\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "104\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "105\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "106\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "107\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "108\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "109\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "110\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "111\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "112\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "113\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "114\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "115\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "117\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "118\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "119\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "120\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "121\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "122\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "123\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "124\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "125\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "126\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "127\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "128\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "129\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "130\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "131\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "132\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "133\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "134\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "135\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "136\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "137\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "138\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "139\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "140\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "141\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "142\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "143\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "144\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "145\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "146\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "147\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "149\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "150\t393\t0.0\t6\t0 0 0 0 0 0 393\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 674928 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 37.5%\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  C: 28.7%\n",
      "  G: 24.7%\n",
      "  T: 9.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t393919\t429550.1\t0\t393919\n",
      "4\t107604\t107387.5\t0\t107604\n",
      "5\t33288\t26846.9\t0\t33288\n",
      "6\t15983\t6711.7\t0\t15983\n",
      "7\t7666\t1677.9\t0\t7666\n",
      "8\t6681\t419.5\t0\t6681\n",
      "9\t6862\t104.9\t0\t6148 714\n",
      "10\t6540\t26.2\t1\t5822 718\n",
      "11\t6161\t6.6\t1\t5832 329\n",
      "12\t5630\t1.6\t1\t5292 338\n",
      "13\t5465\t0.4\t1\t5211 254\n",
      "14\t5168\t0.1\t1\t4945 223\n",
      "15\t4850\t0.0\t1\t4642 208\n",
      "16\t4758\t0.0\t1\t4571 187\n",
      "17\t4601\t0.0\t1\t4396 205\n",
      "18\t4271\t0.0\t1\t4006 265\n",
      "19\t4102\t0.0\t1\t3819 273 10\n",
      "20\t4020\t0.0\t2\t3682 285 53\n",
      "21\t3893\t0.0\t2\t3499 328 66\n",
      "22\t3735\t0.0\t2\t3347 323 65\n",
      "23\t3514\t0.0\t2\t3204 263 47\n",
      "24\t3228\t0.0\t2\t2933 249 46\n",
      "25\t3067\t0.0\t2\t2803 219 45\n",
      "26\t2946\t0.0\t2\t2624 274 48\n",
      "27\t2699\t0.0\t2\t2378 283 38\n",
      "28\t2605\t0.0\t2\t2274 286 45\n",
      "29\t2543\t0.0\t2\t2208 277 55 3\n",
      "30\t2411\t0.0\t3\t2055 284 49 23\n",
      "31\t2391\t0.0\t3\t2052 257 48 34\n",
      "32\t2368\t0.0\t3\t2048 249 50 21\n",
      "33\t2153\t0.0\t3\t1853 192 82 26\n",
      "34\t2118\t0.0\t3\t1 1839 181 97\n",
      "35\t1803\t0.0\t3\t0 15 1616 172\n",
      "36\t1539\t0.0\t3\t0 2 75 1462\n",
      "37\t1568\t0.0\t3\t0 0 48 1502 18\n",
      "38\t1423\t0.0\t3\t0 0 7 55 1361\n",
      "39\t1332\t0.0\t3\t0 0 0 1281 51\n",
      "40\t14\t0.0\t4\t0 0 0 3 11\n",
      "41\t4\t0.0\t4\t0 0 0 0 4\n",
      "42\t1\t0.0\t4\t0 0 0 0 1\n",
      "43\t3\t0.0\t4\t0 0 0 0 3\n",
      "44\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_1.fastq ./20211105180528381-5936/original/DEFAULT/a14_rep3_+T4_t2_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 51.18 s (2 µs/read; 33.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         28,453,081\n",
      "  Read 1 with adapter:                 885,916 (3.1%)\n",
      "  Read 2 with adapter:                 666,854 (2.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 163 (0.0%)\n",
      "Pairs written (passing filters):    28,452,918 (100.0%)\n",
      "\n",
      "Total basepairs processed: 8,535,924,300 bp\n",
      "  Read 1: 4,267,962,150 bp\n",
      "  Read 2: 4,267,962,150 bp\n",
      "Total written (filtered):  8,527,558,114 bp (99.9%)\n",
      "  Read 1: 4,263,406,209 bp\n",
      "  Read 2: 4,264,151,905 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 885916 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.1%\n",
      "  C: 23.8%\n",
      "  G: 17.4%\n",
      "  T: 30.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t549972\t444579.4\t0\t549972\n",
      "4\t179690\t111144.8\t0\t179690\n",
      "5\t47291\t27786.2\t0\t47291\n",
      "6\t14603\t6946.6\t0\t14603\n",
      "7\t6631\t1736.6\t0\t6631\n",
      "8\t4604\t434.2\t0\t4604\n",
      "9\t5165\t108.5\t0\t4479 686\n",
      "10\t5067\t27.1\t1\t4195 872\n",
      "11\t4515\t6.8\t1\t4089 426\n",
      "12\t4289\t1.7\t1\t3993 296\n",
      "13\t3639\t0.4\t1\t3492 147\n",
      "14\t3829\t0.1\t1\t3691 138\n",
      "15\t3472\t0.0\t1\t3304 168\n",
      "16\t3410\t0.0\t1\t3273 137\n",
      "17\t3170\t0.0\t1\t3016 154\n",
      "18\t3098\t0.0\t1\t2954 140 4\n",
      "19\t3191\t0.0\t1\t3041 146 4\n",
      "20\t2984\t0.0\t2\t2819 135 30\n",
      "21\t2787\t0.0\t2\t2608 142 37\n",
      "22\t2617\t0.0\t2\t2462 127 28\n",
      "23\t2549\t0.0\t2\t2394 123 32\n",
      "24\t2323\t0.0\t2\t2176 124 23\n",
      "25\t2313\t0.0\t2\t2176 99 38\n",
      "26\t2109\t0.0\t2\t1995 92 22\n",
      "27\t2027\t0.0\t2\t1910 97 20\n",
      "28\t1975\t0.0\t2\t1836 114 25\n",
      "29\t1933\t0.0\t2\t1800 106 27\n",
      "30\t1853\t0.0\t3\t1702 94 42 15\n",
      "31\t1780\t0.0\t3\t1639 95 30 16\n",
      "32\t1778\t0.0\t3\t1616 118 36 8\n",
      "33\t1576\t0.0\t3\t1437 76 43 20\n",
      "34\t1460\t0.0\t3\t0 1368 71 21\n",
      "35\t1469\t0.0\t3\t0 1353 72 44\n",
      "36\t1408\t0.0\t3\t0 15 1305 88\n",
      "37\t1314\t0.0\t3\t0 4 1252 58\n",
      "38\t1247\t0.0\t3\t0 0 4 1241 2\n",
      "39\t1192\t0.0\t3\t0 0 7 1182 3\n",
      "40\t14\t0.0\t4\t0 0 1 8 5\n",
      "41\t24\t0.0\t4\t0 0 0 2 22\n",
      "42\t12\t0.0\t4\t0 0 0 1 11\n",
      "43\t10\t0.0\t4\t0 0 0 0 10\n",
      "50\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "51\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "52\t3\t0.0\t5\t0 0 0 0 0 3\n",
      "53\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "54\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "55\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "56\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "60\t127\t0.0\t6\t0 0 0 0 0 1 126\n",
      "61\t137\t0.0\t6\t0 0 0 0 0 0 137\n",
      "62\t124\t0.0\t6\t0 0 0 0 0 1 123\n",
      "63\t95\t0.0\t6\t0 0 0 0 0 1 94\n",
      "64\t72\t0.0\t6\t0 0 0 0 0 0 72\n",
      "65\t65\t0.0\t6\t0 0 0 0 0 0 65\n",
      "66\t77\t0.0\t6\t0 0 0 0 0 0 77\n",
      "67\t63\t0.0\t6\t0 0 0 0 0 0 63\n",
      "68\t55\t0.0\t6\t0 0 0 0 0 0 55\n",
      "69\t38\t0.0\t6\t0 0 0 0 0 0 38\n",
      "70\t32\t0.0\t6\t0 0 0 0 0 0 32\n",
      "71\t51\t0.0\t6\t0 0 0 0 0 0 51\n",
      "72\t31\t0.0\t6\t0 0 0 0 0 0 31\n",
      "73\t35\t0.0\t6\t0 0 0 0 0 0 35\n",
      "74\t30\t0.0\t6\t0 0 0 0 0 0 30\n",
      "75\t30\t0.0\t6\t0 0 0 0 0 0 30\n",
      "76\t22\t0.0\t6\t0 0 0 0 0 0 22\n",
      "77\t27\t0.0\t6\t0 0 0 0 0 0 27\n",
      "78\t21\t0.0\t6\t0 0 0 0 0 0 21\n",
      "79\t23\t0.0\t6\t0 0 0 0 0 0 23\n",
      "80\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "81\t24\t0.0\t6\t0 0 0 0 0 0 24\n",
      "82\t14\t0.0\t6\t0 0 0 0 0 0 14\n",
      "83\t24\t0.0\t6\t0 0 0 0 0 0 24\n",
      "84\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "85\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "86\t11\t0.0\t6\t0 0 0 0 0 1 10\n",
      "87\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "88\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "89\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "90\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "91\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "92\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "93\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "94\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "95\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "96\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "97\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "98\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "99\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "100\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "103\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "104\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "105\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "106\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "107\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "108\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "109\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "111\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "113\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "118\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "120\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "122\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "123\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "125\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "126\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "127\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "128\t7\t0.0\t6\t0 0 0 0 0 1 6\n",
      "129\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "130\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "131\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "133\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "134\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "135\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "136\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "137\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "138\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "139\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "140\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "141\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "142\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "143\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "144\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "145\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "149\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "150\t69\t0.0\t6\t0 0 0 0 0 0 69\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 666854 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 39.4%\n",
      "  C: 28.6%\n",
      "  G: 22.8%\n",
      "  T: 9.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t414973\t444579.4\t0\t414973\n",
      "4\t108685\t111144.8\t0\t108685\n",
      "5\t31747\t27786.2\t0\t31747\n",
      "6\t14858\t6946.6\t0\t14858\n",
      "7\t5685\t1736.6\t0\t5685\n",
      "8\t4934\t434.2\t0\t4934\n",
      "9\t5234\t108.5\t0\t4420 814\n",
      "10\t4952\t27.1\t1\t4386 566\n",
      "11\t4455\t6.8\t1\t4188 267\n",
      "12\t4374\t1.7\t1\t4086 288\n",
      "13\t4214\t0.4\t1\t3962 252\n",
      "14\t3667\t0.1\t1\t3461 206\n",
      "15\t3847\t0.0\t1\t3639 208\n",
      "16\t3502\t0.0\t1\t3318 184\n",
      "17\t3438\t0.0\t1\t3257 181\n",
      "18\t3183\t0.0\t1\t2933 248 2\n",
      "19\t3114\t0.0\t1\t2873 236 5\n",
      "20\t3249\t0.0\t2\t2909 281 59\n",
      "21\t3008\t0.0\t2\t2671 274 63\n",
      "22\t2809\t0.0\t2\t2506 251 52\n",
      "23\t2632\t0.0\t2\t2317 258 57\n",
      "24\t2566\t0.0\t2\t2296 227 43\n",
      "25\t2338\t0.0\t2\t2091 209 38\n",
      "26\t2326\t0.0\t2\t1997 280 49\n",
      "27\t2124\t0.0\t2\t1821 251 52\n",
      "28\t2048\t0.0\t2\t1782 210 56\n",
      "29\t1997\t0.0\t2\t1706 245 46\n",
      "30\t1982\t0.0\t3\t1685 221 52 24\n",
      "31\t1873\t0.0\t3\t1583 217 52 21\n",
      "32\t1804\t0.0\t3\t1519 205 47 33\n",
      "33\t1809\t0.0\t3\t1509 188 79 33\n",
      "34\t1581\t0.0\t3\t0 1308 219 54\n",
      "35\t1469\t0.0\t3\t0 1254 142 73\n",
      "36\t1445\t0.0\t3\t0 1 1254 190\n",
      "37\t1363\t0.0\t3\t0 0 1191 172\n",
      "38\t1140\t0.0\t3\t0 0 1 1137 2\n",
      "39\t1184\t0.0\t3\t0 0 0 45 1139\n",
      "40\t762\t0.0\t4\t0 0 0 7 755\n",
      "41\t394\t0.0\t4\t0 0 0 4 390\n",
      "42\t42\t0.0\t4\t0 0 0 0 42\n",
      "43\t30\t0.0\t4\t0 0 0 0 30\n",
      "44\t11\t0.0\t4\t0 0 0 0 11\n",
      "45\t5\t0.0\t4\t0 0 0 0 5\n",
      "46\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_1.fastq ./20211105180528381-5936/original/DEFAULT/a15_rep3_+T4_t5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 44.53 s (1 µs/read; 40.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         29,907,205\n",
      "  Read 1 with adapter:                 943,307 (3.2%)\n",
      "  Read 2 with adapter:                 707,893 (2.4%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 192 (0.0%)\n",
      "Pairs written (passing filters):    29,907,013 (100.0%)\n",
      "\n",
      "Total basepairs processed: 8,972,161,500 bp\n",
      "  Read 1: 4,486,080,750 bp\n",
      "  Read 2: 4,486,080,750 bp\n",
      "Total written (filtered):  8,963,522,487 bp (99.9%)\n",
      "  Read 1: 4,481,241,314 bp\n",
      "  Read 2: 4,482,281,173 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 943307 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 24.3%\n",
      "  G: 17.1%\n",
      "  T: 31.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t588334\t467300.1\t0\t588334\n",
      "4\t195924\t116825.0\t0\t195924\n",
      "5\t50527\t29206.3\t0\t50527\n",
      "6\t15035\t7301.6\t0\t15035\n",
      "7\t6554\t1825.4\t0\t6554\n",
      "8\t4481\t456.3\t0\t4481\n",
      "9\t5142\t114.1\t0\t4372 770\n",
      "10\t4715\t28.5\t1\t3919 796\n",
      "11\t4376\t7.1\t1\t3992 384\n",
      "12\t3876\t1.8\t1\t3611 265\n",
      "13\t3872\t0.4\t1\t3705 167\n",
      "14\t3438\t0.1\t1\t3304 134\n",
      "15\t3353\t0.0\t1\t3197 156\n",
      "16\t3214\t0.0\t1\t3065 149\n",
      "17\t3027\t0.0\t1\t2880 147\n",
      "18\t3027\t0.0\t1\t2871 156\n",
      "19\t2860\t0.0\t1\t2684 169 7\n",
      "20\t2862\t0.0\t2\t2662 169 31\n",
      "21\t2572\t0.0\t2\t2404 139 29\n",
      "22\t2544\t0.0\t2\t2377 136 31\n",
      "23\t2367\t0.0\t2\t2197 140 30\n",
      "24\t2115\t0.0\t2\t1960 139 16\n",
      "25\t2152\t0.0\t2\t2012 115 25\n",
      "26\t2042\t0.0\t2\t1884 133 25\n",
      "27\t1945\t0.0\t2\t1793 124 28\n",
      "28\t1879\t0.0\t2\t1722 131 26\n",
      "29\t1872\t0.0\t2\t1705 139 28\n",
      "30\t1774\t0.0\t3\t1621 120 20 13\n",
      "31\t1720\t0.0\t3\t1553 123 29 15\n",
      "32\t1646\t0.0\t3\t1490 115 27 14\n",
      "33\t1528\t0.0\t3\t1382 84 44 18\n",
      "34\t1521\t0.0\t3\t0 1382 85 54\n",
      "35\t1393\t0.0\t3\t0 3 1287 103\n",
      "36\t1315\t0.0\t3\t0 0 1229 86\n",
      "37\t1257\t0.0\t3\t0 0 1161 81 15\n",
      "38\t1203\t0.0\t3\t0 0 6 1114 83\n",
      "39\t1026\t0.0\t3\t0 0 2 1016 8\n",
      "40\t1023\t0.0\t4\t0 0 0 4 1019\n",
      "41\t1\t0.0\t4\t0 0 0 0 1\n",
      "43\t2\t0.0\t4\t0 0 0 0 2\n",
      "44\t2\t0.0\t4\t0 0 0 1 1\n",
      "46\t1\t0.0\t4\t0 0 0 0 1\n",
      "47\t1\t0.0\t4\t0 0 0 0 1\n",
      "48\t1\t0.0\t4\t0 0 0 0 1\n",
      "49\t2\t0.0\t4\t0 0 0 0 0 2\n",
      "50\t335\t0.0\t5\t0 0 0 0 0 335\n",
      "51\t294\t0.0\t5\t0 0 0 0 0 294\n",
      "52\t250\t0.0\t5\t0 0 0 0 0 250\n",
      "53\t226\t0.0\t5\t0 0 0 0 0 226\n",
      "54\t223\t0.0\t5\t0 0 0 0 0 223\n",
      "55\t146\t0.0\t5\t0 0 0 0 1 145\n",
      "56\t140\t0.0\t5\t0 0 0 0 0 140\n",
      "57\t83\t0.0\t5\t0 0 0 0 1 82\n",
      "58\t107\t0.0\t5\t0 0 0 0 0 107\n",
      "59\t115\t0.0\t5\t0 0 0 0 0 113 2\n",
      "60\t171\t0.0\t6\t0 0 0 0 0 80 91\n",
      "61\t138\t0.0\t6\t0 0 0 0 0 72 66\n",
      "62\t134\t0.0\t6\t0 0 0 0 0 64 70\n",
      "63\t123\t0.0\t6\t0 0 0 0 0 65 58\n",
      "64\t96\t0.0\t6\t0 0 0 0 0 42 54\n",
      "65\t85\t0.0\t6\t0 0 0 0 1 37 47\n",
      "66\t70\t0.0\t6\t0 0 0 0 0 30 40\n",
      "67\t56\t0.0\t6\t0 0 0 0 0 31 25\n",
      "68\t62\t0.0\t6\t0 0 0 0 0 29 33\n",
      "69\t62\t0.0\t6\t0 0 0 0 0 25 37\n",
      "70\t64\t0.0\t6\t0 0 0 0 0 27 37\n",
      "71\t56\t0.0\t6\t0 0 0 0 0 28 28\n",
      "72\t51\t0.0\t6\t0 0 0 0 0 20 31\n",
      "73\t36\t0.0\t6\t0 0 0 0 0 10 26\n",
      "74\t45\t0.0\t6\t0 0 0 0 0 17 28\n",
      "75\t42\t0.0\t6\t0 0 0 0 0 17 25\n",
      "76\t37\t0.0\t6\t0 0 0 0 0 15 22\n",
      "77\t34\t0.0\t6\t0 0 0 0 0 13 21\n",
      "78\t37\t0.0\t6\t0 0 0 0 0 20 17\n",
      "79\t29\t0.0\t6\t0 0 0 0 0 11 18\n",
      "80\t13\t0.0\t6\t0 0 0 0 0 5 8\n",
      "81\t27\t0.0\t6\t0 0 0 0 0 13 14\n",
      "82\t14\t0.0\t6\t0 0 0 0 0 7 7\n",
      "83\t21\t0.0\t6\t0 0 0 0 0 10 11\n",
      "84\t16\t0.0\t6\t0 0 0 0 0 5 11\n",
      "85\t18\t0.0\t6\t0 0 0 0 0 9 9\n",
      "86\t12\t0.0\t6\t0 0 0 0 0 4 8\n",
      "87\t15\t0.0\t6\t0 0 0 0 0 6 9\n",
      "88\t18\t0.0\t6\t0 0 0 0 0 4 14\n",
      "89\t15\t0.0\t6\t0 0 0 0 0 4 11\n",
      "90\t10\t0.0\t6\t0 0 0 0 0 2 8\n",
      "91\t8\t0.0\t6\t0 0 0 0 0 1 7\n",
      "92\t10\t0.0\t6\t0 0 0 0 0 5 5\n",
      "93\t14\t0.0\t6\t0 0 0 0 0 4 10\n",
      "94\t7\t0.0\t6\t0 0 0 0 0 4 3\n",
      "95\t7\t0.0\t6\t0 0 0 0 0 3 4\n",
      "96\t8\t0.0\t6\t0 0 0 0 0 2 6\n",
      "97\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "98\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "99\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "100\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "101\t6\t0.0\t6\t0 0 0 0 0 1 5\n",
      "102\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "105\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "106\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "107\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "108\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "110\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "111\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "113\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "114\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "116\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "118\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "119\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "123\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "124\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "125\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "126\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "127\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "128\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "129\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "130\t4\t0.0\t6\t0 0 0 0 0 2 2\n",
      "131\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "133\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "134\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "135\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "136\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "137\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "138\t4\t0.0\t6\t0 0 0 0 0 2 2\n",
      "139\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "140\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "141\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "142\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "143\t5\t0.0\t6\t0 0 0 0 0 2 3\n",
      "144\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "145\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "146\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "149\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "150\t102\t0.0\t6\t0 0 0 0 0 39 63\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 707893 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 41.0%\n",
      "  C: 27.7%\n",
      "  G: 22.3%\n",
      "  T: 8.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t450076\t467300.1\t0\t450076\n",
      "4\t116856\t116825.0\t0\t116856\n",
      "5\t34046\t29206.3\t0\t34046\n",
      "6\t16019\t7301.6\t0\t16019\n",
      "7\t5588\t1825.4\t0\t5588\n",
      "8\t4767\t456.3\t0\t4767\n",
      "9\t5390\t114.1\t0\t4471 919\n",
      "10\t5197\t28.5\t1\t4310 887\n",
      "11\t4379\t7.1\t1\t3970 409\n",
      "12\t4281\t1.8\t1\t3995 286\n",
      "13\t3822\t0.4\t1\t3613 209\n",
      "14\t3911\t0.1\t1\t3735 176\n",
      "15\t3464\t0.0\t1\t3301 163\n",
      "16\t3366\t0.0\t1\t3184 182\n",
      "17\t3237\t0.0\t1\t3080 157\n",
      "18\t3043\t0.0\t1\t2855 188\n",
      "19\t3068\t0.0\t1\t2855 210 3\n",
      "20\t2911\t0.0\t2\t2630 221 60\n",
      "21\t2889\t0.0\t2\t2612 235 42\n",
      "22\t2599\t0.0\t2\t2332 233 34\n",
      "23\t2560\t0.0\t2\t2298 222 40\n",
      "24\t2386\t0.0\t2\t2159 181 46\n",
      "25\t2122\t0.0\t2\t1926 162 34\n",
      "26\t2187\t0.0\t2\t1941 201 45\n",
      "27\t2068\t0.0\t2\t1801 228 39\n",
      "28\t1966\t0.0\t2\t1743 186 37\n",
      "29\t1921\t0.0\t2\t1658 210 51 2\n",
      "30\t1912\t0.0\t3\t1641 204 42 25\n",
      "31\t1813\t0.0\t3\t1568 180 50 15\n",
      "32\t1742\t0.0\t3\t1516 173 28 25\n",
      "33\t1661\t0.0\t3\t1424 152 64 21\n",
      "34\t1550\t0.0\t3\t3 1305 172 70\n",
      "35\t1476\t0.0\t3\t0 12 1317 147\n",
      "36\t1243\t0.0\t3\t0 2 21 1205 15\n",
      "37\t1191\t0.0\t3\t0 0 3 27 1161\n",
      "38\t1124\t0.0\t3\t0 0 0 12 1112\n",
      "39\t49\t0.0\t3\t0 0 0 7 42\n",
      "40\t10\t0.0\t4\t0 0 2 2 6\n",
      "41\t3\t0.0\t4\t0 0 0 0 3\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_1.fastq ./20211105180528381-5936/original/DEFAULT/a16_rep3_+T4_t10_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 44.56 s (2 µs/read; 31.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         23,686,633\n",
      "  Read 1 with adapter:                 744,790 (3.1%)\n",
      "  Read 2 with adapter:                 615,612 (2.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 435 (0.0%)\n",
      "Pairs written (passing filters):    23,686,198 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,105,989,900 bp\n",
      "  Read 1: 3,552,994,950 bp\n",
      "  Read 2: 3,552,994,950 bp\n",
      "Total written (filtered):  7,097,935,579 bp (99.9%)\n",
      "  Read 1: 3,548,794,616 bp\n",
      "  Read 2: 3,549,140,963 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 744790 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 31.3%\n",
      "  C: 23.2%\n",
      "  G: 15.8%\n",
      "  T: 29.6%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t450327\t370103.6\t0\t450327\n",
      "4\t146870\t92525.9\t0\t146870\n",
      "5\t37659\t23131.5\t0\t37659\n",
      "6\t12290\t5782.9\t0\t12290\n",
      "7\t5966\t1445.7\t0\t5966\n",
      "8\t4573\t361.4\t0\t4573\n",
      "9\t4995\t90.4\t0\t4211 784\n",
      "10\t4750\t22.6\t1\t4012 738\n",
      "11\t4558\t5.6\t1\t4030 528\n",
      "12\t4249\t1.4\t1\t3789 460\n",
      "13\t3964\t0.4\t1\t3615 349\n",
      "14\t3844\t0.1\t1\t3512 332\n",
      "15\t3583\t0.0\t1\t3291 292\n",
      "16\t3481\t0.0\t1\t3178 303\n",
      "17\t3226\t0.0\t1\t2953 273\n",
      "18\t3232\t0.0\t1\t2910 318 4\n",
      "19\t3329\t0.0\t1\t2995 326 8\n",
      "20\t3174\t0.0\t2\t2808 305 61\n",
      "21\t2980\t0.0\t2\t2651 287 42\n",
      "22\t2691\t0.0\t2\t2406 248 37\n",
      "23\t2764\t0.0\t2\t2489 231 44\n",
      "24\t2561\t0.0\t2\t2277 251 33\n",
      "25\t2459\t0.0\t2\t2175 250 34\n",
      "26\t2413\t0.0\t2\t2136 237 40\n",
      "27\t2256\t0.0\t2\t1991 227 38\n",
      "28\t2129\t0.0\t2\t1888 213 28\n",
      "29\t2184\t0.0\t2\t1905 245 31 3\n",
      "30\t2059\t0.0\t3\t1790 220 36 13\n",
      "31\t1975\t0.0\t3\t1755 175 25 20\n",
      "32\t1913\t0.0\t3\t1696 180 27 10\n",
      "33\t1824\t0.0\t3\t1576 112 116 20\n",
      "34\t1783\t0.0\t3\t1 1591 112 79\n",
      "35\t1585\t0.0\t3\t0 1 1402 182\n",
      "36\t1461\t0.0\t3\t0 0 1357 104\n",
      "37\t1357\t0.0\t3\t0 0 7 1350\n",
      "38\t1262\t0.0\t3\t0 0 3 1258 1\n",
      "39\t1265\t0.0\t3\t0 0 0 9 1256\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40\t18\t0.0\t4\t0 0 0 1 17\n",
      "41\t1\t0.0\t4\t0 0 0 0 1\n",
      "42\t1\t0.0\t4\t0 0 0 0 1\n",
      "44\t1\t0.0\t4\t0 0 0 0 1\n",
      "49\t2\t0.0\t4\t0 0 0 0 0 2\n",
      "50\t11\t0.0\t5\t0 0 0 0 0 11\n",
      "51\t9\t0.0\t5\t0 0 0 0 0 9\n",
      "52\t6\t0.0\t5\t0 0 0 0 0 6\n",
      "53\t4\t0.0\t5\t0 0 0 0 0 4\n",
      "54\t3\t0.0\t5\t0 0 0 0 0 3\n",
      "55\t5\t0.0\t5\t0 0 0 0 1 4\n",
      "56\t5\t0.0\t5\t0 0 0 0 0 5\n",
      "57\t8\t0.0\t5\t0 0 0 0 0 8\n",
      "58\t1\t0.0\t5\t0 0 0 0 0 0 1\n",
      "59\t5\t0.0\t5\t0 0 0 0 0 5\n",
      "60\t127\t0.0\t6\t0 0 0 0 0 1 126\n",
      "61\t127\t0.0\t6\t0 0 0 0 0 0 127\n",
      "62\t128\t0.0\t6\t0 0 0 0 0 1 127\n",
      "63\t89\t0.0\t6\t0 0 0 0 0 1 88\n",
      "64\t68\t0.0\t6\t0 0 0 0 0 3 65\n",
      "65\t72\t0.0\t6\t0 0 0 0 0 0 72\n",
      "66\t50\t0.0\t6\t0 0 0 0 0 3 47\n",
      "67\t48\t0.0\t6\t0 0 0 0 0 2 46\n",
      "68\t57\t0.0\t6\t0 0 0 0 0 2 55\n",
      "69\t44\t0.0\t6\t0 0 0 0 0 1 43\n",
      "70\t39\t0.0\t6\t0 0 0 0 0 0 39\n",
      "71\t43\t0.0\t6\t0 0 0 0 0 0 43\n",
      "72\t40\t0.0\t6\t0 0 0 0 0 1 39\n",
      "73\t31\t0.0\t6\t0 0 0 0 0 1 30\n",
      "74\t34\t0.0\t6\t0 0 0 0 0 0 34\n",
      "75\t32\t0.0\t6\t0 0 0 0 0 2 30\n",
      "76\t20\t0.0\t6\t0 0 0 0 0 1 19\n",
      "77\t22\t0.0\t6\t0 0 0 0 0 5 17\n",
      "78\t23\t0.0\t6\t0 0 0 0 0 0 23\n",
      "79\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "80\t18\t0.0\t6\t0 0 0 0 0 1 17\n",
      "81\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "82\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "83\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "84\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "85\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "86\t16\t0.0\t6\t0 0 0 0 0 2 14\n",
      "87\t11\t0.0\t6\t0 0 0 0 0 1 10\n",
      "88\t12\t0.0\t6\t0 0 0 0 0 1 11\n",
      "89\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "90\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "91\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "92\t10\t0.0\t6\t0 0 0 0 0 1 9\n",
      "93\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "94\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "95\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "96\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "97\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "98\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "99\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "100\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "101\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "102\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "103\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "104\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "105\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "106\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "107\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "108\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "111\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "114\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "119\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "120\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "121\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "122\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "129\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "131\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "132\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "133\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "134\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "135\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "138\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "139\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "142\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "143\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "144\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "146\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "149\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "150\t387\t0.0\t6\t0 0 0 0 0 11 376\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 615612 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 39.9%\n",
      "  C: 27.1%\n",
      "  G: 23.7%\n",
      "  T: 9.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t368976\t370103.6\t0\t368976\n",
      "4\t100151\t92525.9\t0\t100151\n",
      "5\t28983\t23131.5\t0\t28983\n",
      "6\t13487\t5782.9\t0\t13487\n",
      "7\t6371\t1445.7\t0\t6371\n",
      "8\t4973\t361.4\t0\t4973\n",
      "9\t5493\t90.4\t0\t4778 715\n",
      "10\t5164\t22.6\t1\t4438 726\n",
      "11\t4668\t5.6\t1\t4244 424\n",
      "12\t4492\t1.4\t1\t4277 215\n",
      "13\t4199\t0.4\t1\t4031 168\n",
      "14\t3999\t0.1\t1\t3853 146\n",
      "15\t3868\t0.0\t1\t3731 137\n",
      "16\t3619\t0.0\t1\t3473 146\n",
      "17\t3517\t0.0\t1\t3376 141\n",
      "18\t3237\t0.0\t1\t3034 203\n",
      "19\t3261\t0.0\t1\t3066 193 2\n",
      "20\t3363\t0.0\t2\t3056 274 33\n",
      "21\t3194\t0.0\t2\t2876 275 43\n",
      "22\t2976\t0.0\t2\t2675 254 47\n",
      "23\t2700\t0.0\t2\t2442 228 30\n",
      "24\t2758\t0.0\t2\t2485 238 35\n",
      "25\t2586\t0.0\t2\t2326 216 44\n",
      "26\t2480\t0.0\t2\t2161 267 52\n",
      "27\t2412\t0.0\t2\t2095 255 62\n",
      "28\t2259\t0.0\t2\t1962 253 43 1\n",
      "29\t2146\t0.0\t2\t1846 239 61\n",
      "30\t2197\t0.0\t3\t1863 276 43 15\n",
      "31\t2092\t0.0\t3\t1769 254 54 15\n",
      "32\t2001\t0.0\t3\t1702 232 49 18\n",
      "33\t1926\t0.0\t3\t1665 186 54 21\n",
      "34\t1820\t0.0\t3\t1 1529 234 56\n",
      "35\t1809\t0.0\t3\t0 1516 219 74\n",
      "36\t1579\t0.0\t3\t0 29 1349 201\n",
      "37\t1529\t0.0\t3\t0 0 37 1492\n",
      "38\t1348\t0.0\t3\t0 1 21 1324 2\n",
      "39\t1284\t0.0\t3\t0 0 13 689 582\n",
      "40\t800\t0.0\t4\t0 0 0 21 779\n",
      "41\t630\t0.0\t4\t0 0 0 6 624\n",
      "42\t624\t0.0\t4\t0 0 0 4 620\n",
      "43\t321\t0.0\t4\t0 0 0 3 318\n",
      "44\t223\t0.0\t4\t0 0 0 3 220\n",
      "45\t68\t0.0\t4\t0 0 0 0 68\n",
      "46\t28\t0.0\t4\t0 0 0 1 27\n",
      "47\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a17_rep3_+T4_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 32.40 s (1 µs/read; 48.33 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         26,100,243\n",
      "  Read 1 with adapter:                 764,164 (2.9%)\n",
      "  Read 2 with adapter:                 731,713 (2.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 398 (0.0%)\n",
      "Pairs written (passing filters):    26,099,845 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,830,072,900 bp\n",
      "  Read 1: 3,915,036,450 bp\n",
      "  Read 2: 3,915,036,450 bp\n",
      "Total written (filtered):  7,821,755,813 bp (99.9%)\n",
      "  Read 1: 3,910,827,710 bp\n",
      "  Read 2: 3,910,928,103 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 764164 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.8%\n",
      "  C: 19.3%\n",
      "  G: 16.6%\n",
      "  T: 31.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t478766\t407816.3\t0\t478766\n",
      "4\t144527\t101954.1\t0\t144527\n",
      "5\t33725\t25488.5\t0\t33725\n",
      "6\t11128\t6372.1\t0\t11128\n",
      "7\t6178\t1593.0\t0\t6178\n",
      "8\t4996\t398.3\t0\t4996\n",
      "9\t4911\t99.6\t0\t4601 310\n",
      "10\t4820\t24.9\t1\t4433 387\n",
      "11\t4451\t6.2\t1\t4187 264\n",
      "12\t4199\t1.6\t1\t3968 231\n",
      "13\t4110\t0.4\t1\t3935 175\n",
      "14\t3752\t0.1\t1\t3584 168\n",
      "15\t3653\t0.0\t1\t3501 152\n",
      "16\t3453\t0.0\t1\t3288 165\n",
      "17\t3296\t0.0\t1\t3149 147\n",
      "18\t3250\t0.0\t1\t3081 165 4\n",
      "19\t3127\t0.0\t1\t2956 169 2\n",
      "20\t3197\t0.0\t2\t2990 159 48\n",
      "21\t3022\t0.0\t2\t2837 147 38\n",
      "22\t2770\t0.0\t2\t2591 127 52\n",
      "23\t2588\t0.0\t2\t2396 163 29\n",
      "24\t2528\t0.0\t2\t2339 144 45\n",
      "25\t2394\t0.0\t2\t2237 121 36\n",
      "26\t2239\t0.0\t2\t2096 115 28\n",
      "27\t2113\t0.0\t2\t1964 123 26\n",
      "28\t2130\t0.0\t2\t1984 115 31\n",
      "29\t2014\t0.0\t2\t1863 126 25\n",
      "30\t1931\t0.0\t3\t1777 116 19 19\n",
      "31\t1901\t0.0\t3\t1764 104 21 12\n",
      "32\t1669\t0.0\t3\t1521 107 27 14\n",
      "33\t1626\t0.0\t3\t1489 81 43 13\n",
      "34\t1614\t0.0\t3\t0 1506 65 43\n",
      "35\t1441\t0.0\t3\t0 3 1367 71\n",
      "36\t1372\t0.0\t3\t0 0 1278 94\n",
      "37\t1308\t0.0\t3\t0 0 0 1308\n",
      "38\t1314\t0.0\t3\t0 0 0 1309 5\n",
      "39\t2\t0.0\t3\t0 0 0 2\n",
      "40\t5\t0.0\t4\t0 0 0 0 5\n",
      "41\t1\t0.0\t4\t0 0 0 0 1\n",
      "42\t1\t0.0\t4\t0 0 0 0 1\n",
      "50\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "51\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "58\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "60\t187\t0.0\t6\t0 0 0 0 0 0 187\n",
      "61\t224\t0.0\t6\t0 0 0 0 0 0 224\n",
      "62\t193\t0.0\t6\t0 0 0 0 0 0 193\n",
      "63\t137\t0.0\t6\t0 0 0 0 0 0 137\n",
      "64\t125\t0.0\t6\t0 0 0 0 0 0 125\n",
      "65\t128\t0.0\t6\t0 0 0 0 0 0 128\n",
      "66\t101\t0.0\t6\t0 0 0 0 0 0 101\n",
      "67\t92\t0.0\t6\t0 0 0 0 0 0 92\n",
      "68\t97\t0.0\t6\t0 0 0 0 0 0 97\n",
      "69\t83\t0.0\t6\t0 0 0 0 0 0 83\n",
      "70\t64\t0.0\t6\t0 0 0 0 0 0 64\n",
      "71\t79\t0.0\t6\t0 0 0 0 0 0 79\n",
      "72\t77\t0.0\t6\t0 0 0 0 0 0 77\n",
      "73\t83\t0.0\t6\t0 0 0 0 0 0 83\n",
      "74\t53\t0.0\t6\t0 0 0 0 0 0 53\n",
      "75\t55\t0.0\t6\t0 0 0 0 0 0 55\n",
      "76\t57\t0.0\t6\t0 0 0 0 0 0 57\n",
      "77\t36\t0.0\t6\t0 0 0 0 0 0 36\n",
      "78\t36\t0.0\t6\t0 0 0 0 0 0 36\n",
      "79\t31\t0.0\t6\t0 0 0 0 0 0 31\n",
      "80\t39\t0.0\t6\t0 0 0 0 0 0 39\n",
      "81\t18\t0.0\t6\t0 0 0 0 0 0 18\n",
      "82\t29\t0.0\t6\t0 0 0 0 0 0 29\n",
      "83\t23\t0.0\t6\t0 0 0 0 0 0 23\n",
      "84\t24\t0.0\t6\t0 0 0 0 0 0 24\n",
      "85\t21\t0.0\t6\t0 0 0 0 0 0 21\n",
      "86\t22\t0.0\t6\t0 0 0 0 0 0 22\n",
      "87\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "88\t20\t0.0\t6\t0 0 0 0 0 0 20\n",
      "89\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "90\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "91\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "92\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "93\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "94\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "95\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "96\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "97\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "98\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "99\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "100\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "101\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "102\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "103\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "104\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "105\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "106\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "108\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "109\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "117\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "118\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "120\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "121\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "122\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "123\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "124\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "125\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "126\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "127\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "128\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "129\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "130\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "131\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "132\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "133\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "134\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "135\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "136\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "137\t3\t0.0\t6\t0 0 0 0 0 0 3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "138\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "139\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "140\t37\t0.0\t6\t0 0 0 0 0 0 37\n",
      "141\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "142\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "143\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "144\t20\t0.0\t6\t0 0 0 0 0 0 20\n",
      "145\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "146\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "147\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "148\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "149\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "150\t181\t0.0\t6\t0 0 0 0 0 0 181\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 731713 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 41.3%\n",
      "  C: 28.2%\n",
      "  G: 22.8%\n",
      "  T: 7.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t449496\t407816.3\t0\t449496\n",
      "4\t138957\t101954.1\t0\t138957\n",
      "5\t29137\t25488.5\t0\t29137\n",
      "6\t12931\t6372.1\t0\t12931\n",
      "7\t6868\t1593.0\t0\t6868\n",
      "8\t5070\t398.3\t0\t5070\n",
      "9\t5253\t99.6\t0\t4785 468\n",
      "10\t5211\t24.9\t1\t4656 555\n",
      "11\t4845\t6.2\t1\t4489 356\n",
      "12\t4432\t1.6\t1\t4249 183\n",
      "13\t4233\t0.4\t1\t4048 185\n",
      "14\t4181\t0.1\t1\t4029 152\n",
      "15\t3831\t0.0\t1\t3683 148\n",
      "16\t3718\t0.0\t1\t3537 181\n",
      "17\t3510\t0.0\t1\t3381 129\n",
      "18\t3343\t0.0\t1\t3129 213 1\n",
      "19\t3306\t0.0\t1\t3103 199 4\n",
      "20\t3226\t0.0\t2\t3018 190 18\n",
      "21\t3264\t0.0\t2\t3065 176 23\n",
      "22\t3069\t0.0\t2\t2878 167 24\n",
      "23\t2830\t0.0\t2\t2650 150 30\n",
      "24\t2639\t0.0\t2\t2456 165 18\n",
      "25\t2576\t0.0\t2\t2397 153 26\n",
      "26\t2446\t0.0\t2\t2211 199 36\n",
      "27\t2281\t0.0\t2\t2051 199 31\n",
      "28\t2171\t0.0\t2\t1978 173 20\n",
      "29\t2171\t0.0\t2\t1914 229 27 1\n",
      "30\t2078\t0.0\t3\t1852 184 30 12\n",
      "31\t1959\t0.0\t3\t1749 170 25 15\n",
      "32\t1937\t0.0\t3\t1717 176 26 18\n",
      "33\t1701\t0.0\t3\t1484 167 37 13\n",
      "34\t1642\t0.0\t3\t1444 146 45 7\n",
      "35\t1639\t0.0\t3\t9 1461 118 51\n",
      "36\t1481\t0.0\t3\t0 15 1334 132\n",
      "37\t1397\t0.0\t3\t0 0 17 1374 6\n",
      "38\t1338\t0.0\t3\t0 0 26 1287 25\n",
      "39\t1338\t0.0\t3\t0 0 0 24 1314\n",
      "40\t71\t0.0\t4\t0 0 0 26 45\n",
      "41\t49\t0.0\t4\t0 0 0 1 48\n",
      "42\t37\t0.0\t4\t0 0 0 0 37\n",
      "43\t24\t0.0\t4\t0 0 0 0 24\n",
      "44\t16\t0.0\t4\t0 0 0 0 16\n",
      "45\t9\t0.0\t4\t0 0 0 0 9\n",
      "46\t2\t0.0\t4\t0 0 0 0 2\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a18_rep3_control_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 66.37 s (2 µs/read; 33.37 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         36,916,469\n",
      "  Read 1 with adapter:               1,183,063 (3.2%)\n",
      "  Read 2 with adapter:                 879,237 (2.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,858 (0.0%)\n",
      "Pairs written (passing filters):    36,914,611 (100.0%)\n",
      "\n",
      "Total basepairs processed: 11,074,940,700 bp\n",
      "  Read 1: 5,537,470,350 bp\n",
      "  Read 2: 5,537,470,350 bp\n",
      "Total written (filtered):  11,062,359,783 bp (99.9%)\n",
      "  Read 1: 5,530,097,564 bp\n",
      "  Read 2: 5,532,262,219 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 1183063 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.9%\n",
      "  C: 23.5%\n",
      "  G: 16.8%\n",
      "  T: 29.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t715020\t576819.8\t0\t715020\n",
      "4\t236897\t144205.0\t0\t236897\n",
      "5\t61862\t36051.2\t0\t61862\n",
      "6\t18931\t9012.8\t0\t18931\n",
      "7\t9031\t2253.2\t0\t9031\n",
      "8\t6390\t563.3\t0\t6390\n",
      "9\t6929\t140.8\t0\t6106 823\n",
      "10\t6772\t35.2\t1\t5776 996\n",
      "11\t6022\t8.8\t1\t5618 404\n",
      "12\t5555\t2.2\t1\t5274 281\n",
      "13\t5361\t0.6\t1\t5150 211\n",
      "14\t5025\t0.1\t1\t4811 214\n",
      "15\t4806\t0.0\t1\t4623 183\n",
      "16\t4649\t0.0\t1\t4418 231\n",
      "17\t4417\t0.0\t1\t4207 210\n",
      "18\t4229\t0.0\t1\t4043 183 3\n",
      "19\t4308\t0.0\t1\t4118 185 5\n",
      "20\t4187\t0.0\t2\t3957 189 41\n",
      "21\t3827\t0.0\t2\t3637 156 34\n",
      "22\t3554\t0.0\t2\t3334 188 32\n",
      "23\t3535\t0.0\t2\t3328 172 35\n",
      "24\t3107\t0.0\t2\t2933 142 32\n",
      "25\t3130\t0.0\t2\t2951 138 41\n",
      "26\t2929\t0.0\t2\t2758 140 31\n",
      "27\t2759\t0.0\t2\t2574 157 28\n",
      "28\t2664\t0.0\t2\t2474 154 36\n",
      "29\t2705\t0.0\t2\t2519 152 32 2\n",
      "30\t2423\t0.0\t3\t2235 146 23 19\n",
      "31\t2429\t0.0\t3\t2243 137 36 13\n",
      "32\t2247\t0.0\t3\t2075 128 28 16\n",
      "33\t2153\t0.0\t3\t1969 142 25 17\n",
      "34\t2057\t0.0\t3\t1887 92 59 19\n",
      "35\t1869\t0.0\t3\t0 1741 109 19\n",
      "36\t1848\t0.0\t3\t0 1726 74 48\n",
      "37\t1806\t0.0\t3\t0 2 1682 122\n",
      "38\t1697\t0.0\t3\t0 3 1618 76\n",
      "39\t1596\t0.0\t3\t0 0 7 1585 4\n",
      "40\t1457\t0.0\t4\t0 0 0 10 1447\n",
      "41\t1337\t0.0\t4\t0 0 0 12 1325\n",
      "42\t1282\t0.0\t4\t0 0 1 8 1273\n",
      "43\t1183\t0.0\t4\t0 0 0 3 1180\n",
      "44\t1002\t0.0\t4\t0 0 1 5 996\n",
      "45\t829\t0.0\t4\t0 0 0 5 824\n",
      "46\t801\t0.0\t4\t0 0 0 6 795\n",
      "47\t759\t0.0\t4\t0 0 0 7 752\n",
      "48\t706\t0.0\t4\t0 0 0 4 702\n",
      "49\t718\t0.0\t4\t0 0 0 5 700 13\n",
      "50\t803\t0.0\t5\t0 0 0 3 628 172\n",
      "51\t842\t0.0\t5\t0 0 0 4 620 218\n",
      "52\t763\t0.0\t5\t0 0 0 0 569 194\n",
      "53\t671\t0.0\t5\t0 0 0 2 514 155\n",
      "54\t608\t0.0\t5\t0 0 0 2 438 168\n",
      "55\t541\t0.0\t5\t0 0 0 0 386 155\n",
      "56\t515\t0.0\t5\t0 0 0 0 350 165\n",
      "57\t514\t0.0\t5\t0 0 0 0 364 150\n",
      "58\t499\t0.0\t5\t0 0 0 1 361 137\n",
      "59\t464\t0.0\t5\t0 0 0 1 333 130\n",
      "60\t419\t0.0\t6\t0 0 0 0 256 137 26\n",
      "61\t434\t0.0\t6\t0 0 0 0 268 129 37\n",
      "62\t437\t0.0\t6\t0 0 0 1 275 127 34\n",
      "63\t341\t0.0\t6\t0 0 0 0 191 104 46\n",
      "64\t372\t0.0\t6\t0 0 0 2 214 107 49\n",
      "65\t295\t0.0\t6\t0 0 0 0 167 93 35\n",
      "66\t275\t0.0\t6\t0 0 0 0 155 84 36\n",
      "67\t254\t0.0\t6\t0 0 0 0 151 74 29\n",
      "68\t242\t0.0\t6\t0 0 0 0 141 69 32\n",
      "69\t227\t0.0\t6\t0 0 0 0 115 81 31\n",
      "70\t231\t0.0\t6\t0 0 0 0 140 68 23\n",
      "71\t215\t0.0\t6\t0 0 0 0 131 65 19\n",
      "72\t213\t0.0\t6\t0 0 0 0 121 64 28\n",
      "73\t209\t0.0\t6\t0 0 0 0 113 71 25\n",
      "74\t182\t0.0\t6\t0 0 0 0 97 58 27\n",
      "75\t170\t0.0\t6\t0 0 0 0 98 56 16\n",
      "76\t162\t0.0\t6\t0 0 0 0 86 54 22\n",
      "77\t122\t0.0\t6\t0 0 0 0 68 34 20\n",
      "78\t113\t0.0\t6\t0 0 0 0 61 35 17\n",
      "79\t110\t0.0\t6\t0 0 0 1 69 30 10\n",
      "80\t97\t0.0\t6\t0 0 0 0 52 32 13\n",
      "81\t105\t0.0\t6\t0 0 0 0 61 34 10\n",
      "82\t108\t0.0\t6\t0 0 0 0 62 38 8\n",
      "83\t102\t0.0\t6\t0 0 0 0 54 35 13\n",
      "84\t75\t0.0\t6\t0 0 0 0 42 21 12\n",
      "85\t80\t0.0\t6\t0 0 0 0 44 25 11\n",
      "86\t72\t0.0\t6\t0 0 0 0 42 17 13\n",
      "87\t77\t0.0\t6\t0 0 0 0 45 23 9\n",
      "88\t47\t0.0\t6\t0 0 0 0 21 19 7\n",
      "89\t65\t0.0\t6\t0 0 0 0 34 25 6\n",
      "90\t36\t0.0\t6\t0 0 0 0 19 14 3\n",
      "91\t34\t0.0\t6\t0 0 0 0 23 7 4\n",
      "92\t39\t0.0\t6\t0 0 0 0 21 13 5\n",
      "93\t28\t0.0\t6\t0 0 0 0 13 10 5\n",
      "94\t32\t0.0\t6\t0 0 0 0 16 9 7\n",
      "95\t30\t0.0\t6\t0 0 0 0 19 7 4\n",
      "96\t36\t0.0\t6\t0 0 0 0 16 11 9\n",
      "97\t28\t0.0\t6\t0 0 0 0 13 9 6\n",
      "98\t22\t0.0\t6\t0 0 0 0 10 6 6\n",
      "99\t29\t0.0\t6\t0 0 0 0 14 7 8\n",
      "100\t20\t0.0\t6\t0 0 0 0 9 8 3\n",
      "101\t15\t0.0\t6\t0 0 0 0 5 7 3\n",
      "102\t20\t0.0\t6\t0 0 0 0 5 7 8\n",
      "103\t22\t0.0\t6\t0 0 0 0 8 11 3\n",
      "104\t15\t0.0\t6\t0 0 0 0 7 5 3\n",
      "105\t7\t0.0\t6\t0 0 0 0 1 2 4\n",
      "106\t12\t0.0\t6\t0 0 0 0 7 3 2\n",
      "107\t3\t0.0\t6\t0 0 0 0 1 1 1\n",
      "108\t8\t0.0\t6\t0 0 0 0 5 0 3\n",
      "109\t11\t0.0\t6\t0 0 0 0 5 3 3\n",
      "110\t7\t0.0\t6\t0 0 0 0 2 3 2\n",
      "111\t11\t0.0\t6\t0 0 0 0 5 4 2\n",
      "112\t6\t0.0\t6\t0 0 0 0 1 1 4\n",
      "113\t9\t0.0\t6\t0 0 0 0 2 2 5\n",
      "114\t7\t0.0\t6\t0 0 0 0 1 5 1\n",
      "115\t4\t0.0\t6\t0 0 0 0 1 1 2\n",
      "116\t7\t0.0\t6\t0 0 0 0 3 2 2\n",
      "117\t7\t0.0\t6\t0 0 0 0 3 3 1\n",
      "118\t21\t0.0\t6\t0 0 0 0 7 9 5\n",
      "119\t13\t0.0\t6\t0 0 0 0 2 6 5\n",
      "120\t21\t0.0\t6\t0 0 0 0 10 9 2\n",
      "121\t8\t0.0\t6\t0 0 0 0 3 1 4\n",
      "122\t4\t0.0\t6\t0 0 0 0 0 2 2\n",
      "123\t8\t0.0\t6\t0 0 0 0 2 5 1\n",
      "124\t8\t0.0\t6\t0 0 0 0 3 2 3\n",
      "125\t18\t0.0\t6\t0 0 0 0 7 6 5\n",
      "126\t9\t0.0\t6\t0 0 0 0 4 2 3\n",
      "127\t22\t0.0\t6\t0 0 0 0 10 7 5\n",
      "128\t19\t0.0\t6\t0 0 0 0 9 3 7\n",
      "129\t11\t0.0\t6\t0 0 0 0 7 2 2\n",
      "130\t12\t0.0\t6\t0 0 0 0 6 3 3\n",
      "131\t16\t0.0\t6\t0 0 0 0 3 8 5\n",
      "132\t12\t0.0\t6\t0 0 0 0 4 6 2\n",
      "133\t17\t0.0\t6\t0 0 0 0 9 4 4\n",
      "134\t14\t0.0\t6\t0 0 0 0 7 7\n",
      "135\t66\t0.0\t6\t0 0 0 0 30 22 14\n",
      "136\t11\t0.0\t6\t0 0 0 0 2 5 4\n",
      "137\t13\t0.0\t6\t0 0 0 0 3 3 7\n",
      "138\t23\t0.0\t6\t0 0 0 0 10 8 5\n",
      "139\t18\t0.0\t6\t0 0 0 0 11 7\n",
      "140\t27\t0.0\t6\t0 0 0 0 6 12 9\n",
      "141\t15\t0.0\t6\t0 0 0 0 4 6 5\n",
      "142\t28\t0.0\t6\t0 0 0 0 15 9 4\n",
      "143\t22\t0.0\t6\t0 0 0 0 10 9 3\n",
      "144\t50\t0.0\t6\t0 0 0 0 24 15 11\n",
      "145\t12\t0.0\t6\t0 0 0 0 4 8\n",
      "146\t9\t0.0\t6\t0 0 0 0 3 2 4\n",
      "147\t4\t0.0\t6\t0 0 0 0 1 0 3\n",
      "148\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "149\t33\t0.0\t6\t0 0 0 0 12 13 8\n",
      "150\t1121\t0.0\t6\t0 0 0 2 411 445 263\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 879237 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 40.0%\n",
      "  C: 28.2%\n",
      "  G: 22.6%\n",
      "  T: 9.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t548488\t576819.8\t0\t548488\n",
      "4\t140583\t144205.0\t0\t140583\n",
      "5\t41493\t36051.2\t0\t41493\n",
      "6\t20154\t9012.8\t0\t20154\n",
      "7\t7624\t2253.2\t0\t7624\n",
      "8\t6684\t563.3\t0\t6684\n",
      "9\t7618\t140.8\t0\t6308 1310\n",
      "10\t6732\t35.2\t1\t6089 643\n",
      "11\t6175\t8.8\t1\t5820 355\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12\t5963\t2.2\t1\t5619 344\n",
      "13\t5551\t0.6\t1\t5271 280\n",
      "14\t5406\t0.1\t1\t5191 215\n",
      "15\t5055\t0.0\t1\t4839 216\n",
      "16\t4831\t0.0\t1\t4587 244\n",
      "17\t4679\t0.0\t1\t4472 207\n",
      "18\t4426\t0.0\t1\t4146 280\n",
      "19\t4242\t0.0\t1\t3966 272 4\n",
      "20\t4386\t0.0\t2\t4000 321 65\n",
      "21\t4222\t0.0\t2\t3819 331 72\n",
      "22\t3839\t0.0\t2\t3493 286 60\n",
      "23\t3583\t0.0\t2\t3270 272 41\n",
      "24\t3549\t0.0\t2\t3232 274 43\n",
      "25\t3110\t0.0\t2\t2793 266 51\n",
      "26\t3147\t0.0\t2\t2789 303 55\n",
      "27\t2968\t0.0\t2\t2615 305 48\n",
      "28\t2789\t0.0\t2\t2460 274 55\n",
      "29\t2690\t0.0\t2\t2340 278 71 1\n",
      "30\t2755\t0.0\t3\t2381 294 55 25\n",
      "31\t2460\t0.0\t3\t2130 257 51 22\n",
      "32\t2454\t0.0\t3\t2103 276 49 26\n",
      "33\t2272\t0.0\t3\t1995 182 73 22\n",
      "34\t2163\t0.0\t3\t0 1880 232 51\n",
      "35\t2070\t0.0\t3\t0 1759 211 100\n",
      "36\t1830\t0.0\t3\t0 2 1645 183\n",
      "37\t1647\t0.0\t3\t0 0 8 1635 4\n",
      "38\t1566\t0.0\t3\t0 0 1 1565\n",
      "39\t25\t0.0\t3\t0 0 0 24 1\n",
      "40\t5\t0.0\t4\t0 0 0 2 3\n",
      "41\t1\t0.0\t4\t0 0 0 1\n",
      "42\t1\t0.0\t4\t0 0 0 0 1\n",
      "43\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_1.fastq ./20211105180528381-5936/original/DEFAULT/a1_rep1_control_t-5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 52.79 s (2 µs/read; 29.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         25,986,950\n",
      "  Read 1 with adapter:                 908,085 (3.5%)\n",
      "  Read 2 with adapter:                 727,467 (2.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs written (passing filters):    25,986,950 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,796,085,000 bp\n",
      "  Read 1: 3,898,042,500 bp\n",
      "  Read 2: 3,898,042,500 bp\n",
      "Total written (filtered):  7,784,791,793 bp (99.9%)\n",
      "  Read 1: 3,892,304,531 bp\n",
      "  Read 2: 3,892,487,262 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 908085 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 37.0%\n",
      "  C: 20.7%\n",
      "  G: 15.8%\n",
      "  T: 26.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t501992\t406046.1\t0\t501992\n",
      "4\t165715\t101511.5\t0\t165715\n",
      "5\t47488\t25377.9\t0\t47488\n",
      "6\t18961\t6344.5\t0\t18961\n",
      "7\t11287\t1586.1\t0\t11287\n",
      "8\t9250\t396.5\t0\t9250\n",
      "9\t9413\t99.1\t0\t8757 656\n",
      "10\t9351\t24.8\t1\t8310 1041\n",
      "11\t8542\t6.2\t1\t8035 507\n",
      "12\t8279\t1.5\t1\t7879 400\n",
      "13\t7797\t0.4\t1\t7517 280\n",
      "14\t7188\t0.1\t1\t6917 271\n",
      "15\t6955\t0.0\t1\t6679 276\n",
      "16\t6625\t0.0\t1\t6353 272\n",
      "17\t6195\t0.0\t1\t5931 264\n",
      "18\t5914\t0.0\t1\t5622 290 2\n",
      "19\t5978\t0.0\t1\t5720 255 3\n",
      "20\t5628\t0.0\t2\t5344 235 49\n",
      "21\t5408\t0.0\t2\t5091 259 58\n",
      "22\t5163\t0.0\t2\t4866 236 61\n",
      "23\t4807\t0.0\t2\t4507 243 57\n",
      "24\t4548\t0.0\t2\t4309 204 35\n",
      "25\t4453\t0.0\t2\t4176 229 48\n",
      "26\t4290\t0.0\t2\t4018 229 43\n",
      "27\t4075\t0.0\t2\t3789 227 59\n",
      "28\t3809\t0.0\t2\t3565 197 47\n",
      "29\t3559\t0.0\t2\t3336 182 39 2\n",
      "30\t3517\t0.0\t3\t3240 195 53 29\n",
      "31\t3378\t0.0\t3\t3135 171 53 19\n",
      "32\t3232\t0.0\t3\t2980 188 34 30\n",
      "33\t3036\t0.0\t3\t2836 109 69 22\n",
      "34\t2783\t0.0\t3\t0 2602 106 75\n",
      "35\t2572\t0.0\t3\t0 3 2464 105\n",
      "36\t2470\t0.0\t3\t0 0 11 2459\n",
      "37\t2316\t0.0\t3\t0 0 0 7 2309\n",
      "38\t2089\t0.0\t3\t0 0 0 16 2073\n",
      "39\t13\t0.0\t3\t0 0 0 1 12\n",
      "40\t2\t0.0\t4\t0 0 0 1 1\n",
      "62\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "66\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "67\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "68\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "74\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 727467 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 35.0%\n",
      "  C: 29.1%\n",
      "  G: 26.0%\n",
      "  T: 9.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t380268\t406046.1\t0\t380268\n",
      "4\t104151\t101511.5\t0\t104151\n",
      "5\t34497\t25377.9\t0\t34497\n",
      "6\t18646\t6344.5\t0\t18646\n",
      "7\t10738\t1586.1\t0\t10738\n",
      "8\t9876\t396.5\t0\t9876\n",
      "9\t9930\t99.1\t0\t9131 799\n",
      "10\t9428\t24.8\t1\t8675 753\n",
      "11\t8765\t6.2\t1\t8301 464\n",
      "12\t8484\t1.5\t1\t8024 460\n",
      "13\t8279\t0.4\t1\t7906 373\n",
      "14\t7848\t0.1\t1\t7543 305\n",
      "15\t7227\t0.0\t1\t6906 321\n",
      "16\t7006\t0.0\t1\t6659 347\n",
      "17\t6654\t0.0\t1\t6359 295\n",
      "18\t6215\t0.0\t1\t5811 402 2\n",
      "19\t5952\t0.0\t1\t5559 385 8\n",
      "20\t6040\t0.0\t2\t5496 466 78\n",
      "21\t5655\t0.0\t2\t5108 472 75\n",
      "22\t5435\t0.0\t2\t4910 447 78\n",
      "23\t5194\t0.0\t2\t4710 412 72\n",
      "24\t4823\t0.0\t2\t4319 432 72\n",
      "25\t4577\t0.0\t2\t4131 364 82\n",
      "26\t4486\t0.0\t2\t3907 483 96\n",
      "27\t4315\t0.0\t2\t3732 503 80\n",
      "28\t4078\t0.0\t2\t3571 414 93\n",
      "29\t3833\t0.0\t2\t3325 421 83 4\n",
      "30\t3620\t0.0\t3\t3135 382 77 26\n",
      "31\t3545\t0.0\t3\t3041 394 71 39\n",
      "32\t3412\t0.0\t3\t2913 389 79 31\n",
      "33\t3293\t0.0\t3\t2812 373 77 31\n",
      "34\t3066\t0.0\t3\t2602 363 72 29\n",
      "35\t2839\t0.0\t3\t2430 256 117 36\n",
      "36\t2662\t0.0\t3\t0 2303 264 95\n",
      "37\t2548\t0.0\t3\t0 3 2288 257\n",
      "38\t2246\t0.0\t3\t0 0 2 2217 27\n",
      "39\t2088\t0.0\t3\t0 0 1 895 1192\n",
      "40\t2036\t0.0\t4\t0 0 4 810 1222\n",
      "41\t1835\t0.0\t4\t0 0 0 242 1593\n",
      "42\t851\t0.0\t4\t0 0 0 6 845\n",
      "43\t501\t0.0\t4\t0 0 0 2 499\n",
      "44\t352\t0.0\t4\t0 0 0 0 352\n",
      "45\t152\t0.0\t4\t0 0 0 0 152\n",
      "46\t20\t0.0\t4\t0 0 0 0 20\n",
      "47\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_1.fastq ./20211105180528381-5936/original/DEFAULT/a2_rep1_+T4_t5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 43.65 s (2 µs/read; 37.34 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         27,164,185\n",
      "  Read 1 with adapter:                 860,087 (3.2%)\n",
      "  Read 2 with adapter:                 651,482 (2.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,478 (0.0%)\n",
      "Pairs written (passing filters):    27,162,707 (100.0%)\n",
      "\n",
      "Total basepairs processed: 8,149,255,500 bp\n",
      "  Read 1: 4,074,627,750 bp\n",
      "  Read 2: 4,074,627,750 bp\n",
      "Total written (filtered):  8,140,196,615 bp (99.9%)\n",
      "  Read 1: 4,069,711,680 bp\n",
      "  Read 2: 4,070,484,935 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 860087 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.6%\n",
      "  C: 24.6%\n",
      "  G: 16.8%\n",
      "  T: 29.9%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t532943\t424440.4\t0\t532943\n",
      "4\t169263\t106110.1\t0\t169263\n",
      "5\t46970\t26527.5\t0\t46970\n",
      "6\t14151\t6631.9\t0\t14151\n",
      "7\t6374\t1658.0\t0\t6374\n",
      "8\t4542\t414.5\t0\t4542\n",
      "9\t4861\t103.6\t0\t4164 697\n",
      "10\t4928\t25.9\t1\t4115 813\n",
      "11\t4206\t6.5\t1\t3891 315\n",
      "12\t3958\t1.6\t1\t3747 211\n",
      "13\t3686\t0.4\t1\t3573 113\n",
      "14\t3553\t0.1\t1\t3440 113\n",
      "15\t3287\t0.0\t1\t3153 134\n",
      "16\t3197\t0.0\t1\t3067 130\n",
      "17\t2987\t0.0\t1\t2879 108\n",
      "18\t2967\t0.0\t1\t2830 135 2\n",
      "19\t2883\t0.0\t1\t2755 127 1\n",
      "20\t2776\t0.0\t2\t2645 106 25\n",
      "21\t2617\t0.0\t2\t2445 139 33\n",
      "22\t2448\t0.0\t2\t2285 135 28\n",
      "23\t2377\t0.0\t2\t2247 100 30\n",
      "24\t2186\t0.0\t2\t2069 97 20\n",
      "25\t2122\t0.0\t2\t1998 94 30\n",
      "26\t2080\t0.0\t2\t1967 87 26\n",
      "27\t1923\t0.0\t2\t1782 118 22 1\n",
      "28\t1904\t0.0\t2\t1766 105 33\n",
      "29\t1781\t0.0\t2\t1667 81 31 2\n",
      "30\t1748\t0.0\t3\t1630 87 21 10\n",
      "31\t1645\t0.0\t3\t1513 92 29 11\n",
      "32\t1642\t0.0\t3\t1512 89 26 15\n",
      "33\t1431\t0.0\t3\t1315 73 29 14\n",
      "34\t1441\t0.0\t3\t1321 59 50 11\n",
      "35\t1333\t0.0\t3\t0 1239 56 38\n",
      "36\t1237\t0.0\t3\t0 3 1157 77\n",
      "37\t1210\t0.0\t3\t0 2 1121 58 29\n",
      "38\t1126\t0.0\t3\t0 0 0 1071 55\n",
      "39\t1025\t0.0\t3\t0 0 0 1022 3\n",
      "40\t952\t0.0\t4\t0 0 0 9 943\n",
      "41\t8\t0.0\t4\t0 0 0 0 8\n",
      "42\t8\t0.0\t4\t0 0 0 0 8\n",
      "43\t4\t0.0\t4\t0 0 0 0 4\n",
      "44\t2\t0.0\t4\t0 0 0 0 2\n",
      "45\t3\t0.0\t4\t0 0 0 0 3\n",
      "47\t2\t0.0\t4\t0 0 0 0 2\n",
      "48\t2\t0.0\t4\t0 0 0 0 2\n",
      "49\t4\t0.0\t4\t0 0 0 0 4\n",
      "50\t505\t0.0\t5\t0 0 0 0 1 504\n",
      "51\t386\t0.0\t5\t0 0 0 0 1 385\n",
      "52\t398\t0.0\t5\t0 0 0 0 0 398\n",
      "53\t385\t0.0\t5\t0 0 0 0 1 384\n",
      "54\t304\t0.0\t5\t0 0 0 0 0 304\n",
      "55\t297\t0.0\t5\t0 0 0 0 0 297\n",
      "56\t226\t0.0\t5\t0 0 0 0 1 225\n",
      "57\t216\t0.0\t5\t0 0 0 0 0 216\n",
      "58\t251\t0.0\t5\t0 0 0 0 0 251\n",
      "59\t239\t0.0\t5\t0 0 0 0 1 234 4\n",
      "60\t306\t0.0\t6\t0 0 0 0 0 211 95\n",
      "61\t288\t0.0\t6\t0 0 0 0 0 184 104\n",
      "62\t216\t0.0\t6\t0 0 0 0 0 138 78\n",
      "63\t216\t0.0\t6\t0 0 0 0 0 145 71\n",
      "64\t198\t0.0\t6\t0 0 0 0 1 132 65\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "65\t172\t0.0\t6\t0 0 0 0 1 117 54\n",
      "66\t182\t0.0\t6\t0 0 0 0 1 118 63\n",
      "67\t136\t0.0\t6\t0 0 0 0 0 100 36\n",
      "68\t121\t0.0\t6\t0 0 0 0 0 73 48\n",
      "69\t117\t0.0\t6\t0 0 0 0 0 71 46\n",
      "70\t131\t0.0\t6\t0 0 0 0 0 90 41\n",
      "71\t131\t0.0\t6\t0 0 0 0 0 81 50\n",
      "72\t118\t0.0\t6\t0 0 0 0 0 88 30\n",
      "73\t115\t0.0\t6\t0 0 0 0 0 78 37\n",
      "74\t97\t0.0\t6\t0 0 0 0 0 66 31\n",
      "75\t103\t0.0\t6\t0 0 0 0 0 76 27\n",
      "76\t92\t0.0\t6\t0 0 0 0 1 60 31\n",
      "77\t73\t0.0\t6\t0 0 0 0 0 47 26\n",
      "78\t65\t0.0\t6\t0 0 0 0 0 41 24\n",
      "79\t49\t0.0\t6\t0 0 0 0 0 27 22\n",
      "80\t64\t0.0\t6\t0 0 0 0 0 40 24\n",
      "81\t62\t0.0\t6\t0 0 0 0 0 41 21\n",
      "82\t46\t0.0\t6\t0 0 0 0 0 29 17\n",
      "83\t36\t0.0\t6\t0 0 0 0 0 24 12\n",
      "84\t54\t0.0\t6\t0 0 0 0 0 31 23\n",
      "85\t45\t0.0\t6\t0 0 0 0 0 30 15\n",
      "86\t42\t0.0\t6\t0 0 0 0 0 27 15\n",
      "87\t43\t0.0\t6\t0 0 0 0 0 23 20\n",
      "88\t41\t0.0\t6\t0 0 0 0 0 23 18\n",
      "89\t33\t0.0\t6\t0 0 0 0 0 21 12\n",
      "90\t27\t0.0\t6\t0 0 0 0 0 18 9\n",
      "91\t27\t0.0\t6\t0 0 0 0 0 15 12\n",
      "92\t30\t0.0\t6\t0 0 0 0 0 17 13\n",
      "93\t19\t0.0\t6\t0 0 0 0 0 10 9\n",
      "94\t21\t0.0\t6\t0 0 0 0 0 10 11\n",
      "95\t24\t0.0\t6\t0 0 0 0 0 15 9\n",
      "96\t23\t0.0\t6\t0 0 0 0 0 14 9\n",
      "97\t19\t0.0\t6\t0 0 0 0 0 9 10\n",
      "98\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "99\t9\t0.0\t6\t0 0 0 0 0 6 3\n",
      "100\t7\t0.0\t6\t0 0 0 0 0 4 3\n",
      "101\t9\t0.0\t6\t0 0 0 0 0 4 5\n",
      "102\t6\t0.0\t6\t0 0 0 0 0 5 1\n",
      "103\t6\t0.0\t6\t0 0 0 0 0 1 5\n",
      "104\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "105\t13\t0.0\t6\t0 0 0 0 0 5 8\n",
      "106\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "107\t11\t0.0\t6\t0 0 0 0 0 6 5\n",
      "108\t7\t0.0\t6\t0 0 0 0 0 2 5\n",
      "109\t7\t0.0\t6\t0 0 0 0 0 2 5\n",
      "110\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "111\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "112\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "113\t8\t0.0\t6\t0 0 0 0 0 3 5\n",
      "114\t8\t0.0\t6\t0 0 0 0 1 5 2\n",
      "115\t7\t0.0\t6\t0 0 0 0 1 5 1\n",
      "116\t4\t0.0\t6\t0 0 0 0 0 3 1\n",
      "117\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "118\t8\t0.0\t6\t0 0 0 0 0 3 5\n",
      "119\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "120\t5\t0.0\t6\t0 0 0 0 0 4 1\n",
      "121\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "122\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "123\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "124\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "125\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "126\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "127\t30\t0.0\t6\t0 0 0 0 0 14 16\n",
      "128\t23\t0.0\t6\t0 0 0 0 0 19 4\n",
      "129\t12\t0.0\t6\t0 0 0 0 0 7 5\n",
      "130\t17\t0.0\t6\t0 0 0 0 0 8 9\n",
      "131\t18\t0.0\t6\t0 0 0 0 0 14 4\n",
      "132\t8\t0.0\t6\t0 0 0 0 0 3 5\n",
      "133\t11\t0.0\t6\t0 0 0 0 0 6 5\n",
      "134\t10\t0.0\t6\t0 0 0 0 0 4 6\n",
      "135\t75\t0.0\t6\t0 0 0 0 0 44 31\n",
      "136\t10\t0.0\t6\t0 0 0 0 0 4 6\n",
      "137\t9\t0.0\t6\t0 0 0 0 0 4 5\n",
      "138\t20\t0.0\t6\t0 0 0 0 0 12 8\n",
      "139\t20\t0.0\t6\t0 0 0 0 0 11 9\n",
      "140\t35\t0.0\t6\t0 0 0 0 0 15 20\n",
      "141\t16\t0.0\t6\t0 0 0 0 0 9 7\n",
      "142\t30\t0.0\t6\t0 0 0 0 0 13 17\n",
      "143\t24\t0.0\t6\t0 0 0 0 0 9 15\n",
      "144\t46\t0.0\t6\t0 0 0 0 0 16 30\n",
      "145\t16\t0.0\t6\t0 0 0 0 0 8 8\n",
      "146\t24\t0.0\t6\t0 0 0 0 0 14 10\n",
      "147\t4\t0.0\t6\t0 0 0 0 0 2 2\n",
      "148\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "149\t32\t0.0\t6\t0 0 0 0 0 12 20\n",
      "150\t817\t0.0\t6\t0 0 0 0 0 432 385\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 651482 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 39.6%\n",
      "  C: 28.6%\n",
      "  G: 22.4%\n",
      "  T: 9.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t401137\t424440.4\t0\t401137\n",
      "4\t105287\t106110.1\t0\t105287\n",
      "5\t31696\t26527.5\t0\t31696\n",
      "6\t14403\t6631.9\t0\t14403\n",
      "7\t5515\t1658.0\t0\t5515\n",
      "8\t4727\t414.5\t0\t4727\n",
      "9\t5346\t103.6\t0\t4543 803\n",
      "10\t4856\t25.9\t1\t4162 694\n",
      "11\t4569\t6.5\t1\t4213 356\n",
      "12\t4146\t1.6\t1\t3936 210\n",
      "13\t3947\t0.4\t1\t3765 182\n",
      "14\t3734\t0.1\t1\t3590 144\n",
      "15\t3574\t0.0\t1\t3439 135\n",
      "16\t3318\t0.0\t1\t3166 152\n",
      "17\t3222\t0.0\t1\t3060 162\n",
      "18\t3021\t0.0\t1\t2843 176 2\n",
      "19\t2993\t0.0\t1\t2815 176 2\n",
      "20\t2937\t0.0\t2\t2706 191 40\n",
      "21\t2795\t0.0\t2\t2552 208 35\n",
      "22\t2643\t0.0\t2\t2347 245 51\n",
      "23\t2458\t0.0\t2\t2231 195 32\n",
      "24\t2401\t0.0\t2\t2203 163 35\n",
      "25\t2221\t0.0\t2\t2020 177 24\n",
      "26\t2151\t0.0\t2\t1926 181 44\n",
      "27\t2115\t0.0\t2\t1871 206 38\n",
      "28\t1940\t0.0\t2\t1706 196 38\n",
      "29\t1931\t0.0\t2\t1688 204 39\n",
      "30\t1817\t0.0\t3\t1573 190 41 13\n",
      "31\t1771\t0.0\t3\t1551 170 30 20\n",
      "32\t1679\t0.0\t3\t1467 156 43 13\n",
      "33\t1672\t0.0\t3\t1464 130 59 19\n",
      "34\t1445\t0.0\t3\t0 1286 137 22\n",
      "35\t1438\t0.0\t3\t0 1262 140 36\n",
      "36\t1346\t0.0\t3\t1 1146 166 33\n",
      "37\t1262\t0.0\t3\t0 1081 141 39 1\n",
      "38\t1237\t0.0\t3\t2 1010 174 51\n",
      "39\t1162\t0.0\t3\t0 944 152 63 3\n",
      "40\t1113\t0.0\t4\t0 1 924 155 33\n",
      "41\t1056\t0.0\t4\t0 0 862 152 42\n",
      "42\t1022\t0.0\t4\t0 0 849 116 57\n",
      "43\t950\t0.0\t4\t0 0 27 798 125\n",
      "44\t936\t0.0\t4\t0 0 3 800 133\n",
      "45\t857\t0.0\t4\t0 0 0 27 830\n",
      "46\t620\t0.0\t4\t0 0 0 10 610\n",
      "47\t626\t0.0\t4\t0 0 0 6 620\n",
      "48\t192\t0.0\t4\t0 0 0 2 190\n",
      "49\t98\t0.0\t4\t0 0 0 0 98\n",
      "50\t67\t0.0\t5\t0 0 0 0 67\n",
      "51\t6\t0.0\t5\t0 0 0 0 5 1\n",
      "52\t8\t0.0\t5\t0 0 0 0 8\n",
      "53\t6\t0.0\t5\t0 0 0 0 0 6\n",
      "54\t13\t0.0\t5\t0 0 0 0 0 13\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_1.fastq ./20211105180528381-5936/original/DEFAULT/a3_rep1_+T4_t10_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 42.59 s (2 µs/read; 38.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         27,081,806\n",
      "  Read 1 with adapter:                 906,607 (3.3%)\n",
      "  Read 2 with adapter:                 746,595 (2.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.0%)\n",
      "Pairs written (passing filters):    27,081,803 (100.0%)\n",
      "\n",
      "Total basepairs processed: 8,124,541,800 bp\n",
      "  Read 1: 4,062,270,900 bp\n",
      "  Read 2: 4,062,270,900 bp\n",
      "Total written (filtered):  8,114,030,881 bp (99.9%)\n",
      "  Read 1: 4,056,982,228 bp\n",
      "  Read 2: 4,057,048,653 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 906607 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 34.5%\n",
      "  C: 22.5%\n",
      "  G: 15.5%\n",
      "  T: 27.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t525052\t423153.2\t0\t525052\n",
      "4\t169722\t105788.3\t0\t169722\n",
      "5\t47372\t26447.1\t0\t47372\n",
      "6\t17510\t6611.8\t0\t17510\n",
      "7\t9882\t1652.9\t0\t9882\n",
      "8\t7814\t413.2\t0\t7814\n",
      "9\t8330\t103.3\t0\t7518 812\n",
      "10\t7895\t25.8\t1\t7042 853\n",
      "11\t7109\t6.5\t1\t6699 410\n",
      "12\t6707\t1.6\t1\t6390 317\n",
      "13\t6433\t0.4\t1\t6196 237\n",
      "14\t6117\t0.1\t1\t5893 224\n",
      "15\t5846\t0.0\t1\t5622 224\n",
      "16\t5545\t0.0\t1\t5307 238\n",
      "17\t5317\t0.0\t1\t5103 214\n",
      "18\t5159\t0.0\t1\t4914 245\n",
      "19\t5138\t0.0\t1\t4869 264 5\n",
      "20\t5160\t0.0\t2\t4867 237 56\n",
      "21\t4596\t0.0\t2\t4333 221 42\n",
      "22\t4277\t0.0\t2\t4034 200 43\n",
      "23\t4192\t0.0\t2\t3938 217 37\n",
      "24\t3860\t0.0\t2\t3632 187 41\n",
      "25\t3835\t0.0\t2\t3584 204 47\n",
      "26\t3585\t0.0\t2\t3324 211 50\n",
      "27\t3306\t0.0\t2\t3095 181 30\n",
      "28\t3227\t0.0\t2\t3006 173 48\n",
      "29\t3257\t0.0\t2\t3042 181 33 1\n",
      "30\t3162\t0.0\t3\t2865 218 50 29\n",
      "31\t2998\t0.0\t3\t2746 195 41 16\n",
      "32\t2823\t0.0\t3\t2593 165 41 24\n",
      "33\t2590\t0.0\t3\t2387 124 62 17\n",
      "34\t2444\t0.0\t3\t1 2249 128 66\n",
      "35\t2207\t0.0\t3\t0 2 2098 107\n",
      "36\t2073\t0.0\t3\t0 0 20 2052 1\n",
      "37\t2002\t0.0\t3\t0 0 0 26 1976\n",
      "38\t37\t0.0\t3\t0 0 0 31 6\n",
      "39\t12\t0.0\t3\t0 0 0 2 10\n",
      "41\t1\t0.0\t4\t0 0 0 0 1\n",
      "60\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "61\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "62\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "63\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "64\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "68\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "78\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "80\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "90\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "129\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "140\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 746595 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 37.2%\n",
      "  C: 28.1%\n",
      "  G: 24.9%\n",
      "  T: 9.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t414842\t423153.2\t0\t414842\n",
      "4\t112440\t105788.3\t0\t112440\n",
      "5\t36063\t26447.1\t0\t36063\n",
      "6\t18817\t6611.8\t0\t18817\n",
      "7\t9796\t1652.9\t0\t9796\n",
      "8\t8280\t413.2\t0\t8280\n",
      "9\t8673\t103.3\t0\t7847 826\n",
      "10\t8552\t25.8\t1\t7533 1019\n",
      "11\t7655\t6.5\t1\t7106 549\n",
      "12\t7064\t1.6\t1\t6743 321\n",
      "13\t6694\t0.4\t1\t6390 304\n",
      "14\t6464\t0.1\t1\t6232 232\n",
      "15\t6155\t0.0\t1\t5913 242\n",
      "16\t5882\t0.0\t1\t5612 270\n",
      "17\t5591\t0.0\t1\t5346 245\n",
      "18\t5330\t0.0\t1\t5026 301 3\n",
      "19\t5199\t0.0\t1\t4915 279 5\n",
      "20\t5213\t0.0\t2\t4758 377 78\n",
      "21\t5177\t0.0\t2\t4680 416 81\n",
      "22\t4617\t0.0\t2\t4215 353 49\n",
      "23\t4312\t0.0\t2\t3963 300 49\n",
      "24\t4205\t0.0\t2\t3800 360 45\n",
      "25\t3895\t0.0\t2\t3538 303 54\n",
      "26\t3863\t0.0\t2\t3416 394 53\n",
      "27\t3615\t0.0\t2\t3135 412 67 1\n",
      "28\t3334\t0.0\t2\t2912 351 71\n",
      "29\t3250\t0.0\t2\t2806 363 80 1\n",
      "30\t3326\t0.0\t3\t2882 356 58 30\n",
      "31\t3188\t0.0\t3\t2778 323 61 26\n",
      "32\t3016\t0.0\t3\t2638 303 48 27\n",
      "33\t2860\t0.0\t3\t2481 304 54 21\n",
      "34\t2605\t0.0\t3\t2224 257 94 30\n",
      "35\t2467\t0.0\t3\t1 2139 273 54\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\t2306\t0.0\t3\t0 2045 226 35\n",
      "37\t2241\t0.0\t3\t0 1941 216 84\n",
      "38\t2134\t0.0\t3\t2 1 1905 226\n",
      "39\t2007\t0.0\t3\t0 0 1789 217 1\n",
      "40\t1671\t0.0\t4\t0 0 0 1666 5\n",
      "41\t1627\t0.0\t4\t0 0 0 1561 66\n",
      "42\t1546\t0.0\t4\t0 0 0 0 1546\n",
      "43\t260\t0.0\t4\t0 0 0 0 260\n",
      "44\t208\t0.0\t4\t0 0 0 0 208\n",
      "45\t130\t0.0\t4\t0 0 0 0 130\n",
      "46\t9\t0.0\t4\t0 0 0 0 9\n",
      "47\t14\t0.0\t4\t0 0 0 0 14\n",
      "48\t2\t0.0\t4\t0 0 0 0 2\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_1.fastq ./20211105180528381-5936/original/DEFAULT/a4_rep1_+T4_t15_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 38.55 s (2 µs/read; 38.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         24,850,635\n",
      "  Read 1 with adapter:                 759,963 (3.1%)\n",
      "  Read 2 with adapter:                 665,570 (2.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 866 (0.0%)\n",
      "Pairs written (passing filters):    24,849,769 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,455,190,500 bp\n",
      "  Read 1: 3,727,595,250 bp\n",
      "  Read 2: 3,727,595,250 bp\n",
      "Total written (filtered):  7,446,897,411 bp (99.9%)\n",
      "  Read 1: 3,723,299,333 bp\n",
      "  Read 2: 3,723,598,078 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 759963 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.7%\n",
      "  C: 22.1%\n",
      "  G: 16.1%\n",
      "  T: 29.0%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t465888\t388291.2\t0\t465888\n",
      "4\t143859\t97072.8\t0\t143859\n",
      "5\t37480\t24268.2\t0\t37480\n",
      "6\t12824\t6067.0\t0\t12824\n",
      "7\t6691\t1516.8\t0\t6691\n",
      "8\t5299\t379.2\t0\t5299\n",
      "9\t5453\t94.8\t0\t4840 613\n",
      "10\t5191\t23.7\t1\t4681 510\n",
      "11\t4765\t5.9\t1\t4489 276\n",
      "12\t4503\t1.5\t1\t4300 203\n",
      "13\t4295\t0.4\t1\t4114 181\n",
      "14\t3969\t0.1\t1\t3779 190\n",
      "15\t3920\t0.0\t1\t3754 166\n",
      "16\t3785\t0.0\t1\t3625 160\n",
      "17\t3430\t0.0\t1\t3277 153\n",
      "18\t3379\t0.0\t1\t3201 178\n",
      "19\t3332\t0.0\t1\t3184 147 1\n",
      "20\t3223\t0.0\t2\t3055 138 30\n",
      "21\t3046\t0.0\t2\t2848 166 32\n",
      "22\t2908\t0.0\t2\t2736 141 31\n",
      "23\t2575\t0.0\t2\t2409 144 22\n",
      "24\t2502\t0.0\t2\t2332 143 27\n",
      "25\t2357\t0.0\t2\t2199 133 25\n",
      "26\t2338\t0.0\t2\t2191 121 26\n",
      "27\t2232\t0.0\t2\t2077 124 30 1\n",
      "28\t2091\t0.0\t2\t1940 119 32\n",
      "29\t1989\t0.0\t2\t1841 123 23 2\n",
      "30\t2015\t0.0\t3\t1845 128 27 15\n",
      "31\t1992\t0.0\t3\t1843 112 26 11\n",
      "32\t1739\t0.0\t3\t1577 117 30 15\n",
      "33\t1613\t0.0\t3\t1463 80 53 17\n",
      "34\t1564\t0.0\t3\t0 1436 83 45\n",
      "35\t1438\t0.0\t3\t0 10 1375 53\n",
      "36\t1305\t0.0\t3\t0 0 15 1290\n",
      "37\t1302\t0.0\t3\t0 0 7 1295\n",
      "38\t433\t0.0\t3\t0 0 2 431\n",
      "39\t13\t0.0\t3\t0 0 0 13\n",
      "40\t9\t0.0\t4\t0 0 1 8\n",
      "41\t9\t0.0\t4\t0 0 0 0 9\n",
      "45\t1\t0.0\t4\t0 0 0 0 1\n",
      "50\t3\t0.0\t5\t0 0 0 0 0 3\n",
      "51\t4\t0.0\t5\t0 0 0 0 0 4\n",
      "52\t2\t0.0\t5\t0 0 0 0 0 2\n",
      "53\t3\t0.0\t5\t0 0 0 0 0 3\n",
      "54\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "55\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "57\t4\t0.0\t5\t0 0 0 0 0 4\n",
      "58\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "59\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "60\t209\t0.0\t6\t0 0 0 0 0 0 209\n",
      "61\t188\t0.0\t6\t0 0 0 0 0 0 188\n",
      "62\t180\t0.0\t6\t0 0 0 0 0 1 179\n",
      "63\t171\t0.0\t6\t0 0 0 0 0 0 171\n",
      "64\t124\t0.0\t6\t0 0 0 0 0 0 124\n",
      "65\t122\t0.0\t6\t0 0 0 0 0 1 121\n",
      "66\t118\t0.0\t6\t0 0 0 0 0 1 117\n",
      "67\t101\t0.0\t6\t0 0 0 0 0 0 101\n",
      "68\t82\t0.0\t6\t0 0 0 0 0 0 82\n",
      "69\t92\t0.0\t6\t0 0 0 0 0 0 92\n",
      "70\t75\t0.0\t6\t0 0 0 0 0 1 74\n",
      "71\t70\t0.0\t6\t0 0 0 0 0 0 70\n",
      "72\t67\t0.0\t6\t0 0 0 0 0 0 67\n",
      "73\t51\t0.0\t6\t0 0 0 0 0 0 51\n",
      "74\t56\t0.0\t6\t0 0 0 0 0 0 56\n",
      "75\t54\t0.0\t6\t0 0 0 0 0 0 54\n",
      "76\t35\t0.0\t6\t0 0 0 0 0 0 35\n",
      "77\t39\t0.0\t6\t0 0 0 0 0 0 39\n",
      "78\t57\t0.0\t6\t0 0 0 0 0 1 56\n",
      "79\t48\t0.0\t6\t0 0 0 0 0 0 48\n",
      "80\t31\t0.0\t6\t0 0 0 0 0 0 31\n",
      "81\t38\t0.0\t6\t0 0 0 0 0 0 38\n",
      "82\t30\t0.0\t6\t0 0 0 0 0 0 30\n",
      "83\t22\t0.0\t6\t0 0 0 0 0 0 22\n",
      "84\t31\t0.0\t6\t0 0 0 0 0 0 31\n",
      "85\t29\t0.0\t6\t0 0 0 0 0 0 29\n",
      "86\t26\t0.0\t6\t0 0 0 0 0 0 26\n",
      "87\t23\t0.0\t6\t0 0 0 0 0 0 23\n",
      "88\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "89\t26\t0.0\t6\t0 0 0 0 0 0 26\n",
      "90\t14\t0.0\t6\t0 0 0 0 0 0 14\n",
      "91\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "92\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "93\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "94\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "95\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "96\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "97\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "98\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "99\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "100\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "101\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "102\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "103\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "104\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "105\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "106\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "107\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "108\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "109\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "110\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "111\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "112\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "113\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "114\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "115\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "116\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "117\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "118\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "119\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "120\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "121\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "123\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "124\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "125\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "126\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "127\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "128\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "129\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "130\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "131\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "132\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "133\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "134\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "135\t25\t0.0\t6\t0 0 0 0 0 0 25\n",
      "136\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "137\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "138\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "139\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "140\t28\t0.0\t6\t0 0 0 0 0 0 28\n",
      "141\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "142\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "143\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "144\t29\t0.0\t6\t0 0 0 0 0 0 29\n",
      "145\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "146\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "147\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "148\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "149\t21\t0.0\t6\t0 0 0 0 0 0 21\n",
      "150\t576\t0.0\t6\t0 0 0 0 0 1 575\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 665570 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 40.8%\n",
      "  C: 26.6%\n",
      "  G: 23.8%\n",
      "  T: 8.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t400830\t388291.2\t0\t400830\n",
      "4\t113946\t97072.8\t0\t113946\n",
      "5\t30652\t24268.2\t0\t30652\n",
      "6\t14650\t6067.0\t0\t14650\n",
      "7\t7531\t1516.8\t0\t7531\n",
      "8\t5518\t379.2\t0\t5518\n",
      "9\t5919\t94.8\t0\t5216 703\n",
      "10\t5650\t23.7\t1\t4864 786\n",
      "11\t5171\t5.9\t1\t4725 446\n",
      "12\t4743\t1.5\t1\t4475 268\n",
      "13\t4507\t0.4\t1\t4330 177\n",
      "14\t4329\t0.1\t1\t4158 171\n",
      "15\t4002\t0.0\t1\t3839 163\n",
      "16\t3953\t0.0\t1\t3769 184\n",
      "17\t3823\t0.0\t1\t3643 180\n",
      "18\t3466\t0.0\t1\t3242 223 1\n",
      "19\t3402\t0.0\t1\t3177 221 4\n",
      "20\t3379\t0.0\t2\t3080 248 51\n",
      "21\t3255\t0.0\t2\t2934 269 52\n",
      "22\t3082\t0.0\t2\t2785 261 36\n",
      "23\t2916\t0.0\t2\t2650 240 26\n",
      "24\t2596\t0.0\t2\t2341 218 37\n",
      "25\t2529\t0.0\t2\t2266 238 25\n",
      "26\t2380\t0.0\t2\t2111 240 29\n",
      "27\t2366\t0.0\t2\t2084 236 46\n",
      "28\t2239\t0.0\t2\t1974 221 44\n",
      "29\t2124\t0.0\t2\t1874 198 52\n",
      "30\t2032\t0.0\t3\t1782 206 31 13\n",
      "31\t2020\t0.0\t3\t1757 222 31 10\n",
      "32\t2004\t0.0\t3\t1760 199 28 17\n",
      "33\t1763\t0.0\t3\t1553 133 60 17\n",
      "34\t1625\t0.0\t3\t1 1433 156 35\n",
      "35\t1578\t0.0\t3\t0 1362 149 67\n",
      "36\t1486\t0.0\t3\t0 2 1310 174\n",
      "37\t1381\t0.0\t3\t0 1 1227 153\n",
      "38\t1272\t0.0\t3\t0 0 0 1251 21\n",
      "39\t1182\t0.0\t3\t0 1 0 1092 89\n",
      "40\t84\t0.0\t4\t0 0 0 5 79\n",
      "41\t66\t0.0\t4\t0 0 0 0 66\n",
      "42\t66\t0.0\t4\t0 0 0 0 66\n",
      "43\t41\t0.0\t4\t0 0 0 0 41\n",
      "44\t7\t0.0\t4\t0 0 0 0 7\n",
      "45\t2\t0.0\t4\t0 0 0 0 2\n",
      "46\t3\t0.0\t4\t0 0 0 0 3\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a5_rep1_+T4_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 52.41 s (2 µs/read; 34.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         30,475,132\n",
      "  Read 1 with adapter:               1,005,437 (3.3%)\n",
      "  Read 2 with adapter:                 949,400 (3.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               2,071 (0.0%)\n",
      "Pairs written (passing filters):    30,473,061 (100.0%)\n",
      "\n",
      "Total basepairs processed: 9,142,539,600 bp\n",
      "  Read 1: 4,571,269,800 bp\n",
      "  Read 2: 4,571,269,800 bp\n",
      "Total written (filtered):  9,128,120,284 bp (99.8%)\n",
      "  Read 1: 4,563,639,260 bp\n",
      "  Read 2: 4,564,481,024 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 1005437 times\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 40.5%\n",
      "  C: 18.4%\n",
      "  G: 14.9%\n",
      "  T: 26.2%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t565180\t476173.9\t0\t565180\n",
      "4\t166171\t119043.5\t0\t166171\n",
      "5\t45863\t29760.9\t0\t45863\n",
      "6\t17853\t7440.2\t0\t17853\n",
      "7\t11780\t1860.1\t0\t11780\n",
      "8\t10206\t465.0\t0\t10206\n",
      "9\t10018\t116.3\t0\t9513 505\n",
      "10\t9472\t29.1\t1\t8829 643\n",
      "11\t9175\t7.3\t1\t8651 524\n",
      "12\t8524\t1.8\t1\t8154 370\n",
      "13\t8296\t0.5\t1\t7972 324\n",
      "14\t7490\t0.1\t1\t7135 355\n",
      "15\t7510\t0.0\t1\t7185 325\n",
      "16\t7054\t0.0\t1\t6734 320\n",
      "17\t6598\t0.0\t1\t6329 269\n",
      "18\t6705\t0.0\t1\t6416 287 2\n",
      "19\t6614\t0.0\t1\t6290 321 3\n",
      "20\t6343\t0.0\t2\t5974 293 76\n",
      "21\t6101\t0.0\t2\t5706 313 82\n",
      "22\t5810\t0.0\t2\t5445 304 61\n",
      "23\t5265\t0.0\t2\t4952 245 68\n",
      "24\t4966\t0.0\t2\t4689 225 52\n",
      "25\t4681\t0.0\t2\t4392 234 55\n",
      "26\t4620\t0.0\t2\t4315 244 61\n",
      "27\t4307\t0.0\t2\t4005 247 55\n",
      "28\t4117\t0.0\t2\t3844 217 56\n",
      "29\t3955\t0.0\t2\t3707 199 49\n",
      "30\t3866\t0.0\t3\t3553 241 46 26\n",
      "31\t3753\t0.0\t3\t3452 224 59 18\n",
      "32\t3485\t0.0\t3\t3241 168 51 25\n",
      "33\t3204\t0.0\t3\t2950 187 47 20\n",
      "34\t3089\t0.0\t3\t2879 112 80 18\n",
      "35\t2879\t0.0\t3\t0 2683 113 83\n",
      "36\t2774\t0.0\t3\t0 2 2623 149\n",
      "37\t2705\t0.0\t3\t0 0 2531 174\n",
      "38\t2590\t0.0\t3\t0 2 2459 128 1\n",
      "39\t2376\t0.0\t3\t0 0 2 2359 15\n",
      "40\t2142\t0.0\t4\t0 0 0 6 2136\n",
      "41\t2077\t0.0\t4\t0 0 0 8 2069\n",
      "42\t10\t0.0\t4\t0 0 0 0 10\n",
      "43\t9\t0.0\t4\t0 0 0 0 9\n",
      "44\t4\t0.0\t4\t0 0 0 0 4\n",
      "45\t4\t0.0\t4\t0 0 0 0 4\n",
      "46\t3\t0.0\t4\t0 0 0 0 3\n",
      "47\t1\t0.0\t4\t0 0 0 0 1\n",
      "48\t1\t0.0\t4\t0 0 0 0 1\n",
      "49\t2\t0.0\t4\t0 0 0 0 2\n",
      "50\t960\t0.0\t5\t0 0 0 0 2 958\n",
      "51\t877\t0.0\t5\t0 0 0 0 1 876\n",
      "52\t798\t0.0\t5\t0 0 0 0 0 798\n",
      "53\t741\t0.0\t5\t0 0 0 0 4 737\n",
      "54\t694\t0.0\t5\t0 0 0 0 2 692\n",
      "55\t646\t0.0\t5\t0 0 0 0 1 645\n",
      "56\t515\t0.0\t5\t0 0 0 0 2 513\n",
      "57\t506\t0.0\t5\t0 0 0 0 1 505\n",
      "58\t554\t0.0\t5\t0 0 0 0 0 554\n",
      "59\t486\t0.0\t5\t0 0 0 0 0 473 13\n",
      "60\t561\t0.0\t6\t0 0 0 0 2 392 167\n",
      "61\t615\t0.0\t6\t0 0 0 0 1 418 196\n",
      "62\t524\t0.0\t6\t0 0 0 0 0 357 167\n",
      "63\t493\t0.0\t6\t0 0 0 0 1 355 137\n",
      "64\t426\t0.0\t6\t0 0 0 0 0 305 121\n",
      "65\t377\t0.0\t6\t0 0 0 0 0 274 103\n",
      "66\t324\t0.0\t6\t0 0 0 0 0 214 110\n",
      "67\t293\t0.0\t6\t0 0 0 0 0 203 90\n",
      "68\t262\t0.0\t6\t0 0 0 0 0 177 85\n",
      "69\t258\t0.0\t6\t0 0 0 0 0 171 87\n",
      "70\t232\t0.0\t6\t0 0 0 0 0 159 73\n",
      "71\t215\t0.0\t6\t0 0 0 0 1 156 58\n",
      "72\t216\t0.0\t6\t0 0 0 0 1 160 55\n",
      "73\t214\t0.0\t6\t0 0 0 0 0 141 73\n",
      "74\t187\t0.0\t6\t0 0 0 0 0 132 55\n",
      "75\t156\t0.0\t6\t0 0 0 0 0 112 44\n",
      "76\t140\t0.0\t6\t0 0 0 0 2 91 47\n",
      "77\t133\t0.0\t6\t0 0 0 0 0 81 52\n",
      "78\t171\t0.0\t6\t0 0 0 0 0 117 54\n",
      "79\t100\t0.0\t6\t0 0 0 0 0 73 27\n",
      "80\t109\t0.0\t6\t0 0 0 0 0 78 31\n",
      "81\t95\t0.0\t6\t0 0 0 0 0 62 33\n",
      "82\t91\t0.0\t6\t0 0 0 0 0 56 35\n",
      "83\t84\t0.0\t6\t0 0 0 0 0 65 19\n",
      "84\t68\t0.0\t6\t0 0 0 0 0 51 17\n",
      "85\t81\t0.0\t6\t0 0 0 0 0 61 20\n",
      "86\t67\t0.0\t6\t0 0 0 0 0 37 30\n",
      "87\t50\t0.0\t6\t0 0 0 0 0 34 16\n",
      "88\t47\t0.0\t6\t0 0 0 0 0 31 16\n",
      "89\t50\t0.0\t6\t0 0 0 0 0 41 9\n",
      "90\t38\t0.0\t6\t0 0 0 0 0 23 15\n",
      "91\t28\t0.0\t6\t0 0 0 0 0 19 9\n",
      "92\t27\t0.0\t6\t0 0 0 0 0 15 12\n",
      "93\t32\t0.0\t6\t0 0 0 0 0 19 13\n",
      "94\t46\t0.0\t6\t0 0 0 0 0 26 20\n",
      "95\t32\t0.0\t6\t0 0 0 0 0 24 8\n",
      "96\t23\t0.0\t6\t0 0 0 0 0 14 9\n",
      "97\t18\t0.0\t6\t0 0 0 0 0 13 5\n",
      "98\t22\t0.0\t6\t0 0 0 0 0 17 5\n",
      "99\t25\t0.0\t6\t0 0 0 0 0 21 4\n",
      "100\t11\t0.0\t6\t0 0 0 0 0 10 1\n",
      "101\t16\t0.0\t6\t0 0 0 0 0 8 8\n",
      "102\t16\t0.0\t6\t0 0 0 0 0 11 5\n",
      "103\t10\t0.0\t6\t0 0 0 0 0 8 2\n",
      "104\t6\t0.0\t6\t0 0 0 0 0 1 5\n",
      "105\t7\t0.0\t6\t0 0 0 0 0 6 1\n",
      "106\t6\t0.0\t6\t0 0 0 0 0 4 2\n",
      "107\t7\t0.0\t6\t0 0 0 0 0 2 5\n",
      "108\t8\t0.0\t6\t0 0 0 0 0 6 2\n",
      "109\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "110\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "111\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "112\t6\t0.0\t6\t0 0 0 0 0 6\n",
      "113\t5\t0.0\t6\t0 0 0 0 0 4 1\n",
      "114\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "115\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "116\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "117\t6\t0.0\t6\t0 0 0 0 1 0 5\n",
      "118\t17\t0.0\t6\t0 0 0 0 0 9 8\n",
      "119\t7\t0.0\t6\t0 0 0 0 0 4 3\n",
      "120\t13\t0.0\t6\t0 0 0 0 0 10 3\n",
      "121\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "122\t10\t0.0\t6\t0 0 0 0 0 5 5\n",
      "123\t9\t0.0\t6\t0 0 0 0 0 4 5\n",
      "124\t5\t0.0\t6\t0 0 0 0 0 4 1\n",
      "125\t14\t0.0\t6\t0 0 0 0 0 3 11\n",
      "126\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "127\t25\t0.0\t6\t0 0 0 0 0 15 10\n",
      "128\t18\t0.0\t6\t0 0 0 0 0 13 5\n",
      "129\t18\t0.0\t6\t0 0 0 0 0 12 6\n",
      "130\t26\t0.0\t6\t0 0 0 0 0 14 12\n",
      "131\t33\t0.0\t6\t0 0 0 0 0 21 12\n",
      "132\t17\t0.0\t6\t0 0 0 0 0 7 10\n",
      "133\t11\t0.0\t6\t0 0 0 0 0 4 7\n",
      "134\t17\t0.0\t6\t0 0 0 0 0 5 12\n",
      "135\t69\t0.0\t6\t0 0 0 0 0 36 33\n",
      "136\t13\t0.0\t6\t0 0 0 0 0 11 2\n",
      "137\t16\t0.0\t6\t0 0 0 0 0 11 5\n",
      "138\t40\t0.0\t6\t0 0 0 0 0 22 18\n",
      "139\t25\t0.0\t6\t0 0 0 0 0 13 12\n",
      "140\t147\t0.0\t6\t0 0 0 0 0 94 53\n",
      "141\t56\t0.0\t6\t0 0 0 0 0 36 20\n",
      "142\t62\t0.0\t6\t0 0 0 0 0 32 30\n",
      "143\t40\t0.0\t6\t0 0 0 0 0 22 18\n",
      "144\t174\t0.0\t6\t0 0 0 0 0 99 75\n",
      "145\t26\t0.0\t6\t0 0 0 0 0 14 12\n",
      "146\t58\t0.0\t6\t0 0 0 0 0 26 32\n",
      "147\t10\t0.0\t6\t0 0 0 0 0 4 6\n",
      "148\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "149\t45\t0.0\t6\t0 0 0 0 0 17 28\n",
      "150\t923\t0.0\t6\t0 0 0 0 0 488 435\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 949400 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 38.7%\n",
      "  C: 27.3%\n",
      "  G: 25.3%\n",
      "  T: 8.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t523326\t476173.9\t0\t523326\n",
      "4\t159864\t119043.5\t0\t159864\n",
      "5\t40314\t29760.9\t0\t40314\n",
      "6\t20517\t7440.2\t0\t20517\n",
      "7\t13133\t1860.1\t0\t13133\n",
      "8\t10368\t465.0\t0\t10368\n",
      "9\t10749\t116.3\t0\t10047 702\n",
      "10\t10557\t29.1\t1\t9673 884\n",
      "11\t9459\t7.3\t1\t8936 523\n",
      "12\t9176\t1.8\t1\t8777 399\n",
      "13\t8598\t0.5\t1\t8237 361\n",
      "14\t8404\t0.1\t1\t8126 278\n",
      "15\t7600\t0.0\t1\t7279 321\n",
      "16\t7616\t0.0\t1\t7282 334\n",
      "17\t7165\t0.0\t1\t6866 299\n",
      "18\t6718\t0.0\t1\t6297 421\n",
      "19\t6808\t0.0\t1\t6398 401 9\n",
      "20\t6758\t0.0\t2\t6228 462 68\n",
      "21\t6455\t0.0\t2\t5898 482 75\n",
      "22\t6163\t0.0\t2\t5596 505 62\n",
      "23\t5876\t0.0\t2\t5366 445 65\n",
      "24\t5343\t0.0\t2\t4872 411 60\n",
      "25\t5031\t0.0\t2\t4605 358 68\n",
      "26\t4753\t0.0\t2\t4204 478 71\n",
      "27\t4678\t0.0\t2\t4163 444 71\n",
      "28\t4367\t0.0\t2\t3856 444 67\n",
      "29\t4157\t0.0\t2\t3654 435 66 2\n",
      "30\t4037\t0.0\t3\t3523 436 62 16\n",
      "31\t3919\t0.0\t3\t3415 393 74 37\n",
      "32\t3788\t0.0\t3\t3316 377 75 20\n",
      "33\t3533\t0.0\t3\t3081 310 110 32\n",
      "34\t3235\t0.0\t3\t0 2825 347 63\n",
      "35\t3084\t0.0\t3\t0 2663 327 94\n",
      "36\t2859\t0.0\t3\t0 2 2559 298\n",
      "37\t2775\t0.0\t3\t0 0 2470 305\n",
      "38\t2669\t0.0\t3\t0 2 2368 299\n",
      "39\t2380\t0.0\t3\t0 0 1 2372 7\n",
      "40\t2259\t0.0\t4\t0 0 0 71 2188\n",
      "41\t262\t0.0\t4\t0 0 0 0 262\n",
      "42\t298\t0.0\t4\t0 0 0 1 297\n",
      "43\t122\t0.0\t4\t0 0 0 2 120\n",
      "44\t131\t0.0\t4\t0 0 0 1 130\n",
      "45\t82\t0.0\t4\t0 0 0 0 82\n",
      "46\t14\t0.0\t4\t0 0 0 0 14\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_1.fastq ./20211105180528381-5936/original/DEFAULT/a6_rep1_control_t20_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 57.99 s (2 µs/read; 32.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         31,813,117\n",
      "  Read 1 with adapter:               1,026,603 (3.2%)\n",
      "  Read 2 with adapter:                 769,080 (2.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:               1,385 (0.0%)\n",
      "Pairs written (passing filters):    31,811,732 (100.0%)\n",
      "\n",
      "Total basepairs processed: 9,543,935,100 bp\n",
      "  Read 1: 4,771,967,550 bp\n",
      "  Read 2: 4,771,967,550 bp\n",
      "Total written (filtered):  9,532,919,342 bp (99.9%)\n",
      "  Read 1: 4,765,717,501 bp\n",
      "  Read 2: 4,767,201,841 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 1026603 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.5%\n",
      "  C: 23.6%\n",
      "  G: 16.6%\n",
      "  T: 29.2%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t616552\t497080.0\t0\t616552\n",
      "4\t202446\t124270.0\t0\t202446\n",
      "5\t54794\t31067.5\t0\t54794\n",
      "6\t17239\t7766.9\t0\t17239\n",
      "7\t8510\t1941.7\t0\t8510\n",
      "8\t6236\t485.4\t0\t6236\n",
      "9\t6669\t121.4\t0\t5891 778\n",
      "10\t6585\t30.3\t1\t5557 1028\n",
      "11\t5919\t7.6\t1\t5487 432\n",
      "12\t5374\t1.9\t1\t5096 278\n",
      "13\t5141\t0.5\t1\t4972 169\n",
      "14\t4960\t0.1\t1\t4793 167\n",
      "15\t4764\t0.0\t1\t4604 160\n",
      "16\t4468\t0.0\t1\t4285 183\n",
      "17\t4217\t0.0\t1\t4040 177\n",
      "18\t4164\t0.0\t1\t3944 220\n",
      "19\t4073\t0.0\t1\t3899 171 3\n",
      "20\t3785\t0.0\t2\t3554 200 31\n",
      "21\t3649\t0.0\t2\t3424 182 43\n",
      "22\t3531\t0.0\t2\t3329 175 27\n",
      "23\t3259\t0.0\t2\t3057 161 41\n",
      "24\t3121\t0.0\t2\t2932 151 38\n",
      "25\t3060\t0.0\t2\t2882 149 29\n",
      "26\t2852\t0.0\t2\t2682 142 28\n",
      "27\t2773\t0.0\t2\t2595 146 32\n",
      "28\t2688\t0.0\t2\t2511 148 29\n",
      "29\t2377\t0.0\t2\t2215 130 31 1\n",
      "30\t2479\t0.0\t3\t2292 137 35 15\n",
      "31\t2254\t0.0\t3\t2062 144 34 14\n",
      "32\t2319\t0.0\t3\t2144 127 31 17\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33\t2130\t0.0\t3\t1958 127 27 18\n",
      "34\t1978\t0.0\t3\t1815 83 58 22\n",
      "35\t1939\t0.0\t3\t0 1820 65 54\n",
      "36\t1690\t0.0\t3\t0 8 1611 71\n",
      "37\t1641\t0.0\t3\t0 0 13 1627 1\n",
      "38\t1506\t0.0\t3\t0 0 8 1498\n",
      "39\t1463\t0.0\t3\t0 0 0 6 1457\n",
      "40\t1440\t0.0\t4\t0 0 1 1428 11\n",
      "41\t1408\t0.0\t4\t0 0 0 3 1405\n",
      "42\t9\t0.0\t4\t0 0 0 1 8\n",
      "43\t10\t0.0\t4\t0 0 0 1 9\n",
      "49\t1\t0.0\t4\t0 0 0 0 1\n",
      "50\t655\t0.0\t5\t0 0 0 0 0 655\n",
      "51\t589\t0.0\t5\t0 0 0 0 0 589\n",
      "52\t525\t0.0\t5\t0 0 0 0 0 525\n",
      "53\t479\t0.0\t5\t0 0 0 0 0 479\n",
      "54\t483\t0.0\t5\t0 0 0 0 0 483\n",
      "55\t432\t0.0\t5\t0 0 0 0 0 432\n",
      "56\t418\t0.0\t5\t0 0 0 0 0 418\n",
      "57\t385\t0.0\t5\t0 0 0 0 0 385\n",
      "58\t293\t0.0\t5\t0 0 0 0 0 293\n",
      "59\t336\t0.0\t5\t0 0 0 0 0 335 1\n",
      "60\t422\t0.0\t6\t0 0 0 0 0 300 122\n",
      "61\t379\t0.0\t6\t0 0 0 0 0 261 118\n",
      "62\t359\t0.0\t6\t0 0 0 0 0 248 111\n",
      "63\t300\t0.0\t6\t0 0 0 0 0 195 105\n",
      "64\t291\t0.0\t6\t0 0 0 0 0 183 108\n",
      "65\t230\t0.0\t6\t0 0 0 0 0 146 84\n",
      "66\t206\t0.0\t6\t0 0 0 0 0 129 77\n",
      "67\t201\t0.0\t6\t0 0 0 0 0 135 66\n",
      "68\t224\t0.0\t6\t0 0 0 0 0 154 70\n",
      "69\t180\t0.0\t6\t0 0 0 0 0 106 74\n",
      "70\t182\t0.0\t6\t0 0 0 0 0 124 58\n",
      "71\t191\t0.0\t6\t0 0 0 0 0 123 68\n",
      "72\t154\t0.0\t6\t0 0 0 0 0 97 57\n",
      "73\t158\t0.0\t6\t0 0 0 0 0 102 56\n",
      "74\t149\t0.0\t6\t0 0 0 0 0 95 54\n",
      "75\t155\t0.0\t6\t0 0 0 0 0 97 58\n",
      "76\t103\t0.0\t6\t0 0 0 0 0 62 41\n",
      "77\t124\t0.0\t6\t0 0 0 0 0 81 43\n",
      "78\t105\t0.0\t6\t0 0 0 0 0 58 47\n",
      "79\t89\t0.0\t6\t0 0 0 0 0 64 25\n",
      "80\t97\t0.0\t6\t0 0 0 0 0 64 33\n",
      "81\t76\t0.0\t6\t0 0 0 0 0 61 15\n",
      "82\t89\t0.0\t6\t0 0 0 0 0 62 27\n",
      "83\t72\t0.0\t6\t0 0 0 0 0 47 25\n",
      "84\t84\t0.0\t6\t0 0 0 0 0 49 35\n",
      "85\t56\t0.0\t6\t0 0 0 0 0 38 18\n",
      "86\t50\t0.0\t6\t0 0 0 0 0 32 18\n",
      "87\t62\t0.0\t6\t0 0 0 0 0 36 26\n",
      "88\t47\t0.0\t6\t0 0 0 0 0 30 17\n",
      "89\t41\t0.0\t6\t0 0 0 0 0 23 18\n",
      "90\t45\t0.0\t6\t0 0 0 0 0 31 14\n",
      "91\t29\t0.0\t6\t0 0 0 0 0 19 10\n",
      "92\t32\t0.0\t6\t0 0 0 0 1 22 9\n",
      "93\t26\t0.0\t6\t0 0 0 0 0 18 8\n",
      "94\t29\t0.0\t6\t0 0 0 0 0 17 12\n",
      "95\t20\t0.0\t6\t0 0 0 0 0 13 7\n",
      "96\t20\t0.0\t6\t0 0 0 0 0 10 10\n",
      "97\t24\t0.0\t6\t0 0 0 0 0 14 10\n",
      "98\t21\t0.0\t6\t0 0 0 0 0 11 10\n",
      "99\t12\t0.0\t6\t0 0 0 0 0 7 5\n",
      "100\t16\t0.0\t6\t0 0 0 0 0 11 5\n",
      "101\t14\t0.0\t6\t0 0 0 0 0 7 7\n",
      "102\t11\t0.0\t6\t0 0 0 0 0 7 4\n",
      "103\t13\t0.0\t6\t0 0 0 0 0 9 4\n",
      "104\t8\t0.0\t6\t0 0 0 0 0 4 4\n",
      "105\t6\t0.0\t6\t0 0 0 0 0 5 1\n",
      "106\t10\t0.0\t6\t0 0 0 0 0 7 3\n",
      "107\t8\t0.0\t6\t0 0 0 0 0 5 3\n",
      "108\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "109\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "110\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "111\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "112\t5\t0.0\t6\t0 0 0 0 0 2 3\n",
      "113\t9\t0.0\t6\t0 0 0 0 0 4 5\n",
      "114\t8\t0.0\t6\t0 0 0 0 0 7 1\n",
      "115\t1\t0.0\t6\t0 0 0 0 0 1\n",
      "116\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "117\t11\t0.0\t6\t0 0 0 0 0 6 5\n",
      "118\t18\t0.0\t6\t0 0 0 0 0 9 9\n",
      "119\t6\t0.0\t6\t0 0 0 0 0 4 2\n",
      "120\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "121\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "122\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "123\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "124\t10\t0.0\t6\t0 0 0 0 0 6 4\n",
      "125\t10\t0.0\t6\t0 0 0 0 0 6 4\n",
      "126\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "127\t25\t0.0\t6\t0 0 0 0 0 12 13\n",
      "128\t10\t0.0\t6\t0 0 0 0 0 5 5\n",
      "129\t13\t0.0\t6\t0 0 0 0 0 7 6\n",
      "130\t16\t0.0\t6\t0 0 0 0 0 5 11\n",
      "131\t5\t0.0\t6\t0 0 0 0 0 1 4\n",
      "132\t8\t0.0\t6\t0 0 0 0 0 5 3\n",
      "133\t11\t0.0\t6\t0 0 0 0 0 7 4\n",
      "134\t14\t0.0\t6\t0 0 0 0 0 10 4\n",
      "135\t48\t0.0\t6\t0 0 0 0 0 24 24\n",
      "136\t16\t0.0\t6\t0 0 0 0 0 8 8\n",
      "137\t6\t0.0\t6\t0 0 0 0 0 4 2\n",
      "138\t18\t0.0\t6\t0 0 0 0 0 8 10\n",
      "139\t28\t0.0\t6\t0 0 0 0 0 12 16\n",
      "140\t28\t0.0\t6\t0 0 0 0 0 15 13\n",
      "141\t18\t0.0\t6\t0 0 0 0 0 8 10\n",
      "142\t29\t0.0\t6\t0 0 0 0 0 15 14\n",
      "143\t11\t0.0\t6\t0 0 0 0 0 4 7\n",
      "144\t42\t0.0\t6\t0 0 0 0 0 17 25\n",
      "145\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "146\t27\t0.0\t6\t0 0 0 0 0 16 11\n",
      "147\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "148\t2\t0.0\t6\t0 0 0 0 0 2\n",
      "149\t32\t0.0\t6\t0 0 0 0 0 16 16\n",
      "150\t772\t0.0\t6\t0 0 0 0 0 360 412\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 769080 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 38.8%\n",
      "  C: 28.7%\n",
      "  G: 22.9%\n",
      "  T: 9.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t466322\t497080.0\t0\t466322\n",
      "4\t122160\t124270.0\t0\t122160\n",
      "5\t36687\t31067.5\t0\t36687\n",
      "6\t18122\t7766.9\t0\t18122\n",
      "7\t7493\t1941.7\t0\t7493\n",
      "8\t6603\t485.4\t0\t6603\n",
      "9\t7111\t121.4\t0\t6036 1075\n",
      "10\t6440\t30.3\t1\t5840 600\n",
      "11\t5924\t7.6\t1\t5536 388\n",
      "12\t5822\t1.9\t1\t5451 371\n",
      "13\t5340\t0.5\t1\t5051 289\n",
      "14\t5177\t0.1\t1\t4938 239\n",
      "15\t4976\t0.0\t1\t4732 244\n",
      "16\t4769\t0.0\t1\t4533 236\n",
      "17\t4498\t0.0\t1\t4269 229\n",
      "18\t4208\t0.0\t1\t3902 305 1\n",
      "19\t4197\t0.0\t1\t3873 316 8\n",
      "20\t4120\t0.0\t2\t3699 354 67\n",
      "21\t3816\t0.0\t2\t3376 368 72\n",
      "22\t3672\t0.0\t2\t3264 344 64\n",
      "23\t3534\t0.0\t2\t3133 351 50\n",
      "24\t3294\t0.0\t2\t2929 319 46\n",
      "25\t3129\t0.0\t2\t2783 295 51\n",
      "26\t3080\t0.0\t2\t2665 355 60\n",
      "27\t2861\t0.0\t2\t2466 323 72\n",
      "28\t2787\t0.0\t2\t2408 311 68\n",
      "29\t2721\t0.0\t2\t2278 364 77 2\n",
      "30\t2426\t0.0\t3\t2043 311 48 24\n",
      "31\t2496\t0.0\t3\t2085 321 57 33\n",
      "32\t2253\t0.0\t3\t1907 260 67 19\n",
      "33\t2317\t0.0\t3\t1962 246 82 27\n",
      "34\t2106\t0.0\t3\t0 1786 232 88\n",
      "35\t1934\t0.0\t3\t0 11 1693 230\n",
      "36\t1924\t0.0\t3\t1 1 1644 278\n",
      "37\t1686\t0.0\t3\t0 0 2 1681 3\n",
      "38\t1539\t0.0\t3\t0 0 0 1534 5\n",
      "39\t1348\t0.0\t3\t0 0 0 1345 3\n",
      "40\t56\t0.0\t4\t0 0 0 1 55\n",
      "41\t45\t0.0\t4\t0 0 0 44 1\n",
      "42\t43\t0.0\t4\t0 0 0 0 43\n",
      "43\t16\t0.0\t4\t0 0 0 0 16\n",
      "44\t17\t0.0\t4\t0 0 0 0 17\n",
      "45\t7\t0.0\t4\t0 0 0 0 7\n",
      "46\t3\t0.0\t4\t0 0 0 0 3\n",
      "47\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_1.fastq ./20211105180528381-5936/original/DEFAULT/a7_rep2_control_t-5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 44.57 s (2 µs/read; 31.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         23,405,221\n",
      "  Read 1 with adapter:                 909,813 (3.9%)\n",
      "  Read 2 with adapter:                 725,407 (3.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 596 (0.0%)\n",
      "Pairs written (passing filters):    23,404,625 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,021,566,300 bp\n",
      "  Read 1: 3,510,783,150 bp\n",
      "  Read 2: 3,510,783,150 bp\n",
      "Total written (filtered):  7,007,755,962 bp (99.8%)\n",
      "  Read 1: 3,503,113,861 bp\n",
      "  Read 2: 3,504,642,101 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 909813 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 44.7%\n",
      "  C: 17.6%\n",
      "  G: 13.4%\n",
      "  T: 24.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t442863\t365706.6\t0\t442863\n",
      "4\t157508\t91426.6\t0\t157508\n",
      "5\t43560\t22856.7\t0\t43560\n",
      "6\t21457\t5714.2\t0\t21457\n",
      "7\t13806\t1428.5\t0\t13806\n",
      "8\t11867\t357.1\t0\t11867\n",
      "9\t12031\t89.3\t0\t11471 560\n",
      "10\t11932\t22.3\t1\t10820 1112\n",
      "11\t10902\t5.6\t1\t10349 553\n",
      "12\t10352\t1.4\t1\t9928 424\n",
      "13\t9964\t0.3\t1\t9638 326\n",
      "14\t9404\t0.1\t1\t9075 329\n",
      "15\t8631\t0.0\t1\t8315 316\n",
      "16\t8265\t0.0\t1\t7961 304\n",
      "17\t8158\t0.0\t1\t7883 275\n",
      "18\t7723\t0.0\t1\t7368 354 1\n",
      "19\t7681\t0.0\t1\t7387 289 5\n",
      "20\t7396\t0.0\t2\t6998 341 57\n",
      "21\t6914\t0.0\t2\t6561 294 59\n",
      "22\t6483\t0.0\t2\t6146 273 64\n",
      "23\t5866\t0.0\t2\t5510 297 59\n",
      "24\t5560\t0.0\t2\t5228 280 52\n",
      "25\t5457\t0.0\t2\t5118 284 55\n",
      "26\t5018\t0.0\t2\t4720 241 57\n",
      "27\t4953\t0.0\t2\t4638 269 46\n",
      "28\t4623\t0.0\t2\t4343 231 49\n",
      "29\t4397\t0.0\t2\t4103 234 60\n",
      "30\t4138\t0.0\t3\t3846 224 47 21\n",
      "31\t4186\t0.0\t3\t3872 237 54 23\n",
      "32\t3956\t0.0\t3\t3653 218 46 39\n",
      "33\t3729\t0.0\t3\t3436 182 84 27\n",
      "34\t3567\t0.0\t3\t0 3311 197 59\n",
      "35\t3297\t0.0\t3\t4 3017 220 56\n",
      "36\t3064\t0.0\t3\t0 2844 196 24\n",
      "37\t2846\t0.0\t3\t0 2620 143 83\n",
      "38\t2771\t0.0\t3\t0 0 2601 170\n",
      "39\t2671\t0.0\t3\t0 0 2505 158 8\n",
      "40\t2368\t0.0\t4\t0 0 0 2322 46\n",
      "41\t2145\t0.0\t4\t0 0 0 2129 16\n",
      "42\t2084\t0.0\t4\t0 0 0 2075 9\n",
      "43\t1866\t0.0\t4\t0 0 0 0 1866\n",
      "44\t1798\t0.0\t4\t0 0 0 0 1798\n",
      "45\t1353\t0.0\t4\t0 0 0 0 1353\n",
      "46\t8\t0.0\t4\t0 0 0 0 8\n",
      "47\t5\t0.0\t4\t0 0 0 0 5\n",
      "48\t2\t0.0\t4\t0 0 0 0 2\n",
      "49\t3\t0.0\t4\t0 0 0 0 3\n",
      "50\t890\t0.0\t5\t0 0 0 0 3 887\n",
      "51\t803\t0.0\t5\t0 0 0 0 0 803\n",
      "52\t630\t0.0\t5\t0 0 0 0 0 630\n",
      "53\t638\t0.0\t5\t0 0 0 0 2 636\n",
      "54\t571\t0.0\t5\t0 0 0 0 0 571\n",
      "55\t458\t0.0\t5\t0 0 0 0 0 458\n",
      "56\t445\t0.0\t5\t0 0 0 0 1 444\n",
      "57\t408\t0.0\t5\t0 0 0 0 2 406\n",
      "58\t391\t0.0\t5\t0 0 0 0 0 391\n",
      "59\t344\t0.0\t5\t0 0 0 0 0 340 4\n",
      "60\t447\t0.0\t6\t0 0 0 0 0 262 185\n",
      "61\t429\t0.0\t6\t0 0 0 0 0 268 161\n",
      "62\t389\t0.0\t6\t0 0 0 0 0 209 180\n",
      "63\t327\t0.0\t6\t0 0 0 0 0 204 123\n",
      "64\t337\t0.0\t6\t0 0 0 0 0 182 155\n",
      "65\t284\t0.0\t6\t0 0 0 0 0 156 128\n",
      "66\t247\t0.0\t6\t0 0 0 0 0 128 119\n",
      "67\t235\t0.0\t6\t0 0 0 0 0 126 109\n",
      "68\t225\t0.0\t6\t0 0 0 0 0 120 105\n",
      "69\t178\t0.0\t6\t0 0 0 0 0 102 76\n",
      "70\t167\t0.0\t6\t0 0 0 0 0 85 82\n",
      "71\t178\t0.0\t6\t0 0 0 0 0 100 78\n",
      "72\t138\t0.0\t6\t0 0 0 0 0 73 65\n",
      "73\t156\t0.0\t6\t0 0 0 0 0 76 80\n",
      "74\t122\t0.0\t6\t0 0 0 0 1 52 69\n",
      "75\t117\t0.0\t6\t0 0 0 0 0 62 55\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "76\t88\t0.0\t6\t0 0 0 0 0 48 40\n",
      "77\t96\t0.0\t6\t0 0 0 0 0 49 47\n",
      "78\t89\t0.0\t6\t0 0 0 0 0 50 39\n",
      "79\t60\t0.0\t6\t0 0 0 0 0 31 29\n",
      "80\t68\t0.0\t6\t0 0 0 0 0 35 33\n",
      "81\t75\t0.0\t6\t0 0 0 0 0 41 34\n",
      "82\t61\t0.0\t6\t0 0 0 0 0 23 38\n",
      "83\t51\t0.0\t6\t0 0 0 0 0 33 18\n",
      "84\t43\t0.0\t6\t0 0 0 0 0 22 21\n",
      "85\t48\t0.0\t6\t0 0 0 0 0 23 25\n",
      "86\t46\t0.0\t6\t0 0 0 0 0 24 22\n",
      "87\t41\t0.0\t6\t0 0 0 0 0 26 15\n",
      "88\t32\t0.0\t6\t0 0 0 0 0 15 17\n",
      "89\t33\t0.0\t6\t0 0 0 0 0 15 18\n",
      "90\t34\t0.0\t6\t0 0 0 0 0 15 19\n",
      "91\t30\t0.0\t6\t0 0 0 0 0 15 15\n",
      "92\t26\t0.0\t6\t0 0 0 0 0 15 11\n",
      "93\t23\t0.0\t6\t0 0 0 0 0 10 13\n",
      "94\t15\t0.0\t6\t0 0 0 0 0 8 7\n",
      "95\t16\t0.0\t6\t0 0 0 0 0 10 6\n",
      "96\t18\t0.0\t6\t0 0 0 0 0 9 9\n",
      "97\t12\t0.0\t6\t0 0 0 0 0 5 7\n",
      "98\t8\t0.0\t6\t0 0 0 0 0 2 6\n",
      "99\t11\t0.0\t6\t0 0 0 0 0 6 5\n",
      "100\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "101\t7\t0.0\t6\t0 0 0 0 0 3 4\n",
      "102\t6\t0.0\t6\t0 0 0 0 0 5 1\n",
      "103\t9\t0.0\t6\t0 0 0 0 0 3 6\n",
      "104\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "105\t7\t0.0\t6\t0 0 0 0 0 5 2\n",
      "106\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "107\t6\t0.0\t6\t0 0 0 0 0 1 5\n",
      "108\t7\t0.0\t6\t0 0 0 0 0 2 5\n",
      "109\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "110\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "111\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "112\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "113\t5\t0.0\t6\t0 0 0 0 0 2 3\n",
      "114\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "115\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "116\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "117\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "118\t11\t0.0\t6\t0 0 0 0 0 5 6\n",
      "119\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "120\t4\t0.0\t6\t0 0 0 0 0 2 2\n",
      "121\t4\t0.0\t6\t0 0 0 0 0 3 1\n",
      "122\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "123\t5\t0.0\t6\t0 0 0 0 0 2 3\n",
      "124\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "125\t10\t0.0\t6\t0 0 0 0 0 7 3\n",
      "126\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "127\t10\t0.0\t6\t0 0 0 0 0 6 4\n",
      "128\t10\t0.0\t6\t0 0 0 0 0 2 8\n",
      "129\t9\t0.0\t6\t0 0 0 0 0 4 5\n",
      "130\t10\t0.0\t6\t0 0 0 0 0 4 6\n",
      "131\t7\t0.0\t6\t0 0 0 0 0 4 3\n",
      "132\t6\t0.0\t6\t0 0 0 0 0 1 5\n",
      "133\t11\t0.0\t6\t0 0 0 0 0 7 4\n",
      "134\t10\t0.0\t6\t0 0 0 0 0 6 4\n",
      "135\t22\t0.0\t6\t0 0 0 0 0 14 8\n",
      "136\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "137\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "138\t13\t0.0\t6\t0 0 0 0 0 7 6\n",
      "139\t6\t0.0\t6\t0 0 0 0 0 4 2\n",
      "140\t12\t0.0\t6\t0 0 0 0 0 3 9\n",
      "141\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "142\t12\t0.0\t6\t0 0 0 0 0 3 9\n",
      "143\t14\t0.0\t6\t0 0 0 0 0 7 7\n",
      "144\t20\t0.0\t6\t0 0 0 0 0 11 9\n",
      "145\t16\t0.0\t6\t0 0 0 0 0 8 8\n",
      "146\t11\t0.0\t6\t0 0 0 0 0 6 5\n",
      "147\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "149\t9\t0.0\t6\t0 0 0 0 0 2 7\n",
      "150\t232\t0.0\t6\t0 0 0 0 0 94 138\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 725407 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.5%\n",
      "  C: 29.1%\n",
      "  G: 28.5%\n",
      "  T: 9.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t340004\t365706.6\t0\t340004\n",
      "4\t97353\t91426.6\t0\t97353\n",
      "5\t34757\t22856.7\t0\t34757\n",
      "6\t21404\t5714.2\t0\t21404\n",
      "7\t13662\t1428.5\t0\t13662\n",
      "8\t12687\t357.1\t0\t12687\n",
      "9\t12395\t89.3\t0\t11801 594\n",
      "10\t12217\t22.3\t1\t11456 761\n",
      "11\t11364\t5.6\t1\t10879 485\n",
      "12\t10796\t1.4\t1\t10359 437\n",
      "13\t10350\t0.3\t1\t9940 410\n",
      "14\t10031\t0.1\t1\t9692 339\n",
      "15\t9441\t0.0\t1\t9067 374\n",
      "16\t8650\t0.0\t1\t8304 346\n",
      "17\t8315\t0.0\t1\t7942 373\n",
      "18\t8208\t0.0\t1\t7727 481\n",
      "19\t7788\t0.0\t1\t7300 478 10\n",
      "20\t7748\t0.0\t2\t7118 564 66\n",
      "21\t7422\t0.0\t2\t6764 559 99\n",
      "22\t6940\t0.0\t2\t6327 529 84\n",
      "23\t6494\t0.0\t2\t5979 446 69\n",
      "24\t5889\t0.0\t2\t5337 479 73\n",
      "25\t5596\t0.0\t2\t5111 426 59\n",
      "26\t5477\t0.0\t2\t4760 610 107\n",
      "27\t5062\t0.0\t2\t4464 492 106\n",
      "28\t4981\t0.0\t2\t4353 531 96 1\n",
      "29\t4664\t0.0\t2\t4077 498 86 3\n",
      "30\t4452\t0.0\t3\t3877 459 90 26\n",
      "31\t4159\t0.0\t3\t3645 419 65 30\n",
      "32\t4219\t0.0\t3\t3651 447 83 38\n",
      "33\t3981\t0.0\t3\t3453 364 134 30\n",
      "34\t3737\t0.0\t3\t2 3285 368 82\n",
      "35\t3574\t0.0\t3\t3 3101 395 75\n",
      "36\t3282\t0.0\t3\t0 2815 339 128\n",
      "37\t3039\t0.0\t3\t0 5 2687 347\n",
      "38\t2732\t0.0\t3\t0 0 2423 309\n",
      "39\t2414\t0.0\t3\t0 0 2 2390 22\n",
      "40\t45\t0.0\t4\t0 0 1 5 39\n",
      "41\t35\t0.0\t4\t0 0 0 1 34\n",
      "42\t25\t0.0\t4\t0 0 0 1 24\n",
      "43\t17\t0.0\t4\t0 0 0 0 17\n",
      "44\t1\t0.0\t4\t0 0 0 0 1\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_1.fastq ./20211105180528381-5936/original/DEFAULT/a8_rep2_+T4_t5_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 38.63 s (1 µs/read; 40.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         25,862,607\n",
      "  Read 1 with adapter:                 926,016 (3.6%)\n",
      "  Read 2 with adapter:                 746,388 (2.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 478 (0.0%)\n",
      "Pairs written (passing filters):    25,862,129 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,758,782,100 bp\n",
      "  Read 1: 3,879,391,050 bp\n",
      "  Read 2: 3,879,391,050 bp\n",
      "Total written (filtered):  7,746,649,447 bp (99.8%)\n",
      "  Read 1: 3,873,059,048 bp\n",
      "  Read 2: 3,873,590,399 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 926016 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 37.7%\n",
      "  C: 21.2%\n",
      "  G: 14.9%\n",
      "  T: 26.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t507104\t404103.2\t0\t507104\n",
      "4\t165262\t101025.8\t0\t165262\n",
      "5\t49679\t25256.5\t0\t49679\n",
      "6\t19063\t6314.1\t0\t19063\n",
      "7\t11112\t1578.5\t0\t11112\n",
      "8\t9404\t394.6\t0\t9404\n",
      "9\t9694\t98.7\t0\t8940 754\n",
      "10\t9459\t24.7\t1\t8537 922\n",
      "11\t8552\t6.2\t1\t8072 480\n",
      "12\t8262\t1.5\t1\t7904 358\n",
      "13\t7636\t0.4\t1\t7377 259\n",
      "14\t7360\t0.1\t1\t7096 264\n",
      "15\t7178\t0.0\t1\t6922 256\n",
      "16\t6555\t0.0\t1\t6313 242\n",
      "17\t6277\t0.0\t1\t6062 215\n",
      "18\t6145\t0.0\t1\t5915 230\n",
      "19\t6039\t0.0\t1\t5804 230 5\n",
      "20\t5805\t0.0\t2\t5524 240 41\n",
      "21\t5602\t0.0\t2\t5297 254 51\n",
      "22\t5205\t0.0\t2\t4932 231 42\n",
      "23\t5018\t0.0\t2\t4750 223 45\n",
      "24\t4671\t0.0\t2\t4428 195 48\n",
      "25\t4567\t0.0\t2\t4312 208 47\n",
      "26\t4262\t0.0\t2\t4016 194 52\n",
      "27\t4227\t0.0\t2\t3961 225 41\n",
      "28\t3930\t0.0\t2\t3705 173 51 1\n",
      "29\t3840\t0.0\t2\t3627 168 43 2\n",
      "30\t3625\t0.0\t3\t3374 191 44 16\n",
      "31\t3526\t0.0\t3\t3252 202 51 21\n",
      "32\t3411\t0.0\t3\t3145 202 42 22\n",
      "33\t3099\t0.0\t3\t2859 140 76 24\n",
      "34\t2951\t0.0\t3\t0 2747 137 67\n",
      "35\t2787\t0.0\t3\t0 3 2673 111\n",
      "36\t2583\t0.0\t3\t0 0 5 2578\n",
      "37\t2502\t0.0\t3\t0 0 4 2498\n",
      "38\t2478\t0.0\t3\t0 0 6 2471 1\n",
      "39\t2227\t0.0\t3\t0 0 7 2220\n",
      "40\t9\t0.0\t4\t0 0 0 9\n",
      "41\t5\t0.0\t4\t0 0 0 0 5\n",
      "50\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "53\t3\t0.0\t5\t0 0 0 0 0 3\n",
      "55\t5\t0.0\t5\t0 0 0 0 0 5\n",
      "57\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "58\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "60\t376\t0.0\t6\t0 0 0 0 0 0 376\n",
      "61\t362\t0.0\t6\t0 0 0 0 0 0 362\n",
      "62\t312\t0.0\t6\t0 0 0 0 0 0 312\n",
      "63\t316\t0.0\t6\t0 0 0 0 0 0 316\n",
      "64\t262\t0.0\t6\t0 0 0 0 0 0 262\n",
      "65\t222\t0.0\t6\t0 0 0 0 0 0 222\n",
      "66\t210\t0.0\t6\t0 0 0 0 0 0 210\n",
      "67\t174\t0.0\t6\t0 0 0 0 0 0 174\n",
      "68\t178\t0.0\t6\t0 0 0 0 0 0 178\n",
      "69\t168\t0.0\t6\t0 0 0 0 0 0 168\n",
      "70\t156\t0.0\t6\t0 0 0 0 0 1 155\n",
      "71\t136\t0.0\t6\t0 0 0 0 0 0 136\n",
      "72\t149\t0.0\t6\t0 0 0 0 0 0 149\n",
      "73\t114\t0.0\t6\t0 0 0 0 0 0 114\n",
      "74\t124\t0.0\t6\t0 0 0 0 0 0 124\n",
      "75\t117\t0.0\t6\t0 0 0 0 0 0 117\n",
      "76\t108\t0.0\t6\t0 0 0 0 0 0 108\n",
      "77\t93\t0.0\t6\t0 0 0 0 0 0 93\n",
      "78\t86\t0.0\t6\t0 0 0 0 0 0 86\n",
      "79\t58\t0.0\t6\t0 0 0 0 0 0 58\n",
      "80\t66\t0.0\t6\t0 0 0 0 0 0 66\n",
      "81\t71\t0.0\t6\t0 0 0 0 0 0 71\n",
      "82\t66\t0.0\t6\t0 0 0 0 0 0 66\n",
      "83\t47\t0.0\t6\t0 0 0 0 0 0 47\n",
      "84\t38\t0.0\t6\t0 0 0 0 0 0 38\n",
      "85\t44\t0.0\t6\t0 0 0 0 0 0 44\n",
      "86\t47\t0.0\t6\t0 0 0 0 0 0 47\n",
      "87\t49\t0.0\t6\t0 0 0 0 0 0 49\n",
      "88\t42\t0.0\t6\t0 0 0 0 0 0 42\n",
      "89\t32\t0.0\t6\t0 0 0 0 0 0 32\n",
      "90\t28\t0.0\t6\t0 0 0 0 0 0 28\n",
      "91\t21\t0.0\t6\t0 0 0 0 0 0 21\n",
      "92\t23\t0.0\t6\t0 0 0 0 0 0 23\n",
      "93\t24\t0.0\t6\t0 0 0 0 0 0 24\n",
      "94\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "95\t22\t0.0\t6\t0 0 0 0 0 0 22\n",
      "96\t19\t0.0\t6\t0 0 0 0 0 0 19\n",
      "97\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "98\t15\t0.0\t6\t0 0 0 0 0 0 15\n",
      "99\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "100\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "101\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "102\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "103\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "104\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "105\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "106\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "107\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "108\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "109\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "110\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "111\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "112\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "113\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "114\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "115\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "116\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "117\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "118\t8\t0.0\t6\t0 0 0 0 0 0 8\n",
      "119\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "120\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "121\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "122\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "123\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "124\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "125\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "126\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "127\t6\t0.0\t6\t0 0 0 0 0 0 6\n",
      "128\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "129\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "130\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "131\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "132\t1\t0.0\t6\t0 0 0 0 0 0 1\n",
      "133\t7\t0.0\t6\t0 0 0 0 0 0 7\n",
      "134\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "135\t26\t0.0\t6\t0 0 0 0 0 0 26\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "136\t2\t0.0\t6\t0 0 0 0 0 0 2\n",
      "137\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "138\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "139\t9\t0.0\t6\t0 0 0 0 0 0 9\n",
      "140\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "141\t5\t0.0\t6\t0 0 0 0 0 0 5\n",
      "142\t11\t0.0\t6\t0 0 0 0 0 0 11\n",
      "143\t10\t0.0\t6\t0 0 0 0 0 0 10\n",
      "144\t16\t0.0\t6\t0 0 0 0 0 0 16\n",
      "145\t4\t0.0\t6\t0 0 0 0 0 0 4\n",
      "146\t12\t0.0\t6\t0 0 0 0 0 0 12\n",
      "147\t3\t0.0\t6\t0 0 0 0 0 0 3\n",
      "149\t13\t0.0\t6\t0 0 0 0 0 0 13\n",
      "150\t201\t0.0\t6\t0 0 0 0 0 1 200\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 746388 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 35.6%\n",
      "  C: 29.3%\n",
      "  G: 25.1%\n",
      "  T: 10.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t389848\t404103.2\t0\t389848\n",
      "4\t106748\t101025.8\t0\t106748\n",
      "5\t35796\t25256.5\t0\t35796\n",
      "6\t19348\t6314.1\t0\t19348\n",
      "7\t10990\t1578.5\t0\t10990\n",
      "8\t9571\t394.6\t0\t9571\n",
      "9\t10202\t98.7\t0\t9371 831\n",
      "10\t9847\t24.7\t1\t8928 919\n",
      "11\t9052\t6.2\t1\t8527 525\n",
      "12\t8469\t1.5\t1\t8137 332\n",
      "13\t8267\t0.4\t1\t7905 362\n",
      "14\t7683\t0.1\t1\t7403 280\n",
      "15\t7393\t0.0\t1\t7103 290\n",
      "16\t7202\t0.0\t1\t6911 291\n",
      "17\t6563\t0.0\t1\t6296 267\n",
      "18\t6290\t0.0\t1\t5896 393 1\n",
      "19\t6173\t0.0\t1\t5820 348 5\n",
      "20\t6123\t0.0\t2\t5586 459 78\n",
      "21\t5842\t0.0\t2\t5331 432 79\n",
      "22\t5638\t0.0\t2\t5120 445 73\n",
      "23\t5224\t0.0\t2\t4761 397 66\n",
      "24\t5050\t0.0\t2\t4551 429 70\n",
      "25\t4704\t0.0\t2\t4288 363 53\n",
      "26\t4604\t0.0\t2\t4012 509 83\n",
      "27\t4284\t0.0\t2\t3775 446 62 1\n",
      "28\t4252\t0.0\t2\t3743 435 73 1\n",
      "29\t3961\t0.0\t2\t3453 433 71 4\n",
      "30\t3880\t0.0\t3\t3363 416 66 35\n",
      "31\t3646\t0.0\t3\t3147 403 69 27\n",
      "32\t3565\t0.0\t3\t3099 357 74 35\n",
      "33\t3434\t0.0\t3\t2976 360 72 26\n",
      "34\t3136\t0.0\t3\t2687 296 111 42\n",
      "35\t2958\t0.0\t3\t1 2558 323 76\n",
      "36\t2867\t0.0\t3\t0 2436 323 108\n",
      "37\t2725\t0.0\t3\t0 7 2400 318\n",
      "38\t2591\t0.0\t3\t0 5 2324 262\n",
      "39\t2364\t0.0\t3\t0 0 5 2353 6\n",
      "40\t2109\t0.0\t4\t0 0 0 11 2098\n",
      "41\t2013\t0.0\t4\t0 0 1 16 1996\n",
      "42\t1878\t0.0\t4\t0 0 0 19 1859\n",
      "43\t67\t0.0\t4\t0 0 0 2 65\n",
      "44\t23\t0.0\t4\t0 0 0 0 23\n",
      "45\t4\t0.0\t4\t0 0 0 1 3\n",
      "46\t4\t0.0\t4\t0 0 0 0 4\n",
      "This is cutadapt 4.0 with Python 3.9.7\n",
      "Command line parameters: -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -o ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1_filter.fastq -p ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2_filter.fastq ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_1.fastq ./20211105180528381-5936/original/DEFAULT/a9_rep2_+T4_t10_2.fastq --cores=0 -m 50\n",
      "Processing paired-end reads on 48 cores ...\n",
      "Finished in 40.31 s (2 µs/read; 39.10 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:         26,272,200\n",
      "  Read 1 with adapter:               1,192,728 (4.5%)\n",
      "  Read 2 with adapter:               1,031,170 (3.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                 881 (0.0%)\n",
      "Pairs written (passing filters):    26,271,319 (100.0%)\n",
      "\n",
      "Total basepairs processed: 7,881,660,000 bp\n",
      "  Read 1: 3,940,830,000 bp\n",
      "  Read 2: 3,940,830,000 bp\n",
      "Total written (filtered):  7,860,173,080 bp (99.7%)\n",
      "  Read 1: 3,929,393,492 bp\n",
      "  Read 2: 3,930,779,588 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG; Type: regular 3'; Length: 65; Trimmed: 1192728 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-65 bp: 6\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 52.9%\n",
      "  C: 16.5%\n",
      "  G: 11.2%\n",
      "  T: 19.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t520606\t410503.1\t0\t520606\n",
      "4\t171370\t102625.8\t0\t171370\n",
      "5\t61464\t25656.4\t0\t61464\n",
      "6\t31850\t6414.1\t0\t31850\n",
      "7\t23744\t1603.5\t0\t23744\n",
      "8\t20752\t400.9\t0\t20752\n",
      "9\t20834\t100.2\t0\t19984 850\n",
      "10\t19859\t25.1\t1\t18571 1288\n",
      "11\t19205\t6.3\t1\t18319 886\n",
      "12\t17943\t1.6\t1\t17210 733\n",
      "13\t17061\t0.4\t1\t16466 595\n",
      "14\t16057\t0.1\t1\t15515 542\n",
      "15\t15420\t0.0\t1\t14872 548\n",
      "16\t14816\t0.0\t1\t14278 538\n",
      "17\t14036\t0.0\t1\t13473 563\n",
      "18\t13143\t0.0\t1\t12580 563\n",
      "19\t13280\t0.0\t1\t12711 565 4\n",
      "20\t12634\t0.0\t2\t11955 574 105\n",
      "21\t11707\t0.0\t2\t11068 528 111\n",
      "22\t11210\t0.0\t2\t10616 499 95\n",
      "23\t10455\t0.0\t2\t9912 445 98\n",
      "24\t9581\t0.0\t2\t9034 450 97\n",
      "25\t9371\t0.0\t2\t8821 438 112\n",
      "26\t8907\t0.0\t2\t8413 404 90\n",
      "27\t8363\t0.0\t2\t7813 455 95\n",
      "28\t8062\t0.0\t2\t7564 412 86\n",
      "29\t7694\t0.0\t2\t7221 389 80 4\n",
      "30\t7460\t0.0\t3\t6899 420 95 46\n",
      "31\t6972\t0.0\t3\t6454 406 73 39\n",
      "32\t6596\t0.0\t3\t6119 362 71 44\n",
      "33\t6264\t0.0\t3\t5840 247 138 39\n",
      "34\t5852\t0.0\t3\t0 5471 315 66\n",
      "35\t5542\t0.0\t3\t0 5196 232 114\n",
      "36\t5059\t0.0\t3\t0 8 4853 198\n",
      "37\t4655\t0.0\t3\t0 0 16 4636 3\n",
      "38\t4374\t0.0\t3\t0 0 2 4372\n",
      "39\t4057\t0.0\t3\t0 0 4 4037 16\n",
      "40\t3792\t0.0\t4\t0 0 2 3789 1\n",
      "41\t3\t0.0\t4\t0 0 0 2 1\n",
      "42\t28\t0.0\t4\t0 0 0 1 27\n",
      "43\t27\t0.0\t4\t0 0 0 0 27\n",
      "44\t4\t0.0\t4\t0 0 0 0 4\n",
      "46\t1\t0.0\t4\t0 0 0 0 1\n",
      "48\t2\t0.0\t4\t0 0 0 0 2\n",
      "50\t1630\t0.0\t5\t0 0 0 0 0 1630\n",
      "51\t1494\t0.0\t5\t0 0 0 0 0 1494\n",
      "52\t1398\t0.0\t5\t0 0 0 0 0 1398\n",
      "53\t1286\t0.0\t5\t0 0 0 0 0 1286\n",
      "54\t1072\t0.0\t5\t0 0 0 0 0 1072\n",
      "55\t1049\t0.0\t5\t0 0 0 0 0 1049\n",
      "56\t913\t0.0\t5\t0 0 0 0 0 913\n",
      "57\t851\t0.0\t5\t0 0 0 0 0 851\n",
      "58\t768\t0.0\t5\t0 0 0 0 1 767\n",
      "59\t790\t0.0\t5\t0 0 0 0 0 780 10\n",
      "60\t920\t0.0\t6\t0 0 0 0 0 655 265\n",
      "61\t932\t0.0\t6\t0 0 0 0 0 688 244\n",
      "62\t826\t0.0\t6\t0 0 0 0 0 551 275\n",
      "63\t710\t0.0\t6\t0 0 0 0 0 504 206\n",
      "64\t667\t0.0\t6\t0 0 0 0 0 436 231\n",
      "65\t615\t0.0\t6\t0 0 0 0 0 409 206\n",
      "66\t496\t0.0\t6\t0 0 0 0 0 334 162\n",
      "67\t468\t0.0\t6\t0 0 0 0 0 304 164\n",
      "68\t417\t0.0\t6\t0 0 0 0 0 295 122\n",
      "69\t374\t0.0\t6\t0 0 0 0 0 263 111\n",
      "70\t341\t0.0\t6\t0 0 0 0 0 229 112\n",
      "71\t379\t0.0\t6\t0 0 0 0 0 259 120\n",
      "72\t291\t0.0\t6\t0 0 0 0 0 191 100\n",
      "73\t322\t0.0\t6\t0 0 0 0 0 224 98\n",
      "74\t292\t0.0\t6\t0 0 0 0 0 199 93\n",
      "75\t237\t0.0\t6\t0 0 0 0 0 159 78\n",
      "76\t216\t0.0\t6\t0 0 0 0 0 141 75\n",
      "77\t171\t0.0\t6\t0 0 0 0 0 118 53\n",
      "78\t198\t0.0\t6\t0 0 0 0 0 136 62\n",
      "79\t160\t0.0\t6\t0 0 0 0 0 94 66\n",
      "80\t154\t0.0\t6\t0 0 0 0 0 99 55\n",
      "81\t155\t0.0\t6\t0 0 0 0 0 93 62\n",
      "82\t144\t0.0\t6\t0 0 0 0 0 85 59\n",
      "83\t100\t0.0\t6\t0 0 0 0 0 69 31\n",
      "84\t108\t0.0\t6\t0 0 0 0 0 72 36\n",
      "85\t122\t0.0\t6\t0 0 0 0 0 82 40\n",
      "86\t84\t0.0\t6\t0 0 0 0 0 49 35\n",
      "87\t86\t0.0\t6\t0 0 0 0 0 57 29\n",
      "88\t91\t0.0\t6\t0 0 0 0 0 67 24\n",
      "89\t56\t0.0\t6\t0 0 0 0 0 41 15\n",
      "90\t54\t0.0\t6\t0 0 0 0 0 31 23\n",
      "91\t38\t0.0\t6\t0 0 0 0 0 24 14\n",
      "92\t47\t0.0\t6\t0 0 0 0 0 29 18\n",
      "93\t39\t0.0\t6\t0 0 0 0 0 22 17\n",
      "94\t37\t0.0\t6\t0 0 0 0 0 19 18\n",
      "95\t28\t0.0\t6\t0 0 0 0 0 23 5\n",
      "96\t25\t0.0\t6\t0 0 0 0 0 19 6\n",
      "97\t23\t0.0\t6\t0 0 0 0 0 15 8\n",
      "98\t21\t0.0\t6\t0 0 0 0 0 13 8\n",
      "99\t23\t0.0\t6\t0 0 0 0 0 11 12\n",
      "100\t17\t0.0\t6\t0 0 0 0 0 8 9\n",
      "101\t13\t0.0\t6\t0 0 0 0 0 6 7\n",
      "102\t7\t0.0\t6\t0 0 0 0 0 4 3\n",
      "103\t7\t0.0\t6\t0 0 0 0 0 5 2\n",
      "104\t9\t0.0\t6\t0 0 0 0 0 8 1\n",
      "105\t8\t0.0\t6\t0 0 0 0 0 2 6\n",
      "106\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "107\t6\t0.0\t6\t0 0 0 0 0 4 2\n",
      "108\t5\t0.0\t6\t0 0 0 0 0 4 1\n",
      "109\t8\t0.0\t6\t0 0 0 0 0 8\n",
      "110\t4\t0.0\t6\t0 0 0 0 0 4\n",
      "111\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "112\t6\t0.0\t6\t0 0 0 0 0 3 3\n",
      "113\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "114\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "115\t3\t0.0\t6\t0 0 0 0 0 3\n",
      "116\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "117\t4\t0.0\t6\t0 0 0 0 0 1 3\n",
      "118\t17\t0.0\t6\t0 0 0 0 0 8 9\n",
      "119\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "120\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "121\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "122\t8\t0.0\t6\t0 0 0 0 0 2 6\n",
      "124\t8\t0.0\t6\t0 0 0 0 0 4 4\n",
      "125\t5\t0.0\t6\t0 0 0 0 0 3 2\n",
      "126\t9\t0.0\t6\t0 0 0 0 0 5 4\n",
      "127\t17\t0.0\t6\t0 0 0 0 0 9 8\n",
      "128\t19\t0.0\t6\t0 0 0 0 0 9 10\n",
      "129\t9\t0.0\t6\t0 0 0 0 0 3 6\n",
      "130\t3\t0.0\t6\t0 0 0 0 0 2 1\n",
      "131\t8\t0.0\t6\t0 0 0 0 0 4 4\n",
      "132\t8\t0.0\t6\t0 0 0 0 0 3 5\n",
      "133\t8\t0.0\t6\t0 0 0 0 0 5 3\n",
      "134\t16\t0.0\t6\t0 0 0 0 0 11 5\n",
      "135\t45\t0.0\t6\t0 0 0 0 0 22 23\n",
      "136\t7\t0.0\t6\t0 0 0 0 0 3 4\n",
      "137\t6\t0.0\t6\t0 0 0 0 0 2 4\n",
      "138\t18\t0.0\t6\t0 0 0 0 0 6 12\n",
      "139\t22\t0.0\t6\t0 0 0 0 0 10 12\n",
      "140\t20\t0.0\t6\t0 0 0 0 0 10 10\n",
      "141\t15\t0.0\t6\t0 0 0 0 0 11 4\n",
      "142\t25\t0.0\t6\t0 0 0 0 0 12 13\n",
      "143\t12\t0.0\t6\t0 0 0 0 0 10 2\n",
      "144\t40\t0.0\t6\t0 0 0 0 0 22 18\n",
      "145\t17\t0.0\t6\t0 0 0 0 0 9 8\n",
      "146\t20\t0.0\t6\t0 0 0 0 0 10 10\n",
      "147\t3\t0.0\t6\t0 0 0 0 0 1 2\n",
      "148\t2\t0.0\t6\t0 0 0 0 0 1 1\n",
      "149\t30\t0.0\t6\t0 0 0 0 0 12 18\n",
      "150\t354\t0.0\t6\t0 0 0 0 0 187 167\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT; Type: regular 3'; Length: 58; Trimmed: 1031170 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-58 bp: 5\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.2%\n",
      "  C: 30.1%\n",
      "  G: 28.7%\n",
      "  T: 11.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t413261\t410503.1\t0\t413261\n",
      "4\t125229\t102625.8\t0\t125229\n",
      "5\t51411\t25656.4\t0\t51411\n",
      "6\t32929\t6414.1\t0\t32929\n",
      "7\t24748\t1603.5\t0\t24748\n",
      "8\t22234\t400.9\t0\t22234\n",
      "9\t21717\t100.2\t0\t20868 849\n",
      "10\t21471\t25.1\t1\t20131 1340\n",
      "11\t19642\t6.3\t1\t18825 817\n",
      "12\t19171\t1.6\t1\t18558 613\n",
      "13\t17963\t0.4\t1\t17360 603\n",
      "14\t17116\t0.1\t1\t16608 508\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15\t16077\t0.0\t1\t15508 569\n",
      "16\t15447\t0.0\t1\t14891 556\n",
      "17\t14858\t0.0\t1\t14285 573\n",
      "18\t14072\t0.0\t1\t13348 721 3\n",
      "19\t13246\t0.0\t1\t12582 657 7\n",
      "20\t13441\t0.0\t2\t12666 684 91\n",
      "21\t12670\t0.0\t2\t11878 703 89\n",
      "22\t11769\t0.0\t2\t11069 616 84\n",
      "23\t11227\t0.0\t2\t10576 550 101\n",
      "24\t10557\t0.0\t2\t9875 588 94\n",
      "25\t9616\t0.0\t2\t8970 568 78\n",
      "26\t9462\t0.0\t2\t8501 853 108\n",
      "27\t8956\t0.0\t2\t8031 814 111\n",
      "28\t8409\t0.0\t2\t7554 745 110\n",
      "29\t8117\t0.0\t2\t7218 766 128 5\n",
      "30\t7786\t0.0\t3\t6936 696 104 50\n",
      "31\t7511\t0.0\t3\t6679 669 118 45\n",
      "32\t7009\t0.0\t3\t6173 686 114 36\n",
      "33\t6627\t0.0\t3\t5862 518 195 52\n",
      "34\t6229\t0.0\t3\t2 5509 552 166\n",
      "35\t5830\t0.0\t3\t0 19 5146 665\n",
      "36\t5483\t0.0\t3\t0 1 4929 553\n",
      "37\t5071\t0.0\t3\t0 1 4533 458 79\n",
      "38\t4880\t0.0\t3\t0 0 3 4372 505\n",
      "39\t4573\t0.0\t3\t0 0 3 4181 389\n",
      "40\t3973\t0.0\t4\t0 0 0 306 3667\n",
      "41\t338\t0.0\t4\t0 0 0 3 335\n",
      "42\t306\t0.0\t4\t0 0 0 0 306\n",
      "43\t287\t0.0\t4\t0 0 0 1 286\n",
      "44\t275\t0.0\t4\t0 0 0 1 274\n",
      "45\t128\t0.0\t4\t0 0 0 0 128\n",
      "46\t48\t0.0\t4\t0 0 0 0 48\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "adaptor1=GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGATGACTATCTCGTATGCCGTCTTCTGCTTG\n",
    "adaptor2=AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/*.fastq | grep -Eo 'a[0-9]+_.*t.*[0-9]+_'  | sort | uniq )\n",
    "path_fastq=\"./20211105180528381-5936/original/DEFAULT/\"\n",
    "for ID in $fastq_name\n",
    "    do \n",
    "    R1_file=$path_fastq$ID\"1.fastq\"\n",
    "    filter_R1=$path_fastq$ID\"1_filter.fastq\"\n",
    "    R2_file=$path_fastq$ID\"2.fastq\"\n",
    "    filter_R2=$path_fastq$ID\"2_filter.fastq\"\n",
    "    cutadapt -a $adaptor1 -A $adaptor2 -o $filter_R1 -p $filter_R2 $R1_file $R2_file --cores=0 -m 50\n",
    "    done\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP3: run hisat2 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### build the index of reference genome using the .fa format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"./yan_results/genome_index/genome.*.ht2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Local offset rate: 3 (one in 8)\n",
      "  Local fTable chars: 6\n",
      "  Local sequence length: 57344\n",
      "  Local sequence overlap between two consecutive indexes: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  ./yan_results/Escherichia_coli_K12_and_T1_phage_genome.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to read SNPs and splice sites: 00:00:00\n",
      "Using parameters --bmax 109253 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 109253 --dcv 1024\n",
      "Constructing suffix-array element generator\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 86 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 86 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 86 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 9, merged 39; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 7, merged 6; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 4, merged 3; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 2; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 81779.1 (target: 109252)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering GFM loop\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting block 1 of 57\n",
      "  Reserving size (109253) for bucket 1\n",
      "Getting block 2 of 57\n",
      "  Reserving size (109253) for bucket 2\n",
      "  Calculating Z arrays for bucket 1\n",
      "Getting block 4 of 57\n",
      "  Reserving size (109253) for bucket 4\n",
      "Getting block 3 of 57\n",
      "  Reserving size (109253) for bucket 3\n",
      "  Calculating Z arrays for bucket 4\n",
      "Getting block 5 of 57\n",
      "  Reserving size (109253) for bucket 5\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  Calculating Z arrays for bucket 3\n",
      "Getting block 6 of 57\n",
      "  Reserving size (109253) for bucket 6\n",
      "Getting block 7 of 57\n",
      "  Reserving size (109253) for bucket 7\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Calculating Z arrays for bucket 6\n",
      "Getting block 8 of 57\n",
      "  Reserving size (109253) for bucket 8\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 8: 10%\n",
      "  bucket 7: 10%\n",
      "  bucket 4: 10%\n",
      "  bucket 1: 10%\n",
      "  bucket 3: 10%\n",
      "  bucket 2: 10%\n",
      "  bucket 5: 10%\n",
      "  bucket 8: 20%\n",
      "  bucket 6: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 4: 20%\n",
      "  bucket 1: 20%\n",
      "  bucket 3: 20%\n",
      "  bucket 2: 20%\n",
      "  bucket 5: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 7: 30%\n",
      "  bucket 6: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 4: 30%\n",
      "  bucket 3: 30%\n",
      "  bucket 2: 30%\n",
      "  bucket 8: 40%\n",
      "  bucket 7: 40%\n",
      "  bucket 5: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 6: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 4: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 2: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 1: 50%\n",
      "  bucket 3: 50%\n",
      "  bucket 5: 40%\n",
      "  bucket 6: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 8: 60%\n",
      "  bucket 2: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 7: 60%\n",
      "  bucket 3: 60%\n",
      "  bucket 5: 50%\n",
      "  bucket 6: 50%\n",
      "  bucket 8: 70%\n",
      "  bucket 1: 70%\n",
      "  bucket 3: 70%\n",
      "  bucket 4: 60%\n",
      "  bucket 2: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 8: 80%\n",
      "  bucket 5: 60%\n",
      "  bucket 3: 80%\n",
      "  bucket 6: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 4: 70%\n",
      "  bucket 1: 90%\n",
      "  bucket 3: 90%\n",
      "  bucket 8: 90%\n",
      "  bucket 5: 70%\n",
      "  bucket 7: 90%\n",
      "  bucket 2: 80%\n",
      "  bucket 6: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 76897 for bucket 1\n",
      "  (Using difference cover)\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 36048 for bucket 3\n",
      "  (Using difference cover)\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 60925 for bucket 8\n",
      "  (Using difference cover)\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 80480 for bucket 7\n",
      "  (Using difference cover)\n",
      "  bucket 2: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 36049 for bucket 3\n",
      "  bucket 5: 80%\n",
      "Getting block 9 of 57\n",
      "  Reserving size (109253) for bucket 9\n",
      "  Calculating Z arrays for bucket 9\n",
      "  Entering block accumulator loop for bucket 9:\n",
      "  bucket 4: 90%\n",
      "  bucket 6: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 76898 for bucket 1\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 60926 for bucket 8\n",
      "Getting block 10 of 57\n",
      "  Reserving size (109253) for bucket 10\n",
      "  Calculating Z arrays for bucket 10\n",
      "  Entering block accumulator loop for bucket 10:\n",
      "Getting block 11 of 57\n",
      "  Reserving size (109253) for bucket 11\n",
      "  Calculating Z arrays for bucket 11\n",
      "  Entering block accumulator loop for bucket 11:\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 83806 for bucket 2\n",
      "  (Using difference cover)\n",
      "  bucket 5: 90%\n",
      "  bucket 6: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 80481 for bucket 7\n",
      "  bucket 9: 10%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 92133 for bucket 4\n",
      "  (Using difference cover)\n",
      "Getting block 12 of 57\n",
      "  Reserving size (109253) for bucket 12\n",
      "  Calculating Z arrays for bucket 12\n",
      "  Entering block accumulator loop for bucket 12:\n",
      "  bucket 10: 10%\n",
      "  bucket 11: 10%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 74009 for bucket 5\n",
      "  (Using difference cover)\n",
      "  bucket 9: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 83807 for bucket 2\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 107052 for bucket 6\n",
      "  (Using difference cover)\n",
      "Getting block 13 of 57\n",
      "  Reserving size (109253) for bucket 13\n",
      "  Calculating Z arrays for bucket 13\n",
      "  Entering block accumulator loop for bucket 13:\n",
      "  bucket 12: 10%\n",
      "  bucket 10: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 92134 for bucket 4\n",
      "  bucket 11: 20%\n",
      "  bucket 9: 30%\n",
      "Getting block 14 of 57\n",
      "  Reserving size (109253) for bucket 14\n",
      "  Calculating Z arrays for bucket 14\n",
      "  Entering block accumulator loop for bucket 14:\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 74010 for bucket 5\n",
      "  bucket 12: 20%\n",
      "  bucket 10: 30%\n",
      "  bucket 13: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 107053 for bucket 6\n",
      "Getting block 15 of 57\n",
      "  Reserving size (109253) for bucket 15\n",
      "  Calculating Z arrays for bucket 15\n",
      "  Entering block accumulator loop for bucket 15:\n",
      "  bucket 9: 40%\n",
      "  bucket 11: 30%\n",
      "Getting block 16 of 57\n",
      "  Reserving size (109253) for bucket 16\n",
      "  Calculating Z arrays for bucket 16\n",
      "  Entering block accumulator loop for bucket 16:\n",
      "  bucket 12: 30%\n",
      "  bucket 10: 40%\n",
      "  bucket 14: 10%\n",
      "  bucket 13: 20%\n",
      "  bucket 9: 50%\n",
      "  bucket 11: 40%\n",
      "  bucket 15: 10%\n",
      "  bucket 10: 50%\n",
      "  bucket 16: 10%\n",
      "  bucket 12: 40%\n",
      "  bucket 14: 20%\n",
      "  bucket 13: 30%\n",
      "  bucket 9: 60%\n",
      "  bucket 10: 60%\n",
      "  bucket 11: 50%\n",
      "  bucket 12: 50%\n",
      "  bucket 16: 20%\n",
      "  bucket 15: 20%\n",
      "  bucket 14: 30%\n",
      "  bucket 9: 70%\n",
      "  bucket 13: 40%\n",
      "  bucket 10: 70%\n",
      "  bucket 12: 60%\n",
      "  bucket 11: 60%\n",
      "  bucket 16: 30%\n",
      "  bucket 9: 80%\n",
      "  bucket 10: 80%\n",
      "  bucket 14: 40%\n",
      "  bucket 13: 50%\n",
      "  bucket 15: 30%\n",
      "  bucket 12: 70%\n",
      "  bucket 11: 70%\n",
      "  bucket 9: 90%\n",
      "  bucket 16: 40%\n",
      "  bucket 10: 90%\n",
      "  bucket 12: 80%\n",
      "  bucket 14: 50%\n",
      "  bucket 13: 60%\n",
      "  bucket 15: 40%\n",
      "  bucket 10: 100%\n",
      "  Sorting block of length 67717 for bucket 10\n",
      "  (Using difference cover)\n",
      "  bucket 11: 80%\n",
      "  bucket 9: 100%\n",
      "  Sorting block of length 49845 for bucket 9\n",
      "  (Using difference cover)\n",
      "  bucket 16: 50%\n",
      "  bucket 12: 90%\n",
      "  bucket 14: 60%\n",
      "  bucket 13: 70%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 49846 for bucket 9\n",
      "Getting block 17 of 57\n",
      "  Reserving size (109253) for bucket 17\n",
      "  Calculating Z arrays for bucket 17\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 67718 for bucket 10\n",
      "  Entering block accumulator loop for bucket 17:\n",
      "  bucket 11: 90%\n",
      "  bucket 12: 100%\n",
      "  Sorting block of length 71143 for bucket 12\n",
      "  (Using difference cover)\n",
      "  bucket 16: 60%\n",
      "Getting block 18 of 57\n",
      "  Reserving size (109253) for bucket 18\n",
      "  Calculating Z arrays for bucket 18\n",
      "  Entering block accumulator loop for bucket 18:\n",
      "  bucket 15: 50%\n",
      "  bucket 14: 70%\n",
      "  bucket 13: 80%\n",
      "  bucket 17: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 71144 for bucket 12\n",
      "  bucket 16: 70%\n",
      "  bucket 18: 10%\n",
      "  bucket 11: 100%\n",
      "  Sorting block of length 100162 for bucket 11\n",
      "  (Using difference cover)\n",
      "Getting block 19 of 57\n",
      "  Reserving size (109253) for bucket 19\n",
      "  Calculating Z arrays for bucket 19\n",
      "  Entering block accumulator loop for bucket 19:\n",
      "  bucket 15: 60%\n",
      "  bucket 14: 80%\n",
      "  bucket 13: 90%\n",
      "  bucket 16: 80%\n",
      "  bucket 17: 20%\n",
      "  bucket 18: 20%\n",
      "  bucket 19: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 100163 for bucket 11\n",
      "  bucket 13: 100%\n",
      "  Sorting block of length 78133 for bucket 13\n",
      "  (Using difference cover)\n",
      "  bucket 14: 90%\n",
      "Getting block 20 of 57\n",
      "  Reserving size (109253) for bucket 20\n",
      "  Calculating Z arrays for bucket 20\n",
      "  Entering block accumulator loop for bucket 20:\n",
      "  bucket 15: 70%\n",
      "  bucket 16: 90%\n",
      "  bucket 18: 30%\n",
      "  bucket 17: 30%\n",
      "  bucket 19: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 78134 for bucket 13\n",
      "  bucket 14: 100%\n",
      "  Sorting block of length 91457 for bucket 14\n",
      "  (Using difference cover)\n",
      "  bucket 16: 100%\n",
      "  Sorting block of length 34248 for bucket 16\n",
      "  (Using difference cover)\n",
      "  bucket 20: 10%\n",
      "Getting block 21 of 57\n",
      "  Reserving size (109253) for bucket 21\n",
      "  Calculating Z arrays for bucket 21\n",
      "  Entering block accumulator loop for bucket 21:\n",
      "  bucket 18: 40%\n",
      "  bucket 17: 40%\n",
      "  bucket 15: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 34249 for bucket 16\n",
      "Getting block 22 of 57\n",
      "  Reserving size (109253) for bucket 22\n",
      "  Calculating Z arrays for bucket 22\n",
      "  Entering block accumulator loop for bucket 22:\n",
      "  bucket 19: 30%\n",
      "  bucket 20: 20%\n",
      "  bucket 21: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 91458 for bucket 14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  bucket 18: 50%\n",
      "  bucket 17: 50%\n",
      "Getting block 23 of 57\n",
      "  Reserving size (109253) for bucket 23\n",
      "  Calculating Z arrays for bucket 23\n",
      "  Entering block accumulator loop for bucket 23:\n",
      "  bucket 15: 90%\n",
      "  bucket 19: 40%\n",
      "  bucket 22: 10%\n",
      "  bucket 18: 60%\n",
      "  bucket 20: 30%\n",
      "  bucket 21: 20%\n",
      "  bucket 17: 60%\n",
      "  bucket 19: 50%\n",
      "  bucket 22: 20%\n",
      "  bucket 23: 10%\n",
      "  bucket 15: 100%\n",
      "  Sorting block of length 100722 for bucket 15\n",
      "  (Using difference cover)\n",
      "  bucket 18: 70%\n",
      "  bucket 20: 40%\n",
      "  bucket 21: 30%\n",
      "  bucket 17: 70%\n",
      "  bucket 19: 60%\n",
      "  bucket 22: 30%\n",
      "  bucket 23: 20%\n",
      "  bucket 18: 80%\n",
      "  bucket 20: 50%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 100723 for bucket 15\n",
      "  bucket 21: 40%\n",
      "  bucket 17: 80%\n",
      "  bucket 19: 70%\n",
      "  bucket 18: 90%\n",
      "  bucket 22: 40%\n",
      "Getting block 24 of 57\n",
      "  Reserving size (109253) for bucket 24\n",
      "  Calculating Z arrays for bucket 24\n",
      "  Entering block accumulator loop for bucket 24:\n",
      "  bucket 23: 30%\n",
      "  bucket 20: 60%\n",
      "  bucket 17: 90%\n",
      "  bucket 21: 50%\n",
      "  bucket 19: 80%\n",
      "  bucket 18: 100%\n",
      "  Sorting block of length 77458 for bucket 18\n",
      "  (Using difference cover)\n",
      "  bucket 22: 50%\n",
      "  bucket 24: 10%\n",
      "  bucket 20: 70%\n",
      "  bucket 23: 40%\n",
      "  bucket 17: 100%\n",
      "  Sorting block of length 77527 for bucket 17\n",
      "  (Using difference cover)\n",
      "  bucket 21: 60%\n",
      "  bucket 19: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 77459 for bucket 18\n",
      "  bucket 22: 60%\n",
      "Getting block 25 of 57\n",
      "  Reserving size (109253) for bucket 25\n",
      "  Calculating Z arrays for bucket 25\n",
      "  Entering block accumulator loop for bucket 25:\n",
      "  bucket 24: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 77528 for bucket 17\n",
      "  bucket 20: 80%\n",
      "  bucket 23: 50%\n",
      "Getting block 26 of 57\n",
      "  Reserving size (109253) for bucket 26\n",
      "  Calculating Z arrays for bucket 26\n",
      "  Entering block accumulator loop for bucket 26:\n",
      "  bucket 21: 70%\n",
      "  bucket 19: 100%\n",
      "  Sorting block of length 94176 for bucket 19\n",
      "  (Using difference cover)\n",
      "  bucket 22: 70%\n",
      "  bucket 25: 10%\n",
      "  bucket 24: 30%\n",
      "  bucket 20: 90%\n",
      "  bucket 23: 60%\n",
      "  bucket 21: 80%\n",
      "  bucket 26: 10%\n",
      "  bucket 22: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 94177 for bucket 19\n",
      "  bucket 25: 20%\n",
      "  bucket 24: 40%\n",
      "  bucket 20: 100%\n",
      "  Sorting block of length 65820 for bucket 20\n",
      "  (Using difference cover)\n",
      "Getting block 27 of 57\n",
      "  Reserving size (109253) for bucket 27\n",
      "  Calculating Z arrays for bucket 27\n",
      "  Entering block accumulator loop for bucket 27:\n",
      "  bucket 21: 90%\n",
      "  bucket 23: 70%\n",
      "  bucket 26: 20%\n",
      "  bucket 22: 90%\n",
      "  bucket 25: 30%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 65821 for bucket 20\n",
      "Getting block 28 of 57\n",
      "  Reserving size (109253) for bucket 28\n",
      "  Calculating Z arrays for bucket 28\n",
      "  Entering block accumulator loop for bucket 28:\n",
      "  bucket 27: 10%\n",
      "  bucket 24: 50%\n",
      "  bucket 21: 100%\n",
      "  Sorting block of length 109122 for bucket 21\n",
      "  (Using difference cover)\n",
      "  bucket 26: 30%\n",
      "  bucket 23: 80%\n",
      "  bucket 22: 100%\n",
      "  Sorting block of length 99233 for bucket 22\n",
      "  (Using difference cover)\n",
      "  bucket 25: 40%\n",
      "  bucket 28: 10%\n",
      "  bucket 27: 20%\n",
      "  bucket 24: 60%\n",
      "  bucket 26: 40%\n",
      "  bucket 25: 50%\n",
      "  bucket 23: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 99234 for bucket 22\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 109123 for bucket 21\n",
      "  bucket 28: 20%\n",
      "  bucket 27: 30%\n",
      "Getting block 29 of 57\n",
      "  Reserving size (109253) for bucket 29\n",
      "  Calculating Z arrays for bucket 29\n",
      "  Entering block accumulator loop for bucket 29:\n",
      "Getting block 30 of 57\n",
      "  Reserving size (109253) for bucket 30\n",
      "  Calculating Z arrays for bucket 30\n",
      "  Entering block accumulator loop for bucket 30:\n",
      "  bucket 24: 70%\n",
      "  bucket 25: 60%\n",
      "  bucket 26: 50%\n",
      "  bucket 23: 100%\n",
      "  Sorting block of length 99668 for bucket 23\n",
      "  (Using difference cover)\n",
      "  bucket 28: 30%\n",
      "  bucket 27: 40%\n",
      "  bucket 29: 10%\n",
      "  bucket 30: 10%\n",
      "  bucket 24: 80%\n",
      "  bucket 25: 70%\n",
      "  bucket 26: 60%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 99669 for bucket 23\n",
      "  bucket 28: 40%\n",
      "  bucket 27: 50%\n",
      "  bucket 29: 20%\n",
      "Getting block 31 of 57\n",
      "  Reserving size (109253) for bucket 31\n",
      "  Calculating Z arrays for bucket 31\n",
      "  Entering block accumulator loop for bucket 31:\n",
      "  bucket 30: 20%\n",
      "  bucket 24: 90%\n",
      "  bucket 25: 80%\n",
      "  bucket 26: 70%\n",
      "  bucket 28: 50%\n",
      "  bucket 27: 60%\n",
      "  bucket 29: 30%\n",
      "  bucket 31: 10%\n",
      "  bucket 24: 100%\n",
      "  Sorting block of length 55615 for bucket 24\n",
      "  (Using difference cover)\n",
      "  bucket 25: 90%\n",
      "  bucket 30: 30%\n",
      "  bucket 26: 80%\n",
      "  bucket 28: 60%\n",
      "  bucket 27: 70%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 55616 for bucket 24\n",
      "  bucket 25: 100%\n",
      "  Sorting block of length 105100 for bucket 25\n",
      "  (Using difference cover)\n",
      "  bucket 31: 20%\n",
      "  bucket 29: 40%\n",
      "Getting block 32 of 57\n",
      "  Reserving size (109253) for bucket 32\n",
      "  Calculating Z arrays for bucket 32\n",
      "  Entering block accumulator loop for bucket 32:\n",
      "  bucket 26: 90%\n",
      "  bucket 30: 40%\n",
      "  bucket 28: 70%\n",
      "  bucket 27: 80%\n",
      "  bucket 31: 30%\n",
      "  bucket 29: 50%\n",
      "  bucket 32: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 105101 for bucket 25\n",
      "  bucket 26: 100%\n",
      "  Sorting block of length 81772 for bucket 26\n",
      "  (Using difference cover)\n",
      "  bucket 30: 50%\n",
      "  bucket 28: 80%\n",
      "Getting block 33 of 57\n",
      "  Reserving size (109253) for bucket 33\n",
      "  Calculating Z arrays for bucket 33\n",
      "  Entering block accumulator loop for bucket 33:\n",
      "  bucket 27: 90%\n",
      "  bucket 31: 40%\n",
      "  bucket 29: 60%\n",
      "  bucket 32: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 81773 for bucket 26\n",
      "  bucket 28: 90%\n",
      "Getting block 34 of 57\n",
      "  Reserving size (109253) for bucket 34\n",
      "  Calculating Z arrays for bucket 34\n",
      "  Entering block accumulator loop for bucket 34:\n",
      "  bucket 33: 10%\n",
      "  bucket 27: 100%\n",
      "  Sorting block of length 94976 for bucket 27\n",
      "  (Using difference cover)\n",
      "  bucket 30: 60%\n",
      "  bucket 31: 50%\n",
      "  bucket 29: 70%\n",
      "  bucket 32: 30%\n",
      "  bucket 33: 20%\n",
      "  bucket 28: 100%\n",
      "  Sorting block of length 55113 for bucket 28\n",
      "  (Using difference cover)\n",
      "  bucket 34: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 94977 for bucket 27\n",
      "  bucket 30: 70%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 55114 for bucket 28\n",
      "  bucket 31: 60%\n",
      "  bucket 29: 80%\n",
      "Getting block 35 of 57\n",
      "  Reserving size (109253) for bucket 35\n",
      "  Calculating Z arrays for bucket 35\n",
      "  Entering block accumulator loop for bucket 35:\n",
      "  bucket 32: 40%\n",
      "Getting block 36 of 57\n",
      "  Reserving size (109253) for bucket 36\n",
      "  Calculating Z arrays for bucket 36\n",
      "  Entering block accumulator loop for bucket 36:\n",
      "  bucket 33: 30%\n",
      "  bucket 34: 20%\n",
      "  bucket 30: 80%\n",
      "  bucket 35: 10%\n",
      "  bucket 31: 70%\n",
      "  bucket 29: 90%\n",
      "  bucket 32: 50%\n",
      "  bucket 33: 40%\n",
      "  bucket 36: 10%\n",
      "  bucket 34: 30%\n",
      "  bucket 30: 90%\n",
      "  bucket 35: 20%\n",
      "  bucket 29: 100%\n",
      "  Sorting block of length 89941 for bucket 29\n",
      "  (Using difference cover)\n",
      "  bucket 31: 80%\n",
      "  bucket 32: 60%\n",
      "  bucket 33: 50%\n",
      "  bucket 36: 20%\n",
      "  bucket 34: 40%\n",
      "  bucket 35: 30%\n",
      "  bucket 30: 100%\n",
      "  Sorting block of length 78083 for bucket 30\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 89942 for bucket 29\n",
      "  bucket 32: 70%\n",
      "  bucket 33: 60%\n",
      "  bucket 31: 90%\n",
      "Getting block 37 of 57\n",
      "  Reserving size (109253) for bucket 37\n",
      "  Calculating Z arrays for bucket 37\n",
      "  Entering block accumulator loop for bucket 37:\n",
      "  bucket 36: 30%\n",
      "  bucket 34: 50%\n",
      "  bucket 35: 40%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 78084 for bucket 30\n",
      "  bucket 33: 70%\n",
      "  bucket 32: 80%\n",
      "  bucket 31: 100%\n",
      "  Sorting block of length 78116 for bucket 31\n",
      "  (Using difference cover)\n",
      "Getting block 38 of 57\n",
      "  Reserving size (109253) for bucket 38\n",
      "  Calculating Z arrays for bucket 38\n",
      "  Entering block accumulator loop for bucket 38:\n",
      "  bucket 37: 10%\n",
      "  bucket 36: 40%\n",
      "  bucket 35: 50%\n",
      "  bucket 34: 60%\n",
      "  bucket 33: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 78117 for bucket 31\n",
      "  bucket 32: 90%\n",
      "  bucket 38: 10%\n",
      "Getting block 39 of 57\n",
      "  Reserving size (109253) for bucket 39\n",
      "  Calculating Z arrays for bucket 39\n",
      "  Entering block accumulator loop for bucket 39:\n",
      "  bucket 37: 20%\n",
      "  bucket 36: 50%\n",
      "  bucket 35: 60%\n",
      "  bucket 34: 70%\n",
      "  bucket 33: 90%\n",
      "  bucket 32: 100%\n",
      "  Sorting block of length 61661 for bucket 32\n",
      "  (Using difference cover)\n",
      "  bucket 38: 20%\n",
      "  bucket 39: 10%\n",
      "  bucket 37: 30%\n",
      "  bucket 36: 60%\n",
      "  bucket 35: 70%\n",
      "  bucket 34: 80%\n",
      "  bucket 33: 100%\n",
      "  Sorting block of length 96953 for bucket 33\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 61662 for bucket 32\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  bucket 38: 30%\n",
      "Getting block 40 of 57\n",
      "  Reserving size (109253) for bucket 40\n",
      "  Calculating Z arrays for bucket 40\n",
      "  Entering block accumulator loop for bucket 40:\n",
      "  bucket 39: 20%\n",
      "  bucket 37: 40%\n",
      "  bucket 35: 80%\n",
      "  bucket 36: 70%\n",
      "  bucket 34: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 96954 for bucket 33\n",
      "  bucket 38: 40%\n",
      "Getting block 41 of 57\n",
      "  Reserving size (109253) for bucket 41\n",
      "  Calculating Z arrays for bucket 41\n",
      "  Entering block accumulator loop for bucket 41:\n",
      "  bucket 40: 10%\n",
      "  bucket 39: 30%\n",
      "  bucket 37: 50%\n",
      "  bucket 35: 90%\n",
      "  bucket 36: 80%\n",
      "  bucket 34: 100%\n",
      "  Sorting block of length 82279 for bucket 34\n",
      "  (Using difference cover)\n",
      "  bucket 41: 10%\n",
      "  bucket 38: 50%\n",
      "  bucket 40: 20%\n",
      "  bucket 39: 40%\n",
      "  bucket 35: 100%\n",
      "  Sorting block of length 55094 for bucket 35\n",
      "  (Using difference cover)\n",
      "  bucket 37: 60%\n",
      "  bucket 36: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 82280 for bucket 34\n",
      "  bucket 41: 20%\n",
      "Getting block 42 of 57\n",
      "  Reserving size (109253) for bucket 42\n",
      "  Calculating Z arrays for bucket 42\n",
      "  Entering block accumulator loop for bucket 42:\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 55095 for bucket 35\n",
      "  bucket 38: 60%\n",
      "Getting block 43 of 57\n",
      "  Reserving size (109253) for bucket 43\n",
      "  Calculating Z arrays for bucket 43\n",
      "  Entering block accumulator loop for bucket 43:\n",
      "  bucket 39: 50%\n",
      "  bucket 40: 30%\n",
      "  bucket 37: 70%\n",
      "  bucket 36: 100%\n",
      "  Sorting block of length 109105 for bucket 36\n",
      "  (Using difference cover)\n",
      "  bucket 41: 30%\n",
      "  bucket 42: 10%\n",
      "  bucket 38: 70%\n",
      "  bucket 43: 10%\n",
      "  bucket 39: 60%\n",
      "  bucket 40: 40%\n",
      "  bucket 37: 80%\n",
      "  bucket 41: 40%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 109106 for bucket 36\n",
      "  bucket 38: 80%\n",
      "  bucket 42: 20%\n",
      "  bucket 43: 20%\n",
      "Getting block 44 of 57\n",
      "  Reserving size (109253) for bucket 44\n",
      "  Calculating Z arrays for bucket 44\n",
      "  Entering block accumulator loop for bucket 44:\n",
      "  bucket 39: 70%\n",
      "  bucket 40: 50%\n",
      "  bucket 37: 90%\n",
      "  bucket 41: 50%\n",
      "  bucket 38: 90%\n",
      "  bucket 43: 30%\n",
      "  bucket 42: 30%\n",
      "  bucket 39: 80%\n",
      "  bucket 44: 10%\n",
      "  bucket 41: 60%\n",
      "  bucket 37: 100%\n",
      "  Sorting block of length 88232 for bucket 37\n",
      "  (Using difference cover)\n",
      "  bucket 40: 60%\n",
      "  bucket 43: 40%\n",
      "  bucket 38: 100%\n",
      "  Sorting block of length 67429 for bucket 38\n",
      "  (Using difference cover)\n",
      "  bucket 42: 40%\n",
      "  bucket 41: 70%\n",
      "  bucket 39: 90%\n",
      "  bucket 44: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 88233 for bucket 37\n",
      "  bucket 40: 70%\n",
      "Getting block 45 of 57\n",
      "  Reserving size (109253) for bucket 45\n",
      "  Calculating Z arrays for bucket 45\n",
      "  Entering block accumulator loop for bucket 45:\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 67430 for bucket 38\n",
      "  bucket 43: 50%\n",
      "Getting block 46 of 57\n",
      "  Reserving size (109253) for bucket 46\n",
      "  Calculating Z arrays for bucket 46\n",
      "  Entering block accumulator loop for bucket 46:\n",
      "  bucket 42: 50%\n",
      "  bucket 41: 80%\n",
      "  bucket 39: 100%\n",
      "  Sorting block of length 62928 for bucket 39\n",
      "  (Using difference cover)\n",
      "  bucket 44: 30%\n",
      "  bucket 40: 80%\n",
      "  bucket 45: 10%\n",
      "  bucket 43: 60%\n",
      "  bucket 46: 10%\n",
      "  bucket 41: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 62929 for bucket 39\n",
      "  bucket 42: 60%\n",
      "Getting block 47 of 57\n",
      "  Reserving size (109253) for bucket 47\n",
      "  Calculating Z arrays for bucket 47\n",
      "  Entering block accumulator loop for bucket 47:\n",
      "  bucket 44: 40%\n",
      "  bucket 45: 20%\n",
      "  bucket 40: 90%\n",
      "  bucket 43: 70%\n",
      "  bucket 46: 20%\n",
      "  bucket 41: 100%\n",
      "  Sorting block of length 94226 for bucket 41\n",
      "  (Using difference cover)\n",
      "  bucket 42: 70%\n",
      "  bucket 45: 30%\n",
      "  bucket 47: 10%\n",
      "  bucket 44: 50%\n",
      "  bucket 43: 80%\n",
      "  bucket 40: 100%\n",
      "  Sorting block of length 95878 for bucket 40\n",
      "  (Using difference cover)\n",
      "  bucket 46: 30%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 94227 for bucket 41\n",
      "  bucket 42: 80%\n",
      "  bucket 45: 40%\n",
      "Getting block 48 of 57\n",
      "  Reserving size (109253) for bucket 48\n",
      "  Calculating Z arrays for bucket 48\n",
      "  Entering block accumulator loop for bucket 48:\n",
      "  bucket 47: 20%\n",
      "  bucket 43: 90%\n",
      "  bucket 44: 60%\n",
      "  bucket 46: 40%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 95879 for bucket 40\n",
      "  bucket 45: 50%\n",
      "  bucket 48: 10%\n",
      "  bucket 42: 90%\n",
      "  bucket 43: 100%\n",
      "  Sorting block of length 44927 for bucket 43\n",
      "  (Using difference cover)\n",
      "Getting block 49 of 57\n",
      "  Reserving size (109253) for bucket 49\n",
      "  Calculating Z arrays for bucket 49\n",
      "  Entering block accumulator loop for bucket 49:\n",
      "  bucket 47: 30%\n",
      "  bucket 44: 70%\n",
      "  bucket 46: 50%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 44928 for bucket 43\n",
      "  bucket 48: 20%\n",
      "  bucket 45: 60%\n",
      "Getting block 50 of 57\n",
      "  Reserving size (109253) for bucket 50\n",
      "  Calculating Z arrays for bucket 50\n",
      "  Entering block accumulator loop for bucket 50:\n",
      "  bucket 42: 100%\n",
      "  Sorting block of length 102218 for bucket 42\n",
      "  (Using difference cover)\n",
      "  bucket 47: 40%\n",
      "  bucket 49: 10%\n",
      "  bucket 46: 60%\n",
      "  bucket 44: 80%\n",
      "  bucket 48: 30%\n",
      "  bucket 45: 70%\n",
      "  bucket 50: 10%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 102219 for bucket 42\n",
      "  bucket 47: 50%\n",
      "  bucket 46: 70%\n",
      "  bucket 49: 20%\n",
      "  bucket 48: 40%\n",
      "Getting block 51 of 57\n",
      "  Reserving size (109253) for bucket 51\n",
      "  Calculating Z arrays for bucket 51\n",
      "  Entering block accumulator loop for bucket 51:\n",
      "  bucket 44: 90%\n",
      "  bucket 45: 80%\n",
      "  bucket 50: 20%\n",
      "  bucket 46: 80%\n",
      "  bucket 48: 50%\n",
      "  bucket 47: 60%\n",
      "  bucket 51: 10%\n",
      "  bucket 49: 30%\n",
      "  bucket 45: 90%\n",
      "  bucket 50: 30%\n",
      "  bucket 44: 100%\n",
      "  Sorting block of length 88889 for bucket 44\n",
      "  (Using difference cover)\n",
      "  bucket 46: 90%\n",
      "  bucket 48: 60%\n",
      "  bucket 47: 70%\n",
      "  bucket 51: 20%\n",
      "  bucket 50: 40%\n",
      "  bucket 45: 100%\n",
      "  Sorting block of length 98395 for bucket 45\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 88890 for bucket 44\n",
      "  bucket 49: 40%\n",
      "Getting block 52 of 57\n",
      "  Reserving size (109253) for bucket 52\n",
      "  Calculating Z arrays for bucket 52\n",
      "  Entering block accumulator loop for bucket 52:\n",
      "  bucket 46: 100%\n",
      "  Sorting block of length 92364 for bucket 46\n",
      "  (Using difference cover)\n",
      "  bucket 48: 70%\n",
      "  bucket 50: 50%\n",
      "  bucket 51: 30%\n",
      "  bucket 47: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 98396 for bucket 45\n",
      "  bucket 49: 50%\n",
      "  bucket 52: 10%\n",
      "Getting block 53 of 57\n",
      "  Reserving size (109253) for bucket 53\n",
      "  Calculating Z arrays for bucket 53\n",
      "  Entering block accumulator loop for bucket 53:\n",
      "  bucket 48: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 92365 for bucket 46\n",
      "  bucket 50: 60%\n",
      "Getting block 54 of 57\n",
      "  Reserving size (109253) for bucket 54\n",
      "  Calculating Z arrays for bucket 54\n",
      "  Entering block accumulator loop for bucket 54:\n",
      "  bucket 51: 40%\n",
      "  bucket 47: 90%\n",
      "  bucket 53: 10%\n",
      "  bucket 52: 20%\n",
      "  bucket 48: 90%\n",
      "  bucket 49: 60%\n",
      "  bucket 50: 70%\n",
      "  bucket 51: 50%\n",
      "  bucket 54: 10%\n",
      "  bucket 47: 100%\n",
      "  Sorting block of length 102042 for bucket 47\n",
      "  (Using difference cover)\n",
      "  bucket 48: 100%\n",
      "  Sorting block of length 107496 for bucket 48\n",
      "  (Using difference cover)\n",
      "  bucket 52: 30%\n",
      "  bucket 53: 20%\n",
      "  bucket 49: 70%\n",
      "  bucket 50: 80%\n",
      "  bucket 51: 60%\n",
      "  bucket 54: 20%\n",
      "  bucket 52: 40%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 102043 for bucket 47\n",
      "  bucket 53: 30%\n",
      "  bucket 50: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 107497 for bucket 48\n",
      "Getting block 55 of 57\n",
      "  Reserving size (109253) for bucket 55\n",
      "  Calculating Z arrays for bucket 55\n",
      "  Entering block accumulator loop for bucket 55:\n",
      "  bucket 49: 80%\n",
      "  bucket 51: 70%\n",
      "Getting block 56 of 57\n",
      "  Reserving size (109253) for bucket 56\n",
      "  Calculating Z arrays for bucket 56\n",
      "  Entering block accumulator loop for bucket 56:\n",
      "  bucket 54: 30%\n",
      "  bucket 52: 50%\n",
      "  bucket 53: 40%\n",
      "  bucket 50: 100%\n",
      "  Sorting block of length 86710 for bucket 50\n",
      "  (Using difference cover)\n",
      "  bucket 55: 10%\n",
      "  bucket 56: 10%\n",
      "  bucket 49: 90%\n",
      "  bucket 51: 80%\n",
      "  bucket 52: 60%\n",
      "  bucket 53: 50%\n",
      "  bucket 54: 40%\n",
      "  bucket 56: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 86711 for bucket 50\n",
      "  bucket 55: 20%\n",
      "  bucket 51: 90%\n",
      "Getting block 57 of 57\n",
      "  Reserving size (109253) for bucket 57\n",
      "  Calculating Z arrays for bucket 57\n",
      "  Entering block accumulator loop for bucket 57:\n",
      "  bucket 49: 100%\n",
      "  Sorting block of length 95722 for bucket 49\n",
      "  (Using difference cover)\n",
      "  bucket 52: 70%\n",
      "  bucket 53: 60%\n",
      "  bucket 56: 30%\n",
      "  bucket 57: 10%\n",
      "  bucket 54: 50%\n",
      "  bucket 55: 30%\n",
      "  bucket 51: 100%\n",
      "  Sorting block of length 80125 for bucket 51\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  (Using difference cover)\n",
      "  bucket 57: 20%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 95723 for bucket 49\n",
      "  bucket 53: 70%\n",
      "  bucket 52: 80%\n",
      "  bucket 56: 40%\n",
      "  bucket 57: 30%\n",
      "  bucket 54: 60%\n",
      "  bucket 55: 40%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 80126 for bucket 51\n",
      "  bucket 57: 40%\n",
      "  bucket 53: 80%\n",
      "  bucket 56: 50%\n",
      "  bucket 52: 90%\n",
      "  bucket 57: 50%\n",
      "  bucket 55: 50%\n",
      "  bucket 54: 70%\n",
      "  bucket 56: 60%\n",
      "  bucket 57: 60%\n",
      "  bucket 53: 90%\n",
      "  bucket 52: 100%\n",
      "  Sorting block of length 92223 for bucket 52\n",
      "  (Using difference cover)\n",
      "  bucket 57: 70%\n",
      "  bucket 55: 60%\n",
      "  bucket 56: 70%\n",
      "  bucket 54: 80%\n",
      "  bucket 53: 100%\n",
      "  Sorting block of length 89439 for bucket 53\n",
      "  (Using difference cover)\n",
      "  bucket 57: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 92224 for bucket 52\n",
      "  bucket 55: 70%\n",
      "  bucket 57: 90%\n",
      "  bucket 56: 80%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 89440 for bucket 53\n",
      "  bucket 54: 90%\n",
      "  bucket 57: 100%\n",
      "  Sorting block of length 38235 for bucket 57\n",
      "  (Using difference cover)\n",
      "  bucket 55: 80%\n",
      "  bucket 56: 90%\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 38236 for bucket 57\n",
      "  bucket 55: 90%\n",
      "  bucket 54: 100%\n",
      "  Sorting block of length 57470 for bucket 54\n",
      "  (Using difference cover)\n",
      "  bucket 56: 100%\n",
      "  Sorting block of length 98970 for bucket 56\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 57471 for bucket 54\n",
      "  bucket 55: 100%\n",
      "  Sorting block of length 105902 for bucket 55\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 98971 for bucket 56\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 105903 for bucket 55\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exited GFM loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 1150437\n",
      "fchr[G]: 2332653\n",
      "fchr[T]: 3513594\n",
      "fchr[$]: 4661465\n",
      "Exiting GFM::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 5752637 bytes to primary GFM file: ./yan_results/genome_index/genome.1.ht2\n",
      "Wrote 1165372 bytes to secondary GFM file: ./yan_results/genome_index/genome.2.ht2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from GFM constructor\n",
      "Returning from initFromVector\n",
      "Wrote 2054631 bytes to primary GFM file: ./yan_results/genome_index/genome.5.ht2\n",
      "Wrote 1186356 bytes to secondary GFM file: ./yan_results/genome_index/genome.6.ht2\n",
      "Re-opening _in5 and _in5 as input streams\n",
      "Returning from HGFM constructor\n",
      "Headers:\n",
      "    len: 4661465\n",
      "    gbwtLen: 4661466\n",
      "    nodes: 4661466\n",
      "    sz: 1165367\n",
      "    gbwtSz: 1165367\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 0\n",
      "    eftabSz: 0\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 291342\n",
      "    offsSz: 1165368\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideGbwtSz: 48\n",
      "    sideGbwtLen: 192\n",
      "    numSides: 24279\n",
      "    numLines: 24279\n",
      "    gbwtTotLen: 1553856\n",
      "    gbwtTotSz: 1553856\n",
      "    reverse: 0\n",
      "    linearFM: Yes\n",
      "Total time for call to driver() for forward index: 00:00:01\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "GENOME_file=\"./yan_results/Escherichia_coli_K12_and_T1_phage_genome.fa\"\n",
    "\n",
    "hisat2-build -p 8 $GENOME_file ./yan_results/genome_index/genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### run hisat2 on each filter fastq file, -1 means the reverse sequence file, -2 means the forward sequence file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### note that when distinguish -1 file, -2 file, we need to identify the same suffix of the filter files, here I use the regular expression 'a[0-9]+_.*t.?[0-9]'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# test for the filtering suffix of the file name(success)\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/filter/*.fastq | grep -Eo 'a[0-9]+_.*t.?[0-9]' -c | sort | uniq )\n",
    "echo $fastq_name\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "##move all the result of the filter file to the filter folder\n",
    "cd ./20211105180528381-5936/original/DEFAULT/\n",
    "mv *filter.fastq* ./filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HISAT2 version 2.2.1 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)\n",
      "Usage: \n",
      "  hisat2 [options]* -x <ht2-idx> {-1 <m1> -2 <m2> | -U <r> | --sra-acc <SRA accession number>} [-S <sam>]\n",
      "\n",
      "  <ht2-idx>  Index filename prefix (minus trailing .X.ht2).\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <SRA accession number>        Comma-separated list of SRA accession numbers, e.g. --sra-acc SRR353653,SRR353654.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "  --sra-acc          SRA accession ID\n",
      "\n",
      " Presets:                 Same as:\n",
      "   --fast                 --no-repeat-index\n",
      "   --sensitive            --bowtie2-dp 1 -k 30 --score-min L,0,-0.5\n",
      "   --very-sensitive       --bowtie2-dp 2 -k 50 --score-min L,0,-1\n",
      "\n",
      " Alignment:\n",
      "  --bowtie2-dp <int> use Bowtie2's dynamic programming alignment algorithm (0) - 0: no dynamic programming, 1: conditional dynamic programming, and 2: unconditional dynamic programming (slowest)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-repeat-index  do not use repeat index\n",
      "\n",
      " Spliced Alignment:\n",
      "  --pen-cansplice <int>              penalty for a canonical splice site (0)\n",
      "  --pen-noncansplice <int>           penalty for a non-canonical splice site (12)\n",
      "  --pen-canintronlen <func>          penalty for long introns (G,-8,1) with canonical splice sites\n",
      "  --pen-noncanintronlen <func>       penalty for long introns (G,-8,1) with noncanonical splice sites\n",
      "  --min-intronlen <int>              minimum intron length (20)\n",
      "  --max-intronlen <int>              maximum intron length (500000)\n",
      "  --known-splicesite-infile <path>   provide a list of known splice sites\n",
      "  --novel-splicesite-outfile <path>  report a list of splice sites\n",
      "  --novel-splicesite-infile <path>   provide a list of novel splice sites\n",
      "  --no-temp-splicesite               disable the use of splice sites found\n",
      "  --no-spliced-alignment             disable spliced alignment\n",
      "  --rna-strandness <string>          specify strand-specific information (unstranded)\n",
      "  --tmo                              reports only those alignments within known transcriptome\n",
      "  --dta                              reports alignments tailored for transcript assemblers\n",
      "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks\n",
      "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)\n",
      "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads\n",
      "\n",
      " Scoring:\n",
      "  --mp <int>,<int>   max and min penalties for mismatch; lower qual = lower penalty <6,2>\n",
      "  --sp <int>,<int>   max and min penalties for soft-clipping; lower qual = lower penalty <2,1>\n",
      "  --no-softclip      no soft-clipping\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (L,0.0,-0.2)\n",
      "\n",
      " Reporting:\n",
      "  -k <int>           It searches for at most <int> distinct, primary alignments for each read. Primary alignments mean \n",
      "                     alignments whose alignment score is equal to or higher than any other alignments. The search terminates \n",
      "                     when it cannot find more distinct valid alignments, or when it finds <int>, whichever happens first. \n",
      "                     The alignment score for a paired-end alignment equals the sum of the alignment scores of \n",
      "                     the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit \n",
      "                     (which equals 256) set in its FLAGS field. For reads that have more than <int> distinct, \n",
      "                     valid alignments, hisat2 does not guarantee that the <int> alignments reported are the best possible \n",
      "                     in terms of alignment score. Default: 5 (linear index) or 10 (graph index).\n",
      "                     Note: HISAT2 is not designed with large values for -k in mind, and when aligning reads to long, \n",
      "                     repetitive genomes, large -k could make alignment much slower.\n",
      "  --max-seeds <int>  HISAT2, like other aligners, uses seed-and-extend approaches. HISAT2 tries to extend seeds to \n",
      "                     full-length alignments. In HISAT2, --max-seeds is used to control the maximum number of seeds that \n",
      "                     will be extended. For DNA-read alignment (--no-spliced-alignment), HISAT2 extends up to these many seeds\n",
      "                     and skips the rest of the seeds. For RNA-read alignment, HISAT2 skips extending seeds and reports \n",
      "                     no alignments if the number of seeds is larger than the number specified with the option, \n",
      "                     to be compatible with previous versions of HISAT2. Large values for --max-seeds may improve alignment \n",
      "                     sensitivity, but HISAT2 is not designed with large values for --max-seeds in mind, and when aligning \n",
      "                     reads to long, repetitive genomes, large --max-seeds could make alignment much slower. \n",
      "                     The default value is the maximum of 5 and the value that comes with -k times 2.\n",
      "  -a/--all           HISAT2 reports all alignments it can find. Using the option is equivalent to using both --max-seeds \n",
      "                     and -k with the maximum value that a 64-bit signed integer can represent (9,223,372,036,854,775,807).\n",
      "  --repeat           report alignments to repeat sequences directly\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0), only valid with --no-spliced-alignment\n",
      "  -X/--maxins <int>  maximum fragment length (500), only valid with --no-spliced-alignment\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>           write unpaired reads that didn't align to <path>\n",
      "  --al <path>           write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>      write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>      write pairs that aligned concordantly at least once to <path>\n",
      "  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --summary-file <path> print alignment summary to this file.\n",
      "  --new-summary         print alignment summary in a new style, which is more machine-friendly.\n",
      "  --quiet               print nothing to stderr except serious errors\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  --met-file <path>     send metrics to file at <path> (off)\n",
      "  --met-stderr          send metrics to stderr (off)\n",
      "  --met <int>           report internal counters & metrics every <int> secs (1)\n",
      "  --no-head             suppress header lines, i.e. lines starting with @\n",
      "  --no-sq               suppress @SQ header lines\n",
      "  --rg-id <text>        set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>           add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                        Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq        put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "\n",
      " Performance:\n",
      "  -o/--offrate <int> override offrate of index; must be >= index's offrate\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'hisat2's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --remove-chrname   remove 'chr' from reference names in alignment\n",
      "  --add-chrname      add 'chr' to reference names in alignment \n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "hisat2 --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "25645193 reads; of these:\n",
      "  25645193 (100.00%) were paired; of these:\n",
      "    970589 (3.78%) aligned concordantly 0 times\n",
      "    24450969 (95.34%) aligned concordantly exactly 1 time\n",
      "    223635 (0.87%) aligned concordantly >1 times\n",
      "    ----\n",
      "    970589 pairs aligned concordantly 0 times; of these:\n",
      "      184109 (18.97%) aligned discordantly 1 time\n",
      "    ----\n",
      "    786480 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1572960 mates make up the pairs; of these:\n",
      "        962371 (61.18%) aligned 0 times\n",
      "        600734 (38.19%) aligned exactly 1 time\n",
      "        9855 (0.63%) aligned >1 times\n",
      "98.12% overall alignment rate\n",
      "24662790 reads; of these:\n",
      "  24662790 (100.00%) were paired; of these:\n",
      "    945163 (3.83%) aligned concordantly 0 times\n",
      "    23443898 (95.06%) aligned concordantly exactly 1 time\n",
      "    273729 (1.11%) aligned concordantly >1 times\n",
      "    ----\n",
      "    945163 pairs aligned concordantly 0 times; of these:\n",
      "      88103 (9.32%) aligned discordantly 1 time\n",
      "    ----\n",
      "    857060 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1714120 mates make up the pairs; of these:\n",
      "        985683 (57.50%) aligned 0 times\n",
      "        717677 (41.87%) aligned exactly 1 time\n",
      "        10760 (0.63%) aligned >1 times\n",
      "98.00% overall alignment rate\n",
      "24933926 reads; of these:\n",
      "  24933926 (100.00%) were paired; of these:\n",
      "    796091 (3.19%) aligned concordantly 0 times\n",
      "    23864505 (95.71%) aligned concordantly exactly 1 time\n",
      "    273330 (1.10%) aligned concordantly >1 times\n",
      "    ----\n",
      "    796091 pairs aligned concordantly 0 times; of these:\n",
      "      96275 (12.09%) aligned discordantly 1 time\n",
      "    ----\n",
      "    699816 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1399632 mates make up the pairs; of these:\n",
      "        947852 (67.72%) aligned 0 times\n",
      "        443360 (31.68%) aligned exactly 1 time\n",
      "        8420 (0.60%) aligned >1 times\n",
      "98.10% overall alignment rate\n",
      "27490537 reads; of these:\n",
      "  27490537 (100.00%) were paired; of these:\n",
      "    897154 (3.26%) aligned concordantly 0 times\n",
      "    26270970 (95.56%) aligned concordantly exactly 1 time\n",
      "    322413 (1.17%) aligned concordantly >1 times\n",
      "    ----\n",
      "    897154 pairs aligned concordantly 0 times; of these:\n",
      "      70748 (7.89%) aligned discordantly 1 time\n",
      "    ----\n",
      "    826406 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1652812 mates make up the pairs; of these:\n",
      "        1113499 (67.37%) aligned 0 times\n",
      "        530221 (32.08%) aligned exactly 1 time\n",
      "        9092 (0.55%) aligned >1 times\n",
      "97.97% overall alignment rate\n",
      "28452918 reads; of these:\n",
      "  28452918 (100.00%) were paired; of these:\n",
      "    895440 (3.15%) aligned concordantly 0 times\n",
      "    27268429 (95.84%) aligned concordantly exactly 1 time\n",
      "    289049 (1.02%) aligned concordantly >1 times\n",
      "    ----\n",
      "    895440 pairs aligned concordantly 0 times; of these:\n",
      "      70396 (7.86%) aligned discordantly 1 time\n",
      "    ----\n",
      "    825044 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1650088 mates make up the pairs; of these:\n",
      "        1101300 (66.74%) aligned 0 times\n",
      "        539110 (32.67%) aligned exactly 1 time\n",
      "        9678 (0.59%) aligned >1 times\n",
      "98.06% overall alignment rate\n",
      "29907013 reads; of these:\n",
      "  29907013 (100.00%) were paired; of these:\n",
      "    926709 (3.10%) aligned concordantly 0 times\n",
      "    28696966 (95.95%) aligned concordantly exactly 1 time\n",
      "    283338 (0.95%) aligned concordantly >1 times\n",
      "    ----\n",
      "    926709 pairs aligned concordantly 0 times; of these:\n",
      "      73652 (7.95%) aligned discordantly 1 time\n",
      "    ----\n",
      "    853057 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1706114 mates make up the pairs; of these:\n",
      "        1122889 (65.82%) aligned 0 times\n",
      "        572579 (33.56%) aligned exactly 1 time\n",
      "        10646 (0.62%) aligned >1 times\n",
      "98.12% overall alignment rate\n",
      "23686198 reads; of these:\n",
      "  23686198 (100.00%) were paired; of these:\n",
      "    855601 (3.61%) aligned concordantly 0 times\n",
      "    22619003 (95.49%) aligned concordantly exactly 1 time\n",
      "    211594 (0.89%) aligned concordantly >1 times\n",
      "    ----\n",
      "    855601 pairs aligned concordantly 0 times; of these:\n",
      "      66628 (7.79%) aligned discordantly 1 time\n",
      "    ----\n",
      "    788973 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1577946 mates make up the pairs; of these:\n",
      "        974109 (61.73%) aligned 0 times\n",
      "        594711 (37.69%) aligned exactly 1 time\n",
      "        9126 (0.58%) aligned >1 times\n",
      "97.94% overall alignment rate\n",
      "26099845 reads; of these:\n",
      "  26099845 (100.00%) were paired; of these:\n",
      "    1070547 (4.10%) aligned concordantly 0 times\n",
      "    24706761 (94.66%) aligned concordantly exactly 1 time\n",
      "    322537 (1.24%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1070547 pairs aligned concordantly 0 times; of these:\n",
      "      81790 (7.64%) aligned discordantly 1 time\n",
      "    ----\n",
      "    988757 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1977514 mates make up the pairs; of these:\n",
      "        1130308 (57.16%) aligned 0 times\n",
      "        834576 (42.20%) aligned exactly 1 time\n",
      "        12630 (0.64%) aligned >1 times\n",
      "97.83% overall alignment rate\n",
      "36914611 reads; of these:\n",
      "  36914611 (100.00%) were paired; of these:\n",
      "    1049210 (2.84%) aligned concordantly 0 times\n",
      "    35440573 (96.01%) aligned concordantly exactly 1 time\n",
      "    424828 (1.15%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1049210 pairs aligned concordantly 0 times; of these:\n",
      "      91212 (8.69%) aligned discordantly 1 time\n",
      "    ----\n",
      "    957998 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1915996 mates make up the pairs; of these:\n",
      "        1266222 (66.09%) aligned 0 times\n",
      "        637940 (33.30%) aligned exactly 1 time\n",
      "        11834 (0.62%) aligned >1 times\n",
      "98.28% overall alignment rate\n",
      "25986950 reads; of these:\n",
      "  25986950 (100.00%) were paired; of these:\n",
      "    850071 (3.27%) aligned concordantly 0 times\n",
      "    24852470 (95.63%) aligned concordantly exactly 1 time\n",
      "    284409 (1.09%) aligned concordantly >1 times\n",
      "    ----\n",
      "    850071 pairs aligned concordantly 0 times; of these:\n",
      "      86498 (10.18%) aligned discordantly 1 time\n",
      "    ----\n",
      "    763573 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1527146 mates make up the pairs; of these:\n",
      "        1054215 (69.03%) aligned 0 times\n",
      "        463138 (30.33%) aligned exactly 1 time\n",
      "        9793 (0.64%) aligned >1 times\n",
      "97.97% overall alignment rate\n",
      "27081803 reads; of these:\n",
      "  27081803 (100.00%) were paired; of these:\n",
      "    874145 (3.23%) aligned concordantly 0 times\n",
      "    25952183 (95.83%) aligned concordantly exactly 1 time\n",
      "    255475 (0.94%) aligned concordantly >1 times\n",
      "    ----\n",
      "    874145 pairs aligned concordantly 0 times; of these:\n",
      "      80011 (9.15%) aligned discordantly 1 time\n",
      "    ----\n",
      "    794134 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1588268 mates make up the pairs; of these:\n",
      "        1064507 (67.02%) aligned 0 times\n",
      "        515379 (32.45%) aligned exactly 1 time\n",
      "        8382 (0.53%) aligned >1 times\n",
      "98.03% overall alignment rate\n",
      "24849769 reads; of these:\n",
      "  24849769 (100.00%) were paired; of these:\n",
      "    788694 (3.17%) aligned concordantly 0 times\n",
      "    23850722 (95.98%) aligned concordantly exactly 1 time\n",
      "    210353 (0.85%) aligned concordantly >1 times\n",
      "    ----\n",
      "    788694 pairs aligned concordantly 0 times; of these:\n",
      "      67174 (8.52%) aligned discordantly 1 time\n",
      "    ----\n",
      "    721520 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1443040 mates make up the pairs; of these:\n",
      "        917160 (63.56%) aligned 0 times\n",
      "        519062 (35.97%) aligned exactly 1 time\n",
      "        6818 (0.47%) aligned >1 times\n",
      "98.15% overall alignment rate\n",
      "30473061 reads; of these:\n",
      "  30473061 (100.00%) were paired; of these:\n",
      "    1100449 (3.61%) aligned concordantly 0 times\n",
      "    29023589 (95.24%) aligned concordantly exactly 1 time\n",
      "    349023 (1.15%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1100449 pairs aligned concordantly 0 times; of these:\n",
      "      112099 (10.19%) aligned discordantly 1 time\n",
      "    ----\n",
      "    988350 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1976700 mates make up the pairs; of these:\n",
      "        1177033 (59.55%) aligned 0 times\n",
      "        785903 (39.76%) aligned exactly 1 time\n",
      "        13764 (0.70%) aligned >1 times\n",
      "98.07% overall alignment rate\n",
      "31811732 reads; of these:\n",
      "  31811732 (100.00%) were paired; of these:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    1007107 (3.17%) aligned concordantly 0 times\n",
      "    30475854 (95.80%) aligned concordantly exactly 1 time\n",
      "    328771 (1.03%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1007107 pairs aligned concordantly 0 times; of these:\n",
      "      84463 (8.39%) aligned discordantly 1 time\n",
      "    ----\n",
      "    922644 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1845288 mates make up the pairs; of these:\n",
      "        1237851 (67.08%) aligned 0 times\n",
      "        595828 (32.29%) aligned exactly 1 time\n",
      "        11609 (0.63%) aligned >1 times\n",
      "98.05% overall alignment rate\n",
      "23404625 reads; of these:\n",
      "  23404625 (100.00%) were paired; of these:\n",
      "    796911 (3.40%) aligned concordantly 0 times\n",
      "    22338205 (95.44%) aligned concordantly exactly 1 time\n",
      "    269509 (1.15%) aligned concordantly >1 times\n",
      "    ----\n",
      "    796911 pairs aligned concordantly 0 times; of these:\n",
      "      94238 (11.83%) aligned discordantly 1 time\n",
      "    ----\n",
      "    702673 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1405346 mates make up the pairs; of these:\n",
      "        962348 (68.48%) aligned 0 times\n",
      "        434006 (30.88%) aligned exactly 1 time\n",
      "        8992 (0.64%) aligned >1 times\n",
      "97.94% overall alignment rate\n",
      "26271319 reads; of these:\n",
      "  26271319 (100.00%) were paired; of these:\n",
      "    914213 (3.48%) aligned concordantly 0 times\n",
      "    25099813 (95.54%) aligned concordantly exactly 1 time\n",
      "    257293 (0.98%) aligned concordantly >1 times\n",
      "    ----\n",
      "    914213 pairs aligned concordantly 0 times; of these:\n",
      "      160017 (17.50%) aligned discordantly 1 time\n",
      "    ----\n",
      "    754196 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      1508392 mates make up the pairs; of these:\n",
      "        1022268 (67.77%) aligned 0 times\n",
      "        476711 (31.60%) aligned exactly 1 time\n",
      "        9413 (0.62%) aligned >1 times\n",
      "98.05% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "path_hisat2_index=\"./yan_results/genome_index\"\n",
    "path_fastq=\"./20211105180528381-5936/original/DEFAULT/filter/\"\n",
    "path_of_result=\"./yan_results/bam_result/sam_files\"\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/filter/*.fastq | grep -Eo 'a[0-9]+_.*t.?[0-9]'  | sort | uniq )\n",
    "for ID in $fastq_name\n",
    "    do \n",
    "    filter_R1=$path_fastq$ID\"_1_filter.fastq\"\n",
    "    filter_R2=$path_fastq$ID\"_2_filter.fastq\"\n",
    "    hisat2 -p 40 -x $path_hisat2_index\"/genome\" --rna-strandness RF -1 $filter_R1 -2 $filter_R2 -S $path_of_result\"/\"$ID\".sam\"\n",
    "    done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP4: transfer the SAMfiles to BAMfiles, and sort them then set the index\n",
    "### STEP5:  transfer bam files to bedgraph format, then visualize them on IGV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ./yan_results/bam_result/F320/\n",
    "for i in $(ls /localdisk/home/openngs/openngs/notebooks/ngs/sander/phage_infection_time_course_e_coli_rnaseq/yan_results/bam_result/sam_files/*.sam)\n",
    "do\n",
    "    name=$(basename $i .sam)\n",
    "    samtools view -b -F 320 $i | samtools sort -o $name\"_F320_sorted.bam\"\n",
    "    samtools index $name\"_F320_sorted.bam\"\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand + > $name\"_F320_plus_strand.bedgraph\"\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand - > $name\"_F320_minus_strand.bedgraph\"\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand + -5 > $name\"_F320_plus_strand_5end.bedgraph\" &\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand - -5 > $name\"_F320_minus_strand_5end.bedgraph\" &\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand + -3 > $name\"_F320_plus_strand_3end.bedgraph\" &\n",
    "    bedtools genomecov -ibam $name\"_F320_sorted.bam\" -bga -strand - -3 > $name\"_F320_minus_strand_3end.bedgraph\" &\n",
    "    \n",
    "    #bedtools genomecov -ibam $i -bga -strand + > $i\\_plus_strand.bedgraph &\n",
    "    #bedtools genomecov -ibam $i -bga -strand - > $i\\_minus_strand.bedgraph &\n",
    "    #bedtools genomecov -ibam $i -bga -strand + -5 > $i\\_plus_strand_5end.bedgraph &\n",
    "    #bedtools genomecov -ibam $i -bga -strand - -5 > $i\\_minus_strand_5end.bedgraph &\n",
    "    #bedtools genomecov -ibam $i -bga -strand + -3 > $i\\_plus_strand_3end.bedgraph &\n",
    "    #bedtools genomecov -ibam $i -bga -strand - -3 > $i\\_minus_strand_3end.bedgraph &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#########################\n",
      "#########################\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 13 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/\n",
    "#head -100 a10_rep2_+T4_t15.sam\n",
    "#samtools view -f 64 a10_rep2_+T4_t15.sam | head -n 5\n",
    "echo \"#########################\"\n",
    "#samtools view a10_rep2_+T4_t15.sam | head -n 10\n",
    "echo \"#########################\"\n",
    "\n",
    "samtools view -b -F 320 ./sam_files/a9_rep2_+T4_t10.sam | samtools sort -o ./F320/a9_rep2_+T4_t10_F320_sorted.bam  \n",
    "samtools index ./F320/a9_rep2_+T4_t10_F320_sorted.bam\n",
    "bedtools genomecov -ibam ./F320/a9_rep2_+T4_t10_F320_sorted.bam -bga -strand + > ./F320/a9_rep2_+T4_t10_F320_plus_strand.bedgraph\n",
    "bedtools genomecov -ibam ./F320/a9_rep2_+T4_t10_F320_sorted.bam -bga -strand - > ./F320/a9_rep2_+T4_t10_F320_minus_strand.bedgraph"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### combine all the replicates of the bedgrapph files into one bedgraph file, then use IGV to visualize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a11_rep2_+T4_t20_F320_minus_strand.bedgraph a17_rep3_+T4_t20_F320_minus_strand.bedgraph a5_rep1_+T4_t20_F320_minus_strand.bedgraph\n",
      "a11_rep2_+T4_t20_F320_plus_strand.bedgraph a17_rep3_+T4_t20_F320_plus_strand.bedgraph a5_rep1_+T4_t20_F320_plus_strand.bedgraph\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "\n",
    "t20_minus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t20_F320_minus_strand.bedgraph'`\n",
    "echo $t20_minus\n",
    "bedtools unionbedg -i $t20_minus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t20_minus_strand.bedgraph\n",
    "\n",
    "t20_plus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t20_F320_plus_strand.bedgraph'`\n",
    "echo $t20_plus\n",
    "bedtools unionbedg -i $t20_plus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t20_plus_strand.bedgraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a15_rep3_+T4_t5_F320_minus_strand.bedgraph a2_rep1_+T4_t5_F320_minus_strand.bedgraph a8_rep2_+T4_t5_F320_minus_strand.bedgraph\n",
      "a15_rep3_+T4_t5_F320_plus_strand.bedgraph a2_rep1_+T4_t5_F320_plus_strand.bedgraph a8_rep2_+T4_t5_F320_plus_strand.bedgraph\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "\n",
    "t5_minus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t5_F320_minus_strand.bedgraph'`\n",
    "echo $t5_minus\n",
    "bedtools unionbedg -i $t5_minus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t5_minus_strand.bedgraph\n",
    "\n",
    "t5_plus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t5_F320_plus_strand.bedgraph'`\n",
    "echo $t5_plus\n",
    "bedtools unionbedg -i $t5_plus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t5_plus_strand.bedgraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a16_rep3_+T4_t10_F320_minus_strand.bedgraph a3_rep1_+T4_t10_F320_minus_strand.bedgraph a9_rep2_+T4_t10_F320_minus_strand.bedgraph\n",
      "a16_rep3_+T4_t10_F320_plus_strand.bedgraph a3_rep1_+T4_t10_F320_plus_strand.bedgraph a9_rep2_+T4_t10_F320_plus_strand.bedgraph\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "\n",
    "t10_minus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t10_F320_minus_strand.bedgraph'`\n",
    "echo $t10_minus\n",
    "bedtools unionbedg -i $t10_minus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t10_minus_strand.bedgraph\n",
    "\n",
    "t10_plus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t10_F320_plus_strand.bedgraph'`\n",
    "echo $t10_plus\n",
    "bedtools unionbedg -i $t10_plus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t10_plus_strand.bedgraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a10_rep2_+T4_t15_F320_minus_strand.bedgraph a4_rep1_+T4_t15_F320_minus_strand.bedgraph\n",
      "a10_rep2_+T4_t15_F320_plus_strand.bedgraph a4_rep1_+T4_t15_F320_plus_strand.bedgraph\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "\n",
    "t15_minus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t15_F320_minus_strand.bedgraph'`\n",
    "echo $t15_minus\n",
    "bedtools unionbedg -i $t15_minus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t15_minus_strand.bedgraph\n",
    "\n",
    "t15_plus=`ls ./*.bedgraph | grep -Eo 'a[0-9]+_.*T4_t15_F320_plus_strand.bedgraph'`\n",
    "echo $t15_plus\n",
    "bedtools unionbedg -i $t15_plus | awk 'OFS=\"\\t\" {sum=0; for (col=4; col<=NF; col++) sum += $col; print $1, $2, $3, sum/(NF-4+1); }' > ./mean_combined_t15_plus_strand.bedgraph"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download all the bedgraph files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mean_combined_t10_minus_strand.bedgraph\n",
      "mean_combined_t10_plus_strand.bedgraph\n",
      "mean_combined_t15_minus_strand.bedgraph\n",
      "mean_combined_t15_plus_strand.bedgraph\n",
      "mean_combined_t20_minus_strand.bedgraph\n",
      "mean_combined_t20_plus_strand.bedgraph\n",
      "mean_combined_t5_minus_strand.bedgraph\n",
      "mean_combined_t5_plus_strand.bedgraph\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "#tar cvf combined_bedgraph.tar combined*\n",
    "tar cvf mean_combined_F320_bedgraph.tar mean_combined*\n",
    "\n",
    "## then click on the download button of the new bedgraph.tar file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### STEP6:  Run the bedtools to get the raw count of per file, and combine the counts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Firstly, create a new file named count.file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ./yan_results/bam_result/F320/\n",
    "touch count_all.file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a10_rep2_+T4_t15 a11_rep2_+T4_t20 a12_rep2_control_t20 a13_rep3_control_t-5 a14_rep3_+T4_t2 a15_rep3_+T4_t5 a16_rep3_+T4_t10 a17_rep3_+T4_t20 a18_rep3_control_t20 a1_rep1_control_t-5 a2_rep1_+T4_t5 a3_rep1_+T4_t10 a4_rep1_+T4_t15 a5_rep1_+T4_t20 a6_rep1_control_t20 a7_rep2_control_t-5 a8_rep2_+T4_t5 a9_rep2_+T4_t10\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/filter/*.fastq | grep -Eo 'a[0-9]+_.*t.?[0-9]' | sort | uniq )\n",
    "echo $fastq_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools multicov -bams ./yan_results/bam_result/F320/a10_rep2_+T4_t15_F320_sorted.bam ./yan_results/bam_result/F320/a11_rep2_+T4_t20_F320_sorted.bam ./yan_results/bam_result/F320/a12_rep2_control_t20_F320_sorted.bam ./yan_results/bam_result/F320/a13_rep3_control_t-5_F320_sorted.bam ./yan_results/bam_result/F320/a14_rep3_+T4_t2_F320_sorted.bam ./yan_results/bam_result/F320/a15_rep3_+T4_t5_F320_sorted.bam ./yan_results/bam_result/F320/a16_rep3_+T4_t10_F320_sorted.bam ./yan_results/bam_result/F320/a17_rep3_+T4_t20_F320_sorted.bam ./yan_results/bam_result/F320/a18_rep3_control_t20_F320_sorted.bam ./yan_results/bam_result/F320/a1_rep1_control_t-5_F320_sorted.bam ./yan_results/bam_result/F320/a2_rep1_+T4_t5_F320_sorted.bam ./yan_results/bam_result/F320/a3_rep1_+T4_t10_F320_sorted.bam ./yan_results/bam_result/F320/a4_rep1_+T4_t15_F320_sorted.bam ./yan_results/bam_result/F320/a5_rep1_+T4_t20_F320_sorted.bam ./yan_results/bam_result/F320/a6_rep1_control_t20_F320_sorted.bam ./yan_results/bam_result/F320/a7_rep2_control_t-5_F320_sorted.bam ./yan_results/bam_result/F320/a8_rep2_+T4_t5_F320_sorted.bam ./yan_results/bam_result/F320/a9_rep2_+T4_t10_F320_sorted.bam -bed ./yan_results/E_coli_flattened_T1.gtf > ./yan_results/bam_result/F320/count_new.file\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fastq_name=$(ls ./20211105180528381-5936/original/DEFAULT/filter/*.fastq | grep -Eo 'a[0-9]+_.*t.?[0-9]' | sort | uniq )\n",
    "command1=\"Chromosome\\tprotein_coding\\t5UTR\\t4\\t189\\t2735.0\\t+\\t.\\tid_list\\t\"\n",
    "command=\"bedtools multicov -bams \" \n",
    "#GTF_file=\"./yan_results/Escherichia_coli_K12_and_T1_phage_genome.gtf\"\n",
    "#GTF_file=\"./yan_results/T1.gtf\"\n",
    "GTF_file=\"./yan_results/E_coli_flattened_T1.gtf\"\n",
    "\n",
    "path_of_result=\"./yan_results/bam_result/F320\"\n",
    "\n",
    "for ID in $fastq_name\n",
    "    do\n",
    "    command+=$path_of_result\"/\"$ID\"_F320_sorted.bam \" \n",
    "    command1+=$ID\"\\t\"\n",
    "    done\n",
    "     command+=\"-bed $GTF_file > ./yan_results/bam_result/F320/count_new.file\"\n",
    "     #echo -e $command1 > ./yan_results/counts_title.txt\n",
    "    echo $command\n",
    "    eval $command\n",
    "    echo \"sed -i '1i\"$command1\"' ./yan_results/bam_result/F320/count_new.file \" > run.sh\n",
    "    bash run.sh \n",
    "    rm run.sh &\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# edit the count file using linux\n",
    "# sort all the columns \n",
    "cd ./yan_results/bam_result/F320\n",
    "awk '{FS=\"\\t\"; OFS=\"\\t\"; {print $9,$19,$25,$13,$24,$12,$18,$20,$26,$15,$21,$27,$16,$22,$10,$23,$11,$17;}}' count_new.file > sorted_count_Ecoli_T1_name.csv\n",
    "#head sorted_count_Ecoli_T1_name.csv\n",
    "awk '{FS=\"\\t\"; OFS=\";\"; {print $1;}}' sorted_count_Ecoli_T1_name.csv > gene_name.file \n",
    "\n",
    "head -4497 gene_name.file > Ecoli_name.file\n",
    "grep \"AY216660\" gene_name.file > T1_name.file \n",
    "\n",
    "# get rid of all the \"\", only keep the gene name\n",
    "cut -d \";\" -f 2 Ecoli_name.file |cut -d \" \" -f 3|tail -n +2| sed 's/\\\"//g' > Ecoli_gene_name.file\n",
    "\n",
    "cut -d \";\" -f 4 T1_name.file |cut -d \" \" -f 3| sed 's/\\\"//g' > T1_gene_name.file\n",
    "\n",
    "#concatenate the gene name of Ecoli and T1\n",
    "cat Ecoli_gene_name.file T1_gene_name.file > all_gene_name.file\n",
    "\n",
    "#add a column name to all the gene\n",
    "echo \"gene_name\" > header_name.file\n",
    "cat header_name.file all_gene_name.file > sum_gene_name.file\n",
    "\n",
    "cut -f 2- sorted_count_Ecoli_T1_name.csv > no_name.file\n",
    "\n",
    "paste sum_gene_name.file no_name.file > sorted_count_table.csv\n",
    "\n",
    "rm *name*\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Result Datasets, they will hold the selected files and current notebook file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Details of the analysis, please overwrite defaults and complete the description\n",
    "\n",
    "title = 'Analysis of Processing_of_fastq_files' # name your analysis\n",
    "notes = '' # describe/comment this analysis\n",
    "\n",
    "from obis_connector import NewDataSet \n",
    " \n",
    "datasets = [] \n",
    " \n",
    "analysed = NewDataSet( \n",
    "    name = title, \n",
    "    type = 'ANALYZED_DATA', \n",
    "    notes = notes, \n",
    "    files = ['folder_with_files','fileName1'], \n",
    "    parents = resultDatasetParents \n",
    ") \n",
    "datasets.append(analysed) \n",
    " \n",
    " \n",
    "# When set to True the uploaded files will be moved into the default location inside working directory \n",
    "# that way subsequent analisis of OpenBIS DataSets will not need to download them before processing \n",
    "# It is safe to set it to True as the files are not deleted so they can be brought back \n",
    "moveFilesOnSuccess = False \n",
    " \n",
    " \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Code versions and execution environment information\n",
    "### You should always call it before uploading data sets to record versions used\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Please only use scripts and annotations files from these folders so they will be tracked with the analysis\n",
    "SCRIPTS=/homes2/openngs/openngs/scripts2\n",
    "ANNOTATIONS=/homes2/openngs/openngs/annotation_files2\n",
    "NGS_TOOLS=/homes2/openngs/openngs/ngs_tools\n",
    "\n",
    "$SCRIPTS/check_dir_status.sh $SCRIPTS || exit 1\n",
    "$SCRIPTS/check_dir_status.sh $ANNOTATIONS || exit 1\n",
    "$SCRIPTS/check_dir_status.sh $NGS_TOOLS || exit 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# explicitly imported global modules\n",
    "import obis_connector\n",
    "\n",
    "print('global modules:\\n')\n",
    "print('\\n'.join(obis_connector.list_imported_modules(globals())))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save notebook before triggering upload and set upload = True below!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Set upload to True to send data to OpenBIS (remember to save the Notebook first!!!!)\n",
    "upload = False\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### You should not need to change anything below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%javascript\n",
    "\n",
    "require([\"base/js/namespace\"],function(Jupyter) {\n",
    "    Jupyter.notebook.save_checkpoint();\n",
    "});\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# You should not need to change this cell\n",
    "\n",
    "import os\n",
    "import sys\n",
    "import time\n",
    "# Result Root node (selected in ELN dialog)\n",
    "data_root= o.get_object('/OPENNGS/JUP_783')\n",
    "\n",
    "# leave true to have new analysis container (sample/object) for the results DataSet (recommended),\n",
    "# otherwise results become directly linked to data_root node you have selected at ELN dialog\n",
    "new_container = True\n",
    "\n",
    "if upload:\n",
    "\n",
    "    # saving current python environment\n",
    "    list_python_dep = '/homes2/openngs/openngs/scripts2/list_python_dep.sh'\n",
    "    if not os.path.exists(list_python_dep):\n",
    "        raise ValueError('Cannot find the environmet listing script')\n",
    "\n",
    "    dep_file = \"python_env.{}.txt\".format(time.time())\n",
    "    !{list_python_dep} {sys.prefix} {dep_file}\n",
    "\n",
    "    # in try so that the env file is deleted on completion\n",
    "    try:\n",
    "\n",
    "        snapshot = obis_connector.save_snapshot(fileName)\n",
    "        jupyter_files = [fileName, snapshot, dep_file]\n",
    "        analysis_props = {'name':title, 'notes':notes}\n",
    "        \n",
    "        res = obis_connector.upload_analysis_to_obis(datasets, \n",
    "                                  data_root = data_root,\n",
    "                                  jupyter_files = jupyter_files,\n",
    "                                  workspace = workspace, moveFilesOnSuccess = moveFilesOnSuccess,\n",
    "                                  new_container = new_container,\n",
    "                                  object_kind = 'JUPYTER_ANALYSIS', analysis_props = analysis_props)\n",
    "        for ds in res: \n",
    "            print(ds.summary) \n",
    "    finally: \n",
    "        os.remove(dep_file)\n",
    "        os.remove(snapshot)\n",
    "else:\n",
    "    raise ValueError('Save notebook and change upload to True to store results in OpenBIS')\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
